<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Allegra G. Hawkins" />
  <meta name="author" content="Joshua A. Shapiro" />
  <meta name="author" content="Stephanie J. Spielman" />
  <meta name="author" content="David S. Mejia" />
  <meta name="author" content="Deepashree Venkatesh Prasad" />
  <meta name="author" content="Nozomi Ichihara" />
  <meta name="author" content="Arkadii Yakovets" />
  <meta name="author" content="Avrohom M. Gottlieb" />
  <meta name="author" content="Kurt G. Wheeler" />
  <meta name="author" content="Chante J. Bethell" />
  <meta name="author" content="Steven M. Foltz" />
  <meta name="author" content="Jennifer O’Malley" />
  <meta name="author" content="Casey S. Greene" />
  <meta name="author" content="Jaclyn N. Taroni" />
  <meta name="dcterms.date" content="2025-04-23" />
  <meta name="keywords" content="single-cell RNA-seq, single-nuclei RNA-seq, pediatric cancer, resource, Nextflow, open science, reproducibility" />
  <title>The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors" />
  <meta name="citation_title" content="The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors" />
  <meta property="og:title" content="The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors" />
  <meta property="twitter:title" content="The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors" />
  <meta name="dc.date" content="2025-04-23" />
  <meta name="citation_publication_date" content="2025-04-23" />
  <meta property="article:published_time" content="2025-04-23" />
  <meta name="dc.modified" content="2025-04-23T21:06:16+00:00" />
  <meta property="article:modified_time" content="2025-04-23T21:06:16+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Allegra G. Hawkins" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0001-6026-3660" />
  <meta name="citation_author" content="Joshua A. Shapiro" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0002-6224-0347" />
  <meta name="citation_author" content="Stephanie J. Spielman" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0002-9090-4788" />
  <meta name="citation_author" content="David S. Mejia" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0003-1679-0353" />
  <meta name="citation_author" content="Deepashree Venkatesh Prasad" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0001-5756-4083" />
  <meta name="citation_author" content="Nozomi Ichihara" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author" content="Arkadii Yakovets" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author" content="Avrohom M. Gottlieb" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author" content="Kurt G. Wheeler" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author" content="Chante J. Bethell" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_institution" content="The University of Texas MD Anderson Cancer Center, UTHealth Houston Graduate School of Biomedical Sciences, Houston, TX, 77030, USA" />
  <meta name="citation_author_orcid" content="0000-0001-9653-8128" />
  <meta name="citation_author" content="Steven M. Foltz" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_institution" content="Department of Pediatrics, Division of Oncology, Children&#39;s Hospital of Philadelphia, Philadelphia, PA, 19104, USA" />
  <meta name="citation_author_orcid" content="0000-0002-9526-8194" />
  <meta name="citation_author" content="Jennifer O&#39;Malley" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author" content="Casey S. Greene" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_institution" content="Center for Health AI, University of Colorado School of Medicine, Aurora, CO, 80045, USA" />
  <meta name="citation_author_institution" content="Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO, 80045, USA" />
  <meta name="citation_author_orcid" content="0000-0001-8713-9213" />
  <meta name="citation_author" content="Jaclyn N. Taroni" />
  <meta name="citation_author_institution" content="Childhood Cancer Data Lab, Alex&#39;s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA" />
  <meta name="citation_author_orcid" content="0000-0003-4734-4508" />
  <link rel="canonical" href="https://AlexsLemonade.github.io/ScPCA-manuscript/" />
  <meta property="og:url" content="https://AlexsLemonade.github.io/ScPCA-manuscript/" />
  <meta property="twitter:url" content="https://AlexsLemonade.github.io/ScPCA-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://AlexsLemonade.github.io/ScPCA-manuscript/" />
  <meta name="citation_pdf_url" content="https://AlexsLemonade.github.io/ScPCA-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://AlexsLemonade.github.io/ScPCA-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://AlexsLemonade.github.io/ScPCA-manuscript/v/5b170f1ceb79dbf0f64efd7716d725b6d7f8d213/" />
  <meta name="manubot_html_url_versioned" content="https://AlexsLemonade.github.io/ScPCA-manuscript/v/5b170f1ceb79dbf0f64efd7716d725b6d7f8d213/" />
  <meta name="manubot_pdf_url_versioned" content="https://AlexsLemonade.github.io/ScPCA-manuscript/v/5b170f1ceb79dbf0f64efd7716d725b6d7f8d213/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">The Single-cell Pediatric Cancer Atlas: Data portal and open-source tools for single-cell transcriptomics of pediatric tumors</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://AlexsLemonade.github.io/ScPCA-manuscript/v/5b170f1ceb79dbf0f64efd7716d725b6d7f8d213/">permalink</a>)
was automatically generated
from <a href="https://github.com/AlexsLemonade/ScPCA-manuscript/tree/5b170f1ceb79dbf0f64efd7716d725b6d7f8d213">AlexsLemonade/ScPCA-manuscript@5b170f1</a>
on April 23, 2025.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Allegra G. Hawkins</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6026-3660">0000-0001-6026-3660</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/allyhawkins">allyhawkins</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Joshua A. Shapiro</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6224-0347">0000-0002-6224-0347</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jashapiro">jashapiro</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Stephanie J. Spielman</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-9090-4788">0000-0002-9090-4788</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/sjspielman">sjspielman</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>David S. Mejia</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1679-0353">0000-0003-1679-0353</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/davidsmejia">davidsmejia</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Deepashree Venkatesh Prasad</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5756-4083">0000-0001-5756-4083</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/dvenprasad">dvenprasad</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Nozomi Ichihara</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/nozomione">nozomione</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Arkadii Yakovets</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/arkid15r">arkid15r</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Avrohom M. Gottlieb</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/avrohomgottlieb">avrohomgottlieb</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Kurt G. Wheeler</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/kurtwheeler">kurtwheeler</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Chante J. Bethell</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9653-8128">0000-0001-9653-8128</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cbethell">cbethell</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA; The University of Texas MD Anderson Cancer Center, UTHealth Houston Graduate School of Biomedical Sciences, Houston, TX, 77030, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Steven M. Foltz</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-9526-8194">0000-0002-9526-8194</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/envest">envest</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA; Department of Pediatrics, Division of Oncology, Children’s Hospital of Philadelphia, Philadelphia, PA, 19104, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Jennifer O’Malley</strong>
<br>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/Jen-OMalley">Jen-OMalley</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Casey S. Greene</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-8713-9213">0000-0001-8713-9213</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/cgreene">cgreene</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA; Center for Health AI, University of Colorado School of Medicine, Aurora, CO, 80045, USA; Department of Biomedical Informatics, University of Colorado School of Medicine, Aurora, CO, 80045, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
<li><p><strong>Jaclyn N. Taroni</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4734-4508">0000-0003-4734-4508</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/jaclyn-taroni">jaclyn-taroni</a>
<br>
<small>
Childhood Cancer Data Lab, Alex’s Lemonade Stand Foundation, Bala Cynwyd, PA, 19004, USA
· Funded by Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab (CCDL)
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — Correspondence possible via <a href="https://github.com/AlexsLemonade/ScPCA-manuscript/issues">GitHub Issues</a>
or email to
Jaclyn N. Taroni &lt;jaclyn.taroni@ccdatalab.org&gt;.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>The Single-cell Pediatric Cancer Atlas (ScPCA) Portal (<a href="https://scpca.alexslemonade.org/" class="uri">https://scpca.alexslemonade.org/</a>) is a data resource for uniformly processed single-cell and single-nuclei RNA sequencing (RNA-seq) data and de-identified metadata from pediatric tumor samples.
Originally comprised of data from 10 projects funded by Alex’s Lemonade Stand Foundation (ALSF), the Portal currently contains summarized gene expression data for over 700 samples across 55 cancer types from ALSF-funded and community-contributed datasets.
Downloads include gene expression data as <code>SingleCellExperiment</code> or <code>AnnData</code> objects containing raw and normalized counts, PCA and UMAP coordinates, and automated cell type annotations, along with summary reports.
Some samples have additional data from bulk RNA-seq, spatial transcriptomics, and/or feature barcoding (e.g., CITE-seq and cell hashing) included in the download.
All data on the Portal were uniformly processed using <code>scpca-nf</code>, an open-source and efficient Nextflow workflow that uses <code>alevin-fry</code> to quantify gene expression.
Comprehensive documentation, including descriptions of file contents and a guide to getting started, is available at <a href="http://scpca.readthedocs.io" class="uri">http://scpca.readthedocs.io</a>.</p>
<h2 id="introduction">Introduction</h2>
<p>Since the introduction of single-cell RNA-seq technology, the number of studies that employ single-cell RNA-seq has grown rapidly <span class="citation" data-cites="ALU8zyix">[<a href="#ref-ALU8zyix" role="doc-biblioref">1</a>]</span>.
Unlike its predecessor, bulk RNA-seq, which averages the expression profiles of all cells within a sample, single-cell technology quantifies gene expression in individual cells.
Tumors are known to be transcriptionally heterogeneous, so many studies have highlighted the importance of using single-cell RNA-seq in studying tumor samples <span class="citation" data-cites="e6vwOccq">[<a href="#ref-e6vwOccq" role="doc-biblioref">2</a>]</span>.
Researchers can use single-cell RNA-seq of samples obtained from patient tumors to analyze and identify individual cell populations that may play important roles in tumor growth, resistance, and metastasis <span class="citation" data-cites="CK2iL88z">[<a href="#ref-CK2iL88z" role="doc-biblioref">3</a>]</span>.
Additionally, single-cell RNA-seq data provides insight into how tumor cells interact with normal cells in the tumor microenvironment <span class="citation" data-cites="10Z0RGQgO">[<a href="#ref-10Z0RGQgO" role="doc-biblioref">4</a>]</span>.</p>
<p>With the growing number of single-cell RNA-seq datasets, efforts have emerged to create central, harmonized sources for datasets.
Harmonized data resources allow researchers to leverage more samples from various biological contexts to complete their analysis and elucidate previously unknown similarities across samples and disease types.
The Human Cell Atlas (HCA) and Human Tumor Atlas Network (HTAN) are two of many such examples.
The HCA, which aims to use single-cell genomics to provide a comprehensive map of all cell types in the human body <span class="citation" data-cites="vk9ZInF3">[<a href="#ref-vk9ZInF3" role="doc-biblioref">5</a>]</span>, contains uniformly processed single-cell RNA-seq data obtained from normal tissue with few samples derived from diseased tissue.
The HTAN also hosts a collection of genomic data collected from tumors across multiple cancer types, including single-cell RNA-seq <span class="citation" data-cites="udPM4KR1">[<a href="#ref-udPM4KR1" role="doc-biblioref">6</a>]</span>.</p>
<p>Existing resources have focused on making large quantities of harmonized data from normal tissue or adult tumor samples publicly available, but there are considerably fewer efforts to harmonize and distribute data from pediatric tumors.
Pediatric cancer is much less common than adult cancer, so the number of available samples from pediatric tumors is smaller compared to the number of adult tumors <span class="citation" data-cites="Hqoq5ZTI">[<a href="#ref-Hqoq5ZTI" role="doc-biblioref">7</a>]</span> and access to data from pediatric tumors is often limited.
Thus, it is imperative to provide harmonized data from pediatric tumors to all pediatric cancer researchers <span class="citation" data-cites="yEQ2v42v">[<a href="#ref-yEQ2v42v" role="doc-biblioref">8</a>]</span>.
To address this unmet need, Alex’s Lemonade Stand Foundation and the Childhood Cancer Data Lab developed and maintain the Single-cell Pediatric Cancer Atlas (ScPCA) Portal (<a href="https://scpca.alexslemonade.org/" class="uri">https://scpca.alexslemonade.org/</a>), an open-source data resource for single-cell and single-nuclei RNA-seq data of pediatric tumors.</p>
<p>The ScPCA Portal holds uniformly processed summarized gene expression from 10x Genomics droplet-based single-cell and single-nuclei RNA-seq for 700 samples from a diverse set of 55 types of pediatric cancers.
Originally comprised of data from ten projects funded by Alex’s Lemonade Stand Foundation, the Portal has since expanded to include data contributed by pediatric cancer research community members.
In addition to gene expression data from single-cell and single-nuclei RNA-seq, the Portal includes data obtained from bulk RNA-seq, spatial transcriptomics, and feature barcoding methods, such as CITE-seq and cell hashing.
All data provided on the portal are available in formats ready for downstream analysis with common workflow ecosystems such as <code>SingleCellExperiment</code> objects used by <code>R/Bioconductor</code><span class="citation" data-cites="z8JnDHpr">[<a href="#ref-z8JnDHpr" role="doc-biblioref">9</a>]</span> or <code>AnnData</code> objects used by <code>Scanpy</code> and related Python modules <span class="citation" data-cites="qIZTS1II">[<a href="#ref-qIZTS1II" role="doc-biblioref">10</a>]</span>.
Downloaded objects contain normalized gene expression counts, dimensionality reduction results, and cell type annotations.</p>
<p>To ensure that all current and future data on the Portal are uniformly processed, we created <code>scpca-nf</code>, an open-source Nextflow <span class="citation" data-cites="4XDvZWxk">[<a href="#ref-4XDvZWxk" role="doc-biblioref">11</a>]</span> pipeline (<a href="https://github.com/AlexsLemonade/scpca-nf" class="uri">https://github.com/AlexsLemonade/scpca-nf</a>).
Using a consistent pipeline for all data increases transparency and allows users to perform analysis across multiple samples and projects without having to do any re-processing.
The <code>scpca-nf</code> workflow uses <code>alevin-fry</code> <span class="citation" data-cites="hYvj5GBC">[<a href="#ref-hYvj5GBC" role="doc-biblioref">12</a>]</span> for fast and efficient quantification of single-cell gene expression for all samples on the Portal, including single-cell RNA-seq data and any associated CITE-seq or cell hash data.
The <code>scpca-nf</code> pipeline also serves as a resource for the community, allowing others to process their own samples for comparison to samples available on the Portal and submit uniformly processed community contributions to the Portal.</p>
<p>Here, we present the Single-cell Pediatric Cancer Atlas as a resource for all pediatric cancer researchers.
The ScPCA Portal provides downloads ready for immediate use, allowing researchers to skip time-consuming data re-processing and wrangling steps.
We provide comprehensive documentation about data processing and the contents of files on the portal, including a guide to getting started working with an ScPCA dataset (<a href="https://scpca.readthedocs.io/" class="uri">https://scpca.readthedocs.io/</a>).
The ScPCA Portal advances pediatric cancer research by accelerating researchers’ ability to answer important biological questions.</p>
<h1 id="results">Results</h1>
<h2 id="the-single-cell-pediatric-cancer-atlas-portal">The Single-cell Pediatric Cancer Atlas Portal</h2>
<p>In March of 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata from pediatric tumor samples available for download.
Data available on the Portal was obtained using two different mechanisms: raw data was accepted from ALSF-funded investigators and processed using our open-source pipeline <code>scpca-nf</code>, or investigators processed their raw data using <code>scpca-nf</code> and submitted the output for inclusion on the Portal.</p>
<p>All samples on the Portal include a core set of metadata obtained from investigators, including age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
Some investigators submitted additional metadata, such as treatment and tumor stage, which can also be found on the Portal.
All submitted metadata was standardized to maintain consistency across projects before adding to the Portal.
In addition to providing a human-readable value for the submitted metadata, we also provide ontology term identifiers, if applicable.
Submitted metadata was mapped to associated ontology term identifiers obtained from HsapDv (age) <span class="citation" data-cites="2n9clgPF">[<a href="#ref-2n9clgPF" role="doc-biblioref">13</a>]</span>, PATO (sex) <span class="citation" data-cites="Jx3SGqPR 19R1xim7A">[<a href="#ref-Jx3SGqPR" role="doc-biblioref">14</a>,<a href="#ref-19R1xim7A" role="doc-biblioref">15</a>]</span>, NCBI taxonomy (organism) <span class="citation" data-cites="RJLosfFq jFq6Uo6f">[<a href="#ref-RJLosfFq" role="doc-biblioref">16</a>,<a href="#ref-jFq6Uo6f" role="doc-biblioref">17</a>]</span>, MONDO (disease) <span class="citation" data-cites="pZyZllZn ZPHZMiua">[<a href="#ref-pZyZllZn" role="doc-biblioref">18</a>,<a href="#ref-ZPHZMiua" role="doc-biblioref">19</a>]</span>, UBERON (tissue) <span class="citation" data-cites="swLRou1q TOC8jpgh XbjdiPPQ">[<a href="#ref-swLRou1q" role="doc-biblioref">20</a>,<a href="#ref-TOC8jpgh" role="doc-biblioref">21</a>,<a href="#ref-XbjdiPPQ" role="doc-biblioref">22</a>]</span>, and Hancestro (ethnicity, if applicable) <span class="citation" data-cites="9Hno2ty7 Opr5odMH">[<a href="#ref-9Hno2ty7" role="doc-biblioref">23</a>,<a href="#ref-Opr5odMH" role="doc-biblioref">24</a>]</span>.
By providing these ontology term identifiers for each sample, users have access to standardized metadata terms that facilitate comparisons among datasets within the Portal as well as to data from other research projects.</p>
<p>The Portal contains data from 700 samples and 55 tumor types <span class="citation" data-cites="9F0oWAO0 HalHlba8 TffkWyy0 sgFUrpYG PJ7Veojz U3XexzPk 9Xh0e2DI">[<a href="#ref-9F0oWAO0" role="doc-biblioref">25</a>,<a href="#ref-HalHlba8" role="doc-biblioref">26</a>,<a href="#ref-TffkWyy0" role="doc-biblioref">27</a>,<a href="#ref-sgFUrpYG" role="doc-biblioref">28</a>,<a href="#ref-PJ7Veojz" role="doc-biblioref">29</a>,<a href="#ref-U3XexzPk" role="doc-biblioref">30</a>,<a href="#ref-9Xh0e2DI" role="doc-biblioref">31</a>]</span>.
<!-- TODO: Update numbers -->
Figure <a href="#fig:fig1">1</a>A summarizes all samples from patient tumors and patient-derived xenografts currently available on the Portal.
The total number of samples for each diagnosis is shown, along with the proportion of samples from each disease stage within a diagnosis group.
The largest number of samples found on the Portal were obtained from patients with leukemia (n = 216).
The Portal also includes samples from sarcoma and soft tissue tumors (n = 194), brain and central nervous system tumors (n = 167), and a variety of other solid tumors (n = 115).
Most samples were collected at initial diagnosis (n = 520), with a smaller number of samples collected either at recurrence (n = 129), during progressive disease (n = 12), during or after treatment (n = 11), or post-mortem (n = 5).
Along with the patient tumors, the Portal contains a small number of human tumor cell line samples (n = 6).</p>
<p>Each of the available samples contains summarized gene expression data from either single-cell or single-nuclei RNA sequencing.
However, some samples also include additional data, such as quantified expression data from tagging cells with antibody-derived tags (ADT), such as CITE-seq antibodies <span class="citation" data-cites="abp5rbw2">[<a href="#ref-abp5rbw2" role="doc-biblioref">32</a>]</span>, or multiplexing samples with hashtag oligonucleotides (HTO) <span class="citation" data-cites="FZrs73Vp">[<a href="#ref-FZrs73Vp" role="doc-biblioref">33</a>]</span> prior to sequencing.
Out of the 518 samples, 96 have associated CITE-seq data, and 19 have associated multiplexing data.
In some cases, multiple libraries from the same sample were collected for additional sequencing, either for bulk RNA-seq or spatial transcriptomics.
Specifically, 118 samples on the Portal were sequenced using bulk RNA-seq and 94 samples were sequenced using spatial transcriptomics.
A summary of the number of samples with each additional modality is shown in Figure <a href="#fig:fig1">1</a>B, and a detailed summary of the total samples with each sequencing method broken down by project is available in Table S1.</p>
<p>Samples on the Portal are organized by project, where each project is a collection of similar samples from an individual lab.
Users can filter projects based on diagnosis, included modalities (e.g., CITE-seq, bulk RNA-seq), 10x Genomics kit version (e.g., 10Xv2, 10Xv3), and whether or not a project includes samples derived from patient-derived xenografts or cell lines.
The project card displays an abstract, the total number of samples included, a list of diagnoses for all samples included in the Project, and links to any external information associated with the project, such as publications and links to external data, such as SRA or GEO (Figure <a href="#fig:fig1">1</a>C).
The project card also indicates the type(s) of sequencing performed, including the 10x Genomics kit version, the suspension type (cell or nucleus), and if additional sequencing is present, like bulk RNA-seq or multiplexing.</p>
<h2 id="uniform-processing-of-data-available-on-the-scpca-portal">Uniform processing of data available on the ScPCA Portal</h2>
<p>We developed <a href="https://github.com/AlexsLemonade/scpca-nf"><code>scpca-nf</code></a>, an open-source and efficient Nextflow <span class="citation" data-cites="4XDvZWxk">[<a href="#ref-4XDvZWxk" role="doc-biblioref">11</a>]</span> workflow for quantifying single-cell and single-nuclei RNA-seq data and processed all data available on the Portal with it.
Using Nextflow as the backbone for the <code>scpca-nf</code> workflow ensures both reproducibility and portability.
All dependencies for the workflow are handled automatically, as each process in the workflow is run in a Docker container.
Nextflow is compatible with various computing environments, including high-performance computing clusters and cloud-based computing, allowing users to run the workflow in their preferred environment.
Setup requires organizing input files and updating a single configuration file for the computing environment after installing Nextflow and either Docker or Singularity.
Nextflow will also handle parallelizing sample processing as allowed by the environment, minimizing run time.
The combination of being able to execute a Nextflow workflow in any environment and run individual processes in Docker containers makes this workflow easily portable for external use.</p>
<p>When building <code>scpca-nf</code>, we sought a fast and memory-efficient tool for gene expression quantification to minimize processing costs.
We expected many users of the Portal to have their own single-cell or single-nuclei data processed with Cell Ranger <span class="citation" data-cites="1CuNC7bWt 87PPOPmp">[<a href="#ref-1CuNC7bWt" role="doc-biblioref">34</a>,<a href="#ref-87PPOPmp" role="doc-biblioref">35</a>]</span>, due to its popularity.
Thus, selecting a tool with comparable results to Cell Ranger was also desirable.
In comparing <code>alevin-fry</code> <span class="citation" data-cites="hYvj5GBC">[<a href="#ref-hYvj5GBC" role="doc-biblioref">12</a>]</span> to Cell Ranger, we found <code>alevin-fry</code> had a lower run time and memory usage (Figure <a href="#fig:figS1">S1</a>A), while retaining comparable mean gene expression for all genes (Figure <a href="#fig:figS1">S1</a>B), total UMIs per cell (Figure <a href="#fig:figS1">S1</a>C), and total genes detected per cell (Figure <a href="#fig:figS1">S1</a>D).
(All analyses comparing gene expression quantification tools are available in a public analysis repository <span class="citation" data-cites="159Z7ABXf">[<a href="#ref-159Z7ABXf" role="doc-biblioref">36</a>]</span>.)
Based on these results, we elected to use <code>salmon alevin</code> and <code>alevin-fry</code> <span class="citation" data-cites="hYvj5GBC">[<a href="#ref-hYvj5GBC" role="doc-biblioref">12</a>]</span> in <code>scpca-nf</code> to quantify gene expression data.</p>
<p><code>scpca-nf</code> takes FASTQ files as input (Figure <a href="#fig:fig2">2</a>A).
Reads are aligned using the selective alignment option in <code>salmon alevin</code> to an index with transcripts corresponding to spliced cDNA and intronic regions, denoted by <code>alevin-fry</code> as a <code>splici</code> index.
The output from <code>alevin-fry</code> includes a gene by cell count matrix for all barcodes identified, even those that may not contain true cells.
This unfiltered counts matrix is stored in a <code>SingleCellExperiment</code> object <span class="citation" data-cites="z8JnDHpr">[<a href="#ref-z8JnDHpr" role="doc-biblioref">9</a>]</span> and output from the workflow as a file with the suffix <code>_unfiltered.rds</code>.
<!-- JAS: I tend to think we should remove the file names and extensions from the Results: they make sense to me in methods but seem a bit deep in the weeds for this section. I simplified them a bit, so I am happier now. Leaving this comment for later reviewers. --></p>
<p><code>scpca-nf</code> performs filtering of empty droplets, removal of low-quality cells, normalization, dimensionality reduction, and cell type annotation (Figure <a href="#fig:fig2">2</a>A).
The unfiltered gene by cell counts matrices are filtered to remove any barcodes that are not likely to contain cells using <code>DropletUtils::emptyDropsCellRanger()</code><span class="citation" data-cites="U2Fa25dp">[<a href="#ref-U2Fa25dp" role="doc-biblioref">37</a>]</span>, and all cells that pass are saved in a <code>SingleCellExperiment</code> object and a file with the suffix <code>_filtered.rds</code>.
Low-quality cells are identified and removed with <code>miQC</code> <span class="citation" data-cites="17Yhk1HAt">[<a href="#ref-17Yhk1HAt" role="doc-biblioref">38</a>]</span>, which jointly models the proportion of mitochondrial reads and detected genes per cell and calculates a probability that each cell is compromised.
The remaining cells’ counts are normalized <span class="citation" data-cites="vpU0FTFY">[<a href="#ref-vpU0FTFY" role="doc-biblioref">39</a>]</span>, and reduced-dimension representations are calculated using both principal component analysis (PCA) and uniform manifold approximation and projection (UMAP) <span class="citation" data-cites="157h5hA34">[<a href="#ref-157h5hA34" role="doc-biblioref">40</a>]</span>.
Finally, cell types are classified using two automated methods, <code>SingleR</code> <span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span> and <code>CellAssign</code> <span class="citation" data-cites="HES0WO6k">[<a href="#ref-HES0WO6k" role="doc-biblioref">42</a>]</span>.
The results from this analysis are stored in a processed <code>SingleCellExperiment</code> object saved to a file with the suffix <code>_processed.rds</code>.</p>
<p>To make downloading from the Portal convenient for R and Python users, downloads are available as either <code>SingleCellExperiment</code> or <code>AnnData</code><span class="citation" data-cites="KRBW8xeZ">[<a href="#ref-KRBW8xeZ" role="doc-biblioref">43</a>]</span> objects.
<code>scpca-nf</code> converts all <code>SingleCellExperiment</code> objects to <code>AnnData</code> objects, which are saved as <code>.h5ad</code> files (Figure <a href="#fig:fig2">2</a>A).
Downloads contain the unfiltered, filtered, and processed objects from <code>scpca-nf</code> to allow users to choose to perform their own filtering and normalization or to start their analysis from a processed object.</p>
<p>All downloads from the Portal include a quality control (QC) report with a summary of processing information (e.g., <code>alevin-fry</code> version), library statistics (e.g., the total number of cells), and a collection of diagnostic plots for each library (Figure <a href="#fig:fig2">2</a>B-G).
A knee plot displaying total UMI counts for all droplets (i.e., including empty droplets) indicates the effects of the empty drop filtering (Figure <a href="#fig:fig2">2</a>B).
For each cell that remains after filtering empty droplets, the number of total UMIs, genes detected, and mitochondrial reads are calculated and summarized in a scatter plot (Figure <a href="#fig:fig2">2</a>C).
We include plots showing the <code>miQC</code> model and which cells are kept and removed after filtering with <code>miQC</code> (Figure <a href="#fig:fig2">2</a>D-E).
A UMAP plot with cells colored by the total number of genes detected and a faceted UMAP plot where cells are colored by the expression of a set of highly variable genes are also provided (Figure <a href="#fig:fig2">2</a>F-G).</p>
<h2 id="processing-samples-with-additional-modalities">Processing samples with additional modalities</h2>
<p><code>scpca-nf</code> includes modules for processing samples with sequencing modalities beyond single-cell or single-nuclei RNA-seq data: corresponding ADT or CITE-seq data <span class="citation" data-cites="abp5rbw2">[<a href="#ref-abp5rbw2" role="doc-biblioref">32</a>]</span>, multiplexed data via cell hashing <span class="citation" data-cites="FZrs73Vp">[<a href="#ref-FZrs73Vp" role="doc-biblioref">33</a>]</span>, spatial transcriptomics, or bulk RNA-seq.</p>
<h3 id="antibody-derived-tags">Antibody-derived tags</h3>
<p>To process ADT libraries, the ADT FASTQ files were provided as input into <code>scpca-nf</code> and quantified using <code>salmon alevin</code> and <code>alevin-fry</code> (Figure <a href="#fig:figS2">S2</a>A).
Along with the FASTQ files, <code>scpca-nf</code> takes a tab-separated values (TSV) file with one row for each ADT – containing the name used for the ADT and associated barcode – required to build an ADT-specific index for quantifying ADT expression with <code>alevin-fry</code>.
The output from <code>alevin-fry</code> is the unfiltered ADT by cell counts matrix.
The ADT by cell counts matrix is read into R alongside the gene by cell counts matrix and saved as an alternative experiment (<code>altExp</code>) within the main <code>SingleCellExperiment</code> object containing the unfiltered RNA counts.
This <code>SingleCellExperiment</code> object containing both RNA and ADT counts is output from the workflow to a file with the suffix <code>_unfiltered.rds</code>.</p>
<p><code>scpca-nf</code> does not filter any cells based on ADT expression or remove cells with low-quality ADT expression.
Any cells removed after filtering empty droplets based on the unfiltered RNA counts matrix are also removed from the ADT counts matrix.
The workflow calculates QC statistics for ADT counts using <code>DropletUtils::cleanTagCounts()</code> that are stored alongside the ADT by cell counts matrix in the filtered <code>SingleCellExperiment</code> object.
The <code>SingleCellExperiment</code> object containing the filtered RNA and ADT counts matrix and associated ADT QC statistics is saved to a file with the suffix <code>_filtered.rds</code>.</p>
<p>The ADT by cell counts matrix is normalized by first determining the ambient profile and then using that profile to calculate median size factors with <code>scuttle::computeMedianFactors()</code> <span class="citation" data-cites="mX3LKFI8 4kYmrOnP">[<a href="#ref-mX3LKFI8" role="doc-biblioref">44</a>,<a href="#ref-4kYmrOnP" role="doc-biblioref">45</a>]</span>.
We skip normalization for cells with low-quality ADT expression, as indicated by <code>DropletUtils::cleanTagCounts()</code>.
Although <code>scpca-nf</code> normalizes ADT counts, the workflow does not perform any dimensionality reduction of ADT data; only the RNA counts data are used as input for dimensionality reduction.
The normalized ADT data are saved as an <code>altExp</code> within the processed <code>SingleCellExperiment</code> containing the normalized RNA data and is output to a file with the suffix <code>_processed.rds</code>.
All files containing <code>SingleCellExperiment</code> objects and associated <code>altExp</code> objects are converted to <code>AnnData</code> objects and exported as separate RNA (<code>_rna.h5ad</code>) and ADT (<code>_adt.h5ad</code>) <code>AnnData</code> objects.</p>
<p>If a library contains associated ADT data, the QC report output by <code>scpca-nf</code> will include an additional section with a summary of ADT-related statistics, such as how many cells express each ADT, and ADT-specific diagnostic plots (Figure <a href="#fig:figS2">S2</a>B-D).
As mentioned above, <code>scpca-nf</code> uses <code>DropletUtils::cleanTagCounts()</code> to calculate QC statistics for each cell using ADT expression but does not filter any cells from the object.
We include plots summarizing the potential effects of removing of low-quality cells based on RNA and ADT counts in the QC report (Figure <a href="#fig:figS2">S2</a>B).
The first quadrant indicates which cells would be kept if the object was filtered using both RNA and ADT quality measures.
The other facets highlight which cells would be removed if filtering was done using only RNA counts, only ADT counts, or both.
The top four ADTs with the most variable expression are also identified and visualized using density plots to show the normalized ADT expression across all cells (Figure <a href="#fig:figS2">S2</a>C) and UMAPs – calculated from RNA data – with cells colored by ADT expression (Figure <a href="#fig:figS2">S2</a>D).</p>
<h3 id="multiplexed-libraries">Multiplexed libraries</h3>
<p>To process multiplexed libraries, the HTO FASTQ files are input to <code>scpca-nf</code> and quantified using <code>salmon alevin</code> and <code>alevin-fry</code> (Figure <a href="#fig:figS2">S2</a>C).
Along with the FASTQ files, <code>scpca-nf</code> requires two TSV files to process multiplexed data: one to build an HTO-specific index for quantifying HTO expression with <code>alevin-fry</code>, and a second to indicate which HTO was used for which sample when multiplexing the library.
The unfiltered HTO by cell counts matrix output from <code>alevin-fry</code> is saved as an alternative experiment (<code>altExp</code>) within the main <code>SingleCellExperiment</code> containing the unfiltered RNA counts.
This <code>SingleCellExperiment</code> object containing both RNA and HTO counts is output from the workflow to a file with the suffix <code>_unfiltered.rds</code>.</p>
<p>As with ADT data, <code>scpca-nf</code> does not filter any cells based on HTO expression, and any cells removed after filtering empty droplets based on the unfiltered RNA counts matrix are also removed from the HTO counts matrix with the remainder saved to a file with the <code>_filtered.rds</code> suffix.
<code>scpca-nf</code> does not perform any additional filtering or processing of the HTO by cell counts matrix, so the same filtered matrix is saved to the file with the <code>_processed.rds</code> suffix.</p>
<p>Although <code>scpca-nf</code> quantifies the HTO data and includes an HTO by cell counts matrix in all objects, <code>scpca-nf</code> does not demultiplex the samples into one sample per library.
Instead, <code>scpca-nf</code> applies multiple demultiplexing methods, including demultiplexing with <code>DropletUtils::hashedDrops()</code> <span class="citation" data-cites="NqZ0Pm2u">[<a href="#ref-NqZ0Pm2u" role="doc-biblioref">46</a>]</span>, demultiplexing with <code>Seurat::HTODemux()</code> <span class="citation" data-cites="FZrs73Vp">[<a href="#ref-FZrs73Vp" role="doc-biblioref">33</a>]</span>, and genetic demultiplexing when bulk RNA-seq data are available.
<code>scpca-nf</code> uses the genetic demultiplexing method described in Weber <em>et al.</em> <span class="citation" data-cites="zLp25QYu">[<a href="#ref-zLp25QYu" role="doc-biblioref">47</a>]</span>, which uses bulk RNA-seq as a reference for the expected genotypes found in each single-cell RNA-seq sample.
The results from all available demultiplexing methods are saved in the filtered and processed <code>SingleCellExperiment</code> objects.</p>
<p>If a library has associated HTO data, an additional section is included in the <code>scpca-nf</code> QC report.
This section summarizes HTO-specific library statistics, such as how many cells express each HTO.
No additional plots are produced, but a table summarizing the results from all three demultiplexing methods is included.</p>
<h3 id="bulk-and-spatial-transcriptomics">Bulk and spatial transcriptomics</h3>
<p>Some samples also included data from bulk RNA-seq and/or spatial transcriptomics libraries.
Both of these additional sequencing methods are supported by <code>scpca-nf</code>.
To quantify bulk RNA-seq data, <code>scpca-nf</code> takes bulk FASTQ files as input, trims reads using <code>fastp</code> <span class="citation" data-cites="pt9vlq4w">[<a href="#ref-pt9vlq4w" role="doc-biblioref">48</a>]</span>, and then aligns and quantifies reads with <code>salmon</code> (Figure <a href="#fig:figS3">S3</a>A) <span class="citation" data-cites="vrqQcFyx">[<a href="#ref-vrqQcFyx" role="doc-biblioref">49</a>]</span>.
The output is a single TSV file with the gene by sample counts matrix for all samples in a given ScPCA project.
This gene by sample matrix is only included with project downloads on the Portal.</p>
<p>To quantify spatial transcriptomics data, <code>scpca-nf</code> takes the RNA FASTQ and slide image as input (Figure <a href="#fig:figS3">S3</a>B).
As <code>alevin-fry</code> does not yet fully support spatial transcriptomics data, <code>scpca-nf</code> uses Space Ranger to quantify all spatial transcriptomics data <span class="citation" data-cites="xo912CsO">[<a href="#ref-xo912CsO" role="doc-biblioref">50</a>]</span>.
The output includes the spot by gene matrix along with a summary report produced by Space Ranger.</p>
<h2 id="downloading-projects-from-the-scpca-portal">Downloading projects from the ScPCA Portal</h2>
<!-- TODO:
⚠️ JNT thinks this level of repetition of facts covered in earlier sections is good.
See what you think during your review.
-->
<p>On the Portal, users can select to download data from individual samples or all data from an entire ScPCA project.
When downloading data for an entire project, users can choose between receiving the individual files for each sample (default) or one file containing the gene expression data and metadata for all samples in the project as a merged object.
Users also have the option to choose their desired format and receive the data as <code>SingleCellExperiment</code> (<code>.rds</code>) or <code>AnnData</code> (<code>.h5ad</code>) objects.</p>
<p>For downloads with samples as individual files, the download folder will include a sub-folder for each sample in the project (Figure <a href="#fig:fig3">3</a>A).
Each sample folder contains all three object types (unfiltered, filtered, and processed) in the requested file format and the QC and cell type summary report for all libraries from the given sample.
The objects house the summarized gene expression data and associated metadata for the library indicated in the filename.</p>
<p>All project downloads include a metadata file, <code>single_cell_metadata.tsv</code>, containing relevant metadata for all samples, and a <code>README.md</code> with information about the contents of each download, contact and citation information, and terms of use for data downloaded from the Portal (Figure <a href="#fig:fig3">3</a>A-B).
If the ScPCA project includes samples with bulk RNA-seq, two additional files are included: a gene by sample counts matrix (<code>bulk_quant.tsv</code>) with the quantified gene expression data for all samples in the project, and a metadata file (<code>bulk_metadata.tsv</code>).</p>
<h3 id="merged-objects">Merged objects</h3>
<p>Providing data for all samples within a single file facilitates performing joint gene-level analyses, such as differential expression or gene set enrichment analyses, on multiple samples simultaneously.
Therefore, we provide a single, merged object for each project containing all raw and normalized gene expression data and metadata for all single-cell and single-nuclei RNA-seq libraries within a given ScPCA project.
We provide merged objects for all projects in the Portal except for those with multiplexing, due to potential ambiguity in identifying samples across multiplexed libraries.
The data in the merged object has simply been combined without further processing; no batch-corrected or integrated data are included.
If downloading data from an ScPCA project as a single, merged file, the download will include a single <code>.rds</code> or <code>.h5ad</code> file, a summary report for the merged object, and a folder with all individual QC and cell type reports for each library found in the merged object (Figure <a href="#fig:fig3">3</a>B).</p>
<p>To build the merged objects, we created an additional stand-alone workflow for merging the output from <code>scpca-nf</code>, <code>merge.nf</code> (Figure <a href="#fig:fig3">3</a>C).
<code>merge.nf</code> takes as input the processed <code>SingleCellExperiment</code> objects output by <code>scpca-nf</code> for all single-cell and single-nuclei libraries included in a given ScPCA project.
The gene expression data stored in all <code>SingleCellExperiment</code> objects are then merged to produce a single merged gene by cell counts matrix containing all cells from all libraries.
The genes available in the merged object will be the same as those in each individual object, as all objects on the Portal were quantified using the same index.
Where possible, library-, cell- and gene-specific metadata found in the individual processed <code>SingleCellExperiment</code> objects are also merged.
The merged normalized counts matrix is then used to select high-variance genes in a library-aware manner before performing dimensionality reduction with both PCA and UMAP.
<code>merge.nf</code> outputs the merged and processed object as a <code>SingleCellExperiment</code> object.
The more samples that are included in a merged object, the larger the object, and the more difficult it is to work with that object in R or Python.
Therefore, we do not provide merged objects for projects with more than 100 samples.</p>
<p>We also account for additional modalities in <code>merge.nf</code>.
If at least one library in a project contains ADT data, the raw and normalized ADT data are also merged and saved as an <code>altExp</code> in the merged <code>SingleCellExperiment</code> object.
If any libraries in a project are multiplexed, no merged object is created, as there is no guarantee that a unique HTO was used for each sample in a given project.
All merged <code>SingleCellExperiment</code> objects are converted to <code>AnnData</code> objects and exported as <code>.h5ad</code> files.
If the merged object contains an <code>altExp</code> with merged ADT data, two <code>AnnData</code> objects are exported to create separate RNA (<code>_rna.h5ad</code>) and ADT (<code>_adt.h5ad</code>) objects.</p>
<p><code>merge.nf</code> outputs a summary report for each merged object, which includes a set of tables summarizing the types of samples and libraries included in the project, such as types of diagnosis, and a faceted UMAP showing all cells from all libraries.
In the UMAP, each panel represents a different library included in the merged object, with all cells from the specified library shown in color, while all other cells are gray.
An example of this UMAP showing a subset of libraries from an ScPCA project is available in Figure <a href="#fig:fig3">3</a>D.</p>
<h2 id="annotating-cell-types">Annotating cell types</h2>
<p>Assigning cell type labels to single-cell and single-nuclei RNA-seq data is often an essential step in analysis.
Cell type annotation requires knowledge of the expected cell types in a dataset and the associated gene expression patterns for each cell type, which is available in publications or other public databases for some biological contexts.
Automated cell type annotation methods leveraging public databases are an excellent initial step in the labeling process, as they can be applied consistently and transparently across all samples in a data set.
As such, we include cell type annotations determined using two different automated methods, <code>SingleR</code> <span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span> and <code>CellAssign</code> <span class="citation" data-cites="HES0WO6k">[<a href="#ref-HES0WO6k" role="doc-biblioref">42</a>]</span>, in all processed <code>SingleCellExperiment</code> and <code>AnnData</code> objects available for download on the Portal, saving users analysis time.</p>
<p>Annotating cell types with automated methods like <code>SingleR</code> and <code>CellAssign</code> requires the use of previously annotated reference data.
For <code>SingleR</code>, this can be in the form of an annotated gene expression dataset from a microarray, bulk RNA-seq, or single-cell RNA-seq experiment.
<code>CellAssign</code> requires a matrix of cell types and expected marker genes.
Most public annotated reference datasets that can be used with these methods – including those we use for the Portal – are derived from normal tissue, making accurately annotating tumor datasets particularly difficult.
Comparing the two methods and observing consistent cell type annotations across methods can indicate higher confidence in the provided labels, so we created a set of ontology-aware rules to assign consensus cell type labels based on the agreement between <code>SingleR</code> and <code>CellAssign</code>.
These consensus cell type assignments can be found in all processed <code>SingleCellExperiment</code> and <code>AnnData</code> objects on the Portal.</p>
<p>For some ScPCA projects, submitters provided their own curated cell type annotations, including annotation of tumor cells and disease-specific cell states.
These submitter-provided annotations can be found in all <code>SingleCellExperiment</code> and <code>AnnData</code> objects (unfiltered, filtered, and processed).</p>
<h3 id="choosing-cell-typing-methods-and-references">Choosing cell typing methods and references</h3>
<p><code>SingleR</code> is a reference-based annotation method that requires an existing bulk or single-cell RNA-seq dataset with annotations.
To identify an appropriate reference to use with <code>SingleR</code>, we annotated a small number of samples across multiple disease types with all human-specific references available in the <code>celldex</code> package <span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span>.
The output from <code>SingleR</code> includes a score matrix containing a score for each cell and all possible cell types found in the reference, where higher scores are associated with assigned cell types.
We calculated the delta median statistic for each cell in the dataset by subtracting the median score from the score associated with the assigned cell type label.
The delta median statistic helps evaluate how confident <code>SingleR</code> is in assigning each cell to a specific cell type, where low delta median values indicate ambiguous assignments and high delta median values indicate confident assignments <span class="citation" data-cites="xpZq9dBF">[<a href="#ref-xpZq9dBF" role="doc-biblioref">51</a>]</span>.
<!-- TODO: ⚠️ For review - What do you think of the next sentence? -->
Using this measure, we found that the <code>BlueprintEncodeData</code> reference <span class="citation" data-cites="MzCvXTgB 15J98V2qM">[<a href="#ref-MzCvXTgB" role="doc-biblioref">52</a>,<a href="#ref-15J98V2qM" role="doc-biblioref">53</a>]</span>, which includes a variety of normal cell types, tended to perform better than or at least similarly to other references across samples from different disease types (Figure <a href="#fig:figS4">S4</a>).
Based on these findings, we used the <code>BlueprintEncodeData</code> reference to annotate cells from all libraries on the Portal, as using a single reference is potentially valuable for cross-project analyses.</p>
<p>In contrast, <code>CellAssign</code> is a marker-gene-based annotation method that requires a binary matrix with all cell types and all associated marker genes as the reference.
We used the list of marker genes available as part of <code>PanglaoDB</code> <span class="citation" data-cites="T3PlZ3Vh">[<a href="#ref-T3PlZ3Vh" role="doc-biblioref">54</a>]</span> to construct organ-specific marker gene matrices with marker genes from all cell types listed for the specified organ.
Since many cancers may have infiltrating immune cells, all immune cells were also included in each organ-specific reference.
For each ScPCA project, we provided the organ-specific marker gene matrix relevant to the disease and tissue type from which the sample was obtained (e.g., for brain tumors, we used a brain-specific marker gene matrix with all brain and immune cell types).
If <code>CellAssign</code> cannot find a likely cell type from the marker gene matrix, it does not assign a cell type.
Because we annotate cells from tumor samples using references containing only normal cells, we anticipate that many cells, particularly the tumor cells, may not have an exact match; reporting this to the end user is valuable.
Indeed, when applying <code>CellAssign</code> to tumor samples with our chosen reference, we observed that many of the cells were unassigned.
We included an example in Figure <a href="#fig:figS5">S5</a>A where unassigned cell types are labeled with <code>Unknown</code>.
When comparing annotations obtained from <code>CellAssign</code> to <code>SingleR</code> annotations and submitter-provided annotations, we noticed the labels for non-tumor cells were similar between <code>CellAssign</code>, <code>SingleR</code>, and submitter annotations, while the tumor cells were not assigned using <code>CellAssign</code> (Figure <a href="#fig:figS5">S5</a>B).</p>
<h3 id="adding-cell-type-annotations-to-the-scpca-portal">Adding cell type annotations to the ScPCA Portal</h3>
<p><code>scpca-nf</code> adds cell type annotations from <code>SingleR</code> and <code>CellAssign</code> to all processed <code>SingleCellExperiment</code> objects (Figure <a href="#fig:fig4">4</a>A).
This requires two additional reference files as input to the workflow: a classification model built from a reference dataset for <code>SingleR</code> and a marker gene by cell type matrix for <code>CellAssign</code>.
<code>SingleR::trainSingleR()</code> was used to build a classification model from the provided <code>BlueprintEncodeData</code> dataset and create the required <code>SingleR</code> input for <code>scpca-nf</code>.
The classification model and processed <code>SingleCellExperiment</code> were used as input for <code>SingleR::classifySingleR()</code>, resulting in annotations for all cells and an associated score matrix.
The score matrix containing a score for all cells and each possible cell type and the assigned cell types are added to the processed <code>SingleCellExperiment</code> object output by <code>scpca-nf</code>.
Simultaneously, processed <code>SingleCellExperiment</code> objects are converted to <code>AnnData</code> objects for classification with <code>CellAssign</code>.
<code>CellAssign</code> uses the converted <code>AnnData</code> object and the marker gene matrix to train a model and predict the most likely cell type from the possible cell types in the marker gene matrix.
The prediction matrix, which contains a probability that each cell is one of each possible cell types, and the assigned cell types are added to the processed <code>SingleCellExperiment</code> object output by <code>scpca-nf</code>.
The processed <code>SingleCellExperiment</code> object is then converted to an <code>AnnData</code> object to ensure cell type annotations are included in both data formats provided by <code>scpca-nf</code>.</p>
<p>An additional cell type report with information about reference sources, comparisons among cell type annotation methods, and diagnostic plots is also output by <code>scpca-nf</code>.
Tables summarizing the number of cells assigned to each cell type for each method are shown alongside UMAPs coloring cells by the assigned cell type.
The concordance of cell type annotations assigned between both methods can indicate higher confidence in the provided annotations.
We therefore used the Jaccard similarity index to compare annotations between the two methods, as well as submitter-provided annotations, if available.
This index is calculated between pairs of labels from each method and ranges from 0-1, with a value close to 1 indicating high agreement and a high proportion of overlapping cells and values close to 0 indicating a low proportion of overlapping cells.
The Jaccard similarity index is displayed in a heatmap, an example of which is shown in Figure <a href="#fig:fig4">4</a>B.</p>
<p>The report also includes a diagnostic plot evaluating the confidence of cell type annotations determined by each method.
To evaluate confidence in <code>SingleR</code> cell type annotations, the delta median statistic is calculated by subtracting the median score from the score associated with the assigned cell type label <span class="citation" data-cites="xpZq9dBF">[<a href="#ref-xpZq9dBF" role="doc-biblioref">51</a>]</span>.
The distribution of delta median values for each cell type is shown in the cell type report, where a higher delta median statistic for a cell indicates higher confidence in the final cell type annotation.
<code>CellAssign</code> calculates the probability that each cell belongs to each possible cell type provided in the reference, and the cell type label with the highest probability is assigned as the cell type for that cell.
<!-- TODO: What exactly do we mean by reliable labels? Labels that are appropriate for the dataset -->
These values range from 0 to 1, with larger values indicating greater confidence in a given cell type label, so we expect more confident labels to have most values close to 1.
A plot displaying the distribution of all probabilities for each cell type is included in the cell type report.</p>
<p>If the submitter provided cell types, the submitter annotations are compared to the annotations from both <code>SingleR</code> and <code>CellAssign</code>.
A summary of this comparison is included in the cell type report along with a table summarizing the submitter cell type annotations and a UMAP plot where each cell is colored by the submitter annotation.
The Jaccard similarity index is calculated for all pairs of cell type labels in submitter annotations and <code>SingleR</code> annotations and in submitter annotations and <code>CellAssign</code> annotations.
The results from both comparisons are displayed in a stacked heatmap available in the report, an example of which is shown in Figure <a href="#fig:figS5">S5</a>B.</p>
<h3 id="assigning-consensus-cell-types">Assigning consensus cell types</h3>
<p><code>SingleR</code> and <code>CellAssign</code> use different references and distinct computational approaches to label cells.
We expect cells with the same or similar cell type labels using both methods will likely be more accurately annotated.
Because of this, we assigned consensus cell type labels when we observed agreement between the two automated methods.
To account for different levels of granularity in reference datasets, we employed an ontology-based approach to assign a consensus cell type label.
Specifically, the consensus cell type annotation is equivalent to the latest common ancestor (LCA) <span class="citation" data-cites="17SbnOJiL">[<a href="#ref-17SbnOJiL" role="doc-biblioref">55</a>]</span> shared between the two predicted cell types.
To ensure specificity in our consensus labels, cells were only assigned a consensus cell type if the identified LCA had no more than 170 descendant terms, with a few exceptions (see Methods for more details).
This threshold was chosen to exclude overly general cell ontology terms, such as lymphocyte, while retaining meaningful classifications like T cell and B cell.
After assigning all consensus cell types, we looked at the expression of cell-type specific marker genes across all cells to validate our assignments (Figure <a href="#fig:fig5">5</a>A, Figure <a href="#fig:figS6">S6</a>).</p>
<p>The addition of the consensus cell type label provides a harmonized cell type annotation for all 700 samples in the ScPCA Portal, making it easy for users to perform downstream analyses across multiple samples.
Consensus annotations can be particularly useful when examining samples from multiple projects submitted by different investigators.
The availability of uniformly processed gene expression data with provided consensus annotations makes it easy to analyze and draw conclusions across a large set of samples.
For example, we show the distribution of cell types observed in all high-grade and low-grade glioma samples in Figure <a href="#fig:fig5">5</a>B, which originate from six different projects and four different investigators.
Here, we can identify similar cell types across all glioma samples, but the composition of cell types present in each sample is heterogeneous.</p>
<p>Previous studies have characterized the glioma immune microenvironment as being predominantly composed of myeloid cells, including microglia and glioma-associated macrophages, with smaller proportions of lymphocytes such as T cells <span class="citation" data-cites="yxJiICJJ 9U8b91g7">[<a href="#ref-yxJiICJJ" role="doc-biblioref">56</a>,<a href="#ref-9U8b91g7" role="doc-biblioref">57</a>]</span>.
Focusing on the immune infiltrate in glioma samples reveals that most immune cells in ScPCA samples are classified as either myeloid or T cell types.
However, there is notable heterogeneity even within HGG and LGG subtypes (Figure <a href="#fig:fig5">5</a>C).
A summary of all the consensus cell types observed in all other ScPCA samples can be found in Figure <a href="#fig:figS7">S7</a>.</p>
<h2 id="analysis-of-bulk-rna-seq">Analysis of bulk RNA-seq</h2>
<p>Several projects in the ScPCA Portal quantified many samples’ expression both at the bulk and single-cell/nuclei level.
Previous research has suggested that, compared to bulk RNA-seq, single-cell/nuclei RNA-seq may fail to capture certain cell types <span class="citation" data-cites="Vz4pQdwU">[<a href="#ref-Vz4pQdwU" role="doc-biblioref">58</a>]</span>, for example due to technical aspects of library preparation.
We therefore sought to ask whether we could identify differences in biological signal between these two modalities that may suggest distinct cell type distributions.
For this analysis, we specifically focused on ScPCA projects with solid tumors, considering only non-multiplexed samples with both sequencing modalities and excluding low-quality single-cell/nuclei libraries.
In total, we analyzed 97 samples across five projects: <code>SCPCP00001</code>, <code>SCPCP000002</code>, <code>SCPCP000006</code>, <code>SCPCP000009</code>, and <code>SCPCP000017</code>.
Projects <code>SCPCP000001</code> and <code>SCPCP000002</code> comprise high- and low-grade gliomas, respectively, and were sequenced at the bulk and single-cell levels, and <code>SCPCP000006</code>, <code>SCPCP000009</code>, and <code>SCPCP000017</code> comprise Wilms tumors, CNS tumors, and osteosarcomas, respectively, and were sequenced at the bulk and single-nuclei levels.
As described in Materials and Methods, we derived pseudobulk expression matrices for each single-cell/nuclei library, and we compared their expression to bulk using a series of linear models (one per ScPCA project) predicting bulk from pseudobulk expression with a random effect controlling for sample (Figure <a href="#fig:fig6">6</a>A, Figure <a href="#fig:figS8">S8</a>A).
Across all projects, we observed an expected positive relationship between bulk and pseudobulk expression.</p>
<p>We next performed an overrepresentation analysis to probe for differences in gene expression that might suggest differences in cell type composition and/or abundance between modalities.
To this end, we calculated the per-gene median of each project’s model residuals and identified outliers, where “positive outliers” are genes with higher bulk RNA-seq expression than predicted from pseudobulk expression, and conversely “negative outliers” are genes with lower bulk RNA-seq expression than predicted from pseudobulk expression.
Using marker gene sets associated with consensus cell types, we calculated the odds ratio in each direction as: the odds a cell type marker gene is present in the given outlier direction compared to other genes.
Following permutation testing and P-value correction to control the FDR at <span class="math inline">\(5%\)</span>, we indeed found that several cell type marker gene sets had higher, but never lower, bulk RNA-seq expression than expected (Figure <a href="#fig:fig6">6</a>B, Figure <a href="#fig:figS8">S8</a>B).</p>
<p>In brain and CNS tumors, we nearly exclusively identified stromal (e.g., endothelial and extracellular matrix secreting cells) and/or neuronal cell types (e.g., glial cells and astrocytes), all of which are known to be prevalent non-immune cells in glioma tumor microenvironments <span class="citation" data-cites="2x15P0J5 DR9mGrXH">[<a href="#ref-2x15P0J5" role="doc-biblioref">59</a>,<a href="#ref-DR9mGrXH" role="doc-biblioref">60</a>]</span>, as overrepresented in bulk (Figure <a href="#fig:fig6">6</a>B).
The only exception to this was monocytes being overrepresented in bulk for <code>SCPCP000009</code> (brain and CNS tumors), which was sequenced at the single-nuclei level while projects <code>SCPCP000001</code> (high-grade gliomas) and <code>SCPCP000002</code> (low-grade gliomas) were sequenced at the single-cell level.
This difference may reflect that single-cell approaches have increased sensitivity to detect immune cells relative to single-nuclei approaches <span class="citation" data-cites="1AegtXQtf">[<a href="#ref-1AegtXQtf" role="doc-biblioref">61</a>]</span>.</p>
<p>Given that our consensus cell type analysis identified a variety of immune cells from high- and low-grade gliomas (Figure <a href="#fig:fig5">5</a>), these results suggest that non-immune cells may have been lost during single-cell library preparation.
Indeed, several of these overrepresented bulk cell types for <code>SCPCP000001</code> and <code>SCPCP000002</code> were not among the single-cell consensus cell types themselves (<code>SCPCP000001</code>: “blood vessel endothelial cell”, “extracellular matrix secreting cell”, “glial cell”, “pericyte”; <code>SCPCP000002</code>: “blood vessel endothelial cell”, “extracellular matrix secreting cell”, “microvascular endothelial cell”), further emphasizing the potential loss of these cell types in the single-cell data.
By contrast, we uncovered a variety of both immune and non-immune cell types overrepresented in bulk RNA-seq <code>SCPCP000017</code> (osteosarcoma; Figure <a href="#fig:figS8">S8</a>B), which may reflect inherent challenges in dissociating bone tissue <span class="citation" data-cites="cnXjUoU1">[<a href="#ref-cnXjUoU1" role="doc-biblioref">62</a>]</span>.
Taken together, these results show that, while bulk and single-cell/nuclei expression is indeed highly correlated, cell type differences may still be present between modalities, potentially driven by cell-type-specific loss in single-cell experiments.</p>
<h2 id="materials-and-methods">Materials and Methods</h2>
<h3 id="data-generation-and-processing">Data generation and processing</h3>
<p>Raw data and metadata were generated and compiled by each lab and institution contributing to the Portal.
Single-cell or single-nuclei libraries were generated using one of the commercially available kits from 10x Genomics.
For bulk RNA-seq, RNA was collected and sequenced using either paired-end or single-end sequencing.
For spatial transcriptomics, cDNA libraries were generated using the Visium kit from 10x Genomics.
All libraries were processed using our open-source pipeline, <code>scpca-nf</code>, to produce summarized gene expression data.
A detailed summary with the total number of samples and libraries collected for each sequencing method broken down by project is available in Table S1.</p>
<h3 id="metadata">Metadata</h3>
<p>Submitters were required to submit the age, sex, organism, diagnosis, subdiagnosis (if applicable), and tissue of origin for each sample.
The submitted metadata was standardized across projects, including converting all ages to years, removing abbreviations used in diagnosis, subdiagnosis, or tissue of origin, and using standard values across projects as much as possible for diagnosis, subdiagnosis, disease timing, and tissue of origin.
For example, all samples obtained at diagnosis were assigned the value <code>Initial diagnosis</code> for disease timing.</p>
<p>In an effort to ensure sample metadata for ScPCA are compatible with CZI’s CELLxGENE, ontology term identifiers were assigned to metadata categories for each sample following the guidelines present in the CELLxGENE schema <span class="citation" data-cites="W0TbGvjF QgMCMjVA">[<a href="#ref-W0TbGvjF" role="doc-biblioref">63</a>,<a href="#ref-QgMCMjVA" role="doc-biblioref">64</a>]</span>, as shown in Table <a href="#tbl:metadata">1</a>.
<br><br></p>
<div id="tbl:metadata" class="tablenos">
<table id="tbl:metadata">
<caption><span>Table 1:</span> Assignment of metadata fields to ontology terms. </caption>
<colgroup>
<col style="width: 12%" />
<col style="width: 87%" />
</colgroup>
<thead>
<tr class="header">
<th>Metadata field</th>
<th>Ontology term description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Age</td>
<td>Ontology term obtained from HsapDv <span class="citation" data-cites="2n9clgPF">[<a href="#ref-2n9clgPF" role="doc-biblioref">13</a>]</span>. For ages 0-11 months, the HsapDv for age in months was used. For ages 12 months and greater, the HsapDv for age in years was used.</td>
</tr>
<tr class="even">
<td>Sex</td>
<td>Ontology term obtained from PATO, either male (PATO:0000384), female (PATO:0000383), or unknown <span class="citation" data-cites="Jx3SGqPR 19R1xim7A">[<a href="#ref-Jx3SGqPR" role="doc-biblioref">14</a>,<a href="#ref-19R1xim7A" role="doc-biblioref">15</a>]</span>.</td>
</tr>
<tr class="odd">
<td>Organism</td>
<td>NCBI taxonomy term for organism. All current samples available on the Portal are from Homo sapiens or NCBITaxon:9606 <span class="citation" data-cites="RJLosfFq jFq6Uo6f">[<a href="#ref-RJLosfFq" role="doc-biblioref">16</a>,<a href="#ref-jFq6Uo6f" role="doc-biblioref">17</a>]</span>.</td>
</tr>
<tr class="even">
<td>Diagnosis</td>
<td>The most appropriate MONDO term based on the provided diagnosis <span class="citation" data-cites="pZyZllZn ZPHZMiua">[<a href="#ref-pZyZllZn" role="doc-biblioref">18</a>,<a href="#ref-ZPHZMiua" role="doc-biblioref">19</a>]</span>. An exact match was identified for most samples, but in a handful of cases, the most closely related term was used.</td>
</tr>
<tr class="odd">
<td>Tissue of origin</td>
<td>The most appropriate UBERON term based on the provided tissue of origin <span class="citation" data-cites="swLRou1q TOC8jpgh XbjdiPPQ">[<a href="#ref-swLRou1q" role="doc-biblioref">20</a>,<a href="#ref-TOC8jpgh" role="doc-biblioref">21</a>,<a href="#ref-XbjdiPPQ" role="doc-biblioref">22</a>]</span>. An exact match was identified for most samples, but in a handful of cases, the most closely related term was used.</td>
</tr>
<tr class="even">
<td>Ethnicity (if applicable)</td>
<td>If the submitter provided ethnicity, the associated Hancestro term <span class="citation" data-cites="9Hno2ty7 Opr5odMH">[<a href="#ref-9Hno2ty7" role="doc-biblioref">23</a>,<a href="#ref-Opr5odMH" role="doc-biblioref">24</a>]</span>. If ethnicity is unavailable, <code>unknown</code> is used.</td>
</tr>
</tbody>
</table>
</div>
<h3 id="processing-single-cell-and-single-nuclei-rna-seq-data-with-alevin-fry">Processing single-cell and single-nuclei RNA-seq data with alevin-fry</h3>
<p>To quantify RNA-seq gene expression for each cell or nucleus in a library, <code>scpca-nf</code> uses <code>salmon alevin</code> <span class="citation" data-cites="91inc6cK">[<a href="#ref-91inc6cK" role="doc-biblioref">65</a>]</span> and <code>alevin-fry</code><span class="citation" data-cites="hYvj5GBC">[<a href="#ref-hYvj5GBC" role="doc-biblioref">12</a>]</span> to generate a gene by cell counts matrix.
Prior to mapping, we generated an index using transcripts from both spliced cDNA and unspliced cDNA sequences, denoted as the <code>splici</code> index <span class="citation" data-cites="hYvj5GBC">[<a href="#ref-hYvj5GBC" role="doc-biblioref">12</a>]</span>.
The index was generated from the human genome, GRCh38, Ensembl version 104.
<code>salmon alevin</code> was run using selective alignment to the <code>splici</code> index with the <code>--rad</code> option to generate a reduced alignment data (RAD) file required for input to <code>alevin-fry</code>.</p>
<p>The RAD file was used as input to the recommended <code>alevin-fry</code> workflow, with the following customizations.
At the <code>generate-permit-list</code> step, we used the <code>--unfiltered-pl</code> option to provide a list of expected barcodes specific to the 10x kit used to generate each library.
The <code>quant</code> step was run using the <code>cr-like-em</code> resolution strategy for feature quantification and UMI de-duplication.</p>
<h3 id="post-alevin-fry-processing-of-single-cell-and-single-nuclei-rna-seq-data">Post alevin-fry processing of single-cell and single-nuclei RNA-seq data</h3>
<p>The output from running <code>alevin-fry</code> includes a gene by cell counts matrix, with reads from both spliced and unspliced reads for all potential cell barcodes.
This output is read into R to create a <code>SingleCellExperiment</code> using <code>fishpond::load_fry()</code>.
The resulting <code>SingleCellExperiment</code> contains a <code>counts</code> assay with a gene by cell counts matrix where all spliced and unspliced reads for a given gene are totaled together.
We also include a <code>spliced</code> assay that contains a gene by cell counts matrix with only spliced reads.
These matrices include all potential cells, including empty droplets, and are provided for all Portal downloads in the unfiltered objects saved as <code>.rds</code> files with the <code>_unfiltered.rds</code> suffix.</p>
<p>Each droplet was tested for deviation from the ambient RNA profile using <code>DropletUtils::emptyDropsCellRanger()</code> and those with an FDR ≤ 0.01 were retained as likely cells.
If a library did not have a sufficient number of droplets and <code>DropletUtils::emptyDropsCellRanger()</code> failed, cells with fewer than 100 UMIs were removed.
Gene expression data for any cells that remain after filtering are provided in the filtered objects saved as <code>.rds</code> files with the <code>_filtered.rds</code> suffix.</p>
<p>In addition to removing empty droplets, <code>scpca-nf</code> also removes cells that are likely to be compromised by damage or low-quality sequencing.
<code>miQC</code> was used to calculate the posterior probability that each cell is compromised <span class="citation" data-cites="17Yhk1HAt">[<a href="#ref-17Yhk1HAt" role="doc-biblioref">38</a>]</span>.
Any cells with a probability of being compromised greater than 0.75 and fewer than 200 genes detected were removed before further processing.
The gene expression counts from the remaining cells were log-normalized using the deconvolution method from Lun, Bach, and Marioni <span class="citation" data-cites="vpU0FTFY">[<a href="#ref-vpU0FTFY" role="doc-biblioref">39</a>]</span>.
<code>scran::modelGeneVar()</code> was used to model gene variance from the log-normalized counts and <code>scran::getTopHVGs()</code> was used to select the top 2000 high-variance genes.
These were used as input to calculate the top 50 principal components using <code>scater::runPCA()</code>.
Finally, UMAP embeddings were calculated from the principal components with <code>scater::runUMAP()</code>.
The raw and log-normalized counts, list of 2000 high-variance genes, principal components, and UMAP embeddings are all stored in the processed objects saved as <code>.rds</code> files with the <code>_processed.rds</code> suffix.</p>
<h3 id="quantifying-gene-expression-for-libraries-with-cite-seq-or-cell-hashing">Quantifying gene expression for libraries with CITE-seq or cell hashing</h3>
<p>All libraries with antibody-derived tags (ADTs) or hashtag oligonucleotides (HTOs) were mapped to a reference index using <code>salmon alevin</code> and quantified using <code>alevin-fry</code>.
The reference indices were constructed using the <code>salmon index</code> command with the <code>--feature</code> option.
References were custom-built for each ScPCA project and constructed using the submitter-provided list of ADTs or HTOs and their barcode sequences.</p>
<p>The ADT by cell or HTO by cell counts matrix produced by <code>alevin-fry</code> were read into R as a <code>SingleCellExperiment</code> object and saved as an alternative experiment (<code>altExp</code>) in the same <code>SingleCellExperiment</code> object with the unfiltered gene expression counts data.
The <code>altExp</code> within the unfiltered object contains all identified ADTs or HTOs and all barcodes identified in the RNA-seq gene expression data.
Any barcodes that only appeared in either ADT or HTO data were discarded, and cell barcodes that were only found in the gene expression data (i.e., did not appear in the ADT or HTO data) were assigned zero counts for all ADTs and HTOs.
Any cells removed after filtering empty droplets were also removed from the ADT and HTO counts matrices and before creating the filtered <code>SingleCellExperiment</code> object.</p>
<h3 id="processing-adt-expression-data-from-cite-seq">Processing ADT expression data from CITE-seq</h3>
<p>The ADT count matrix stored in the unfiltered object was used to calculate an ambient profile with <code>DropletUtils::ambientProfileEmpty()</code>.
This ambient profile was used to calculate quality-control statistics with <code>DropletUtils::cleanTagCounts()</code> for all cells remaining after removing empty droplets.
Any negative or isotype controls were taken into account when calculating QC statistics.
Cells with a high level of ambient contamination or negative/isotype controls were flagged as having low-quality ADT expression, but we did not remove any cells based on ADT quality from the object.
The filtered and processed objects contain the results from running <code>DropletUtils::cleanTagCounts()</code>.</p>
<p>ADT count data were then normalized by calculating median size factors using the ambient profile with <code>scuttle::computeMedianFactors()</code>.
If median-based normalization failed for any reason, ADT counts were log-transformed after adding a pseudocount of 1.
Normalized counts are only available for any cells that would be retained after ADT filtering, and any cells that would be filtered out based on <code>DropletUtils::cleanTagCounts()</code> are assigned <code>NA</code>.
The normalized ADT data are available in the <code>altExp</code> of the processed object.</p>
<h3 id="processing-hto-data-from-multiplexed-libraries">Processing HTO data from multiplexed libraries</h3>
<p>To identify which cells come from which samples in a multiplexed library, we applied three different demultiplexing methods: genetic demultiplexing, HTO demultiplexing using <code>DropletUtils::hashedDrops()</code>, and HTO demultiplexing using <code>Seurat::HTODemux()</code>.
We do not provide separate <code>SingleCellExperiment</code> objects for each sample in a library.
Each multiplexed library object contains the counts data from all samples and the results from all three demultiplexing methods to allow users to select which method(s) to use.</p>
<h4 id="genetic-demultiplexing">Genetic demultiplexing</h4>
<p>If all samples in a multiplexed library were also sequenced using bulk RNA-seq, we performed genetic demultiplexing using genotype data from both bulk RNA-seq and single-cell or single-nuclei RNA-seq <span class="citation" data-cites="zLp25QYu">[<a href="#ref-zLp25QYu" role="doc-biblioref">47</a>]</span>.
If bulk RNA-seq was not available, no genetic demultiplexing was performed.</p>
<p>Bulk RNA-seq reads for each sample were mapped to a reference genome using <code>STAR</code> <span class="citation" data-cites="tTu8Ds9Z">[<a href="#ref-tTu8Ds9Z" role="doc-biblioref">66</a>]</span>, and multiplexed single-cell or single-nuclei RNA-seq reads were mapped to the same reference genome using <code>STARsolo</code><span class="citation" data-cites="rxoRG1BR">[<a href="#ref-rxoRG1BR" role="doc-biblioref">67</a>]</span>.
The mapped bulk reads were used to call variants and assign genotypes with <code>bcftools mpileup</code> <span class="citation" data-cites="eyOvxHVn">[<a href="#ref-eyOvxHVn" role="doc-biblioref">68</a>]</span>.
<code>cellsnp-lite</code> was then used to genotype single-cell data at the identified sites found in the bulk RNA-seq data <span class="citation" data-cites="wd1cI6xh">[<a href="#ref-wd1cI6xh" role="doc-biblioref">69</a>]</span>.
Finally, <code>vireo</code> was used to identify the sample of origin <span class="citation" data-cites="wd1cI6xh">[<a href="#ref-wd1cI6xh" role="doc-biblioref">69</a>]</span>.</p>
<h4 id="hto-demultiplexing">HTO demultiplexing</h4>
<p>For all multiplexed libraries, we performed demultiplexing using <code>DropletUtils::hashedDrops()</code> and <code>Seurat::HTODemux()</code>.
For both methods, we used the default parameters and only performed demultiplexing on the filtered cells present in the filtered object.
The results from both these methods are available in the filtered and processed objects.</p>
<h3 id="quantification-of-spatial-transcriptomics-data">Quantification of spatial transcriptomics data</h3>
<p>10x Genomics’ Space Ranger <span class="citation" data-cites="xo912CsO">[<a href="#ref-xo912CsO" role="doc-biblioref">50</a>]</span> was used to quantify gene expression data from spatial transcriptomics libraries.
<code>cellranger mkref</code> was used to create a reference index from the human genome, GRCh38, Ensembl version 104.
The FASTQ files, microscopic slide image, and slide serial number were provided as input to <code>spaceranger count</code>.
The raw and filtered counts matrix and the summary report output by <code>spaceranger count</code> are included in the folder output from <code>scpca-nf</code>.</p>
<h3 id="quantification-of-bulk-rna-seq-data">Quantification of bulk RNA-seq data</h3>
<p><code>fastp</code> was used to trim adapters and perform quality and length filtering on all FASTQ files from bulk RNA-seq.
We used a decoy-aware reference created from spliced cDNA sequences with the entire human genome sequence (GRCh38, Ensembl version 104) as the decoy <span class="citation" data-cites="vrqQcFyx">[<a href="#ref-vrqQcFyx" role="doc-biblioref">49</a>]</span>.
The trimmed reads were then provided as input to <code>salmon quant</code> for selective alignment.
In addition to using the default parameters for <code>salmon quant</code>, we applied the <code>--seqBias</code> and <code>--gcBias</code> flags to correct for sequence-specific biases due to random hexamer priming and fragment-level GC biases, respectively.</p>
<h3 id="cell-type-annotation">Cell type annotation</h3>
<p>Cell type labels determined by both <code>SingleR</code><span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span> and <code>CellAssign</code><span class="citation" data-cites="HES0WO6k">[<a href="#ref-HES0WO6k" role="doc-biblioref">42</a>]</span> were added to processed <code>SingleCellExperiment</code> objects.
If cell types were obtained from the submitter of the dataset, the submitter-provided annotations were incorporated into all <code>SingleCellExperiment</code> objects (unfiltered, filtered, and processed).</p>
<p>To prepare the references used for assigning cell types, we developed a separate workflow, <code>build-celltype-index.nf</code>, within <code>scpca-nf</code>.
For <code>SingleR</code>, we used the <code>BlueprintEncodeData</code> from the <code>celldex</code> package <span class="citation" data-cites="MzCvXTgB 15J98V2qM">[<a href="#ref-MzCvXTgB" role="doc-biblioref">52</a>,<a href="#ref-15J98V2qM" role="doc-biblioref">53</a>]</span> to train the <code>SingleR</code> classification model with <code>SingleR::trainSingleR()</code>.
In the main <code>scpca-nf</code> workflow, this model and the processed <code>SingleCellExperiment</code> object were input to <code>SingleR::classifySingleR()</code>.
The <code>SingleR</code> output of cell type annotations and a score matrix for each cell and all possible cell types were added to the processed <code>SingleCellExperiment</code> object output.
To evaluate confidence in <code>SingleR</code> cell type assignments, we also calculated a delta median statistic for each cell by subtracting the median cell type score from the score associated with the assigned cell type <span class="citation" data-cites="xpZq9dBF">[<a href="#ref-xpZq9dBF" role="doc-biblioref">51</a>]</span>.</p>
<p>For <code>CellAssign</code>, marker gene references were created using the marker gene lists available on <code>PanglaoDB</code> <span class="citation" data-cites="T3PlZ3Vh">[<a href="#ref-T3PlZ3Vh" role="doc-biblioref">54</a>]</span>.
Organ-specific references were built using all cell types in a specified organ listed in <code>PanglaoDB</code> to accommodate all ScPCA projects encompassing a variety of disease and tissue types.
If a set of disease types in a given project encompassed cells that may be present in multiple organ groups, multiple organs were combined.
For example, we created a reference containing bone, connective tissue, smooth muscle, and immune cells for sarcomas that appear in bone or soft tissue.</p>
<p>Given the processed <code>SingleCellExperiment</code> object and organ-specific reference, <code>scvi.external.CellAssign</code> was used in the main <code>scpca-nf</code> workflow to train the model and predict the assigned cell type.
For each cell, <code>CellAssign</code> calculates a probability of assignment to each cell type in the reference.
The probability matrix and a prediction based on the most probable cell type were added as cell type annotations to the processed <code>SingleCellExperiment</code> object output.</p>
<h4 id="assigning-consensus-cell-types-1">Assigning consensus cell types</h4>
<p>Cell type labels obtained from <code>SingleR</code> and <code>CellAssign</code> were then used to assign an ontology-aware consensus cell type label.
We first assigned each of the cell types present in the <code>PanglaoDB</code> <span class="citation" data-cites="T3PlZ3Vh">[<a href="#ref-T3PlZ3Vh" role="doc-biblioref">54</a>]</span> reference used with <code>CellAssign</code> to an appropriate Cell Ontology term <span class="citation" data-cites="wBFzn1br">[<a href="#ref-wBFzn1br" role="doc-biblioref">70</a>]</span>.
For cell types available in the <code>BlueprintEncodeData</code> reference used with <code>SingleR</code>, we used the provided Cell Ontology terms.</p>
<p>We then created a reference table containing all possible combinations of cell types assigned using <code>SingleR</code> and <code>CellAssign</code> and identified the latest common ancestor (LCA) <span class="citation" data-cites="17SbnOJiL">[<a href="#ref-17SbnOJiL" role="doc-biblioref">55</a>]</span> between the two cell type terms.
The LCA was then used as the consensus cell type label if the following criteria were met, otherwise no consensus cell type was assigned:</p>
<ol type="1">
<li><p>The terms shared only one distinct LCA.
The only exception to this rule was if the terms shared two LCAs, one of which was <code>hematopoietic precursor cell</code>; then <code>hematopoietic precursor cell</code> was used as the consensus label.</p></li>
<li><p>The LCA had fewer than 170 descendants, or was either <code>neuron</code> or <code>epithelial cell</code>.</p></li>
</ol>
<p>We also excluded the following non-specific LCA terms: <code>bone cell</code>, <code>lining cell</code>, <code>blood cell</code>, <code>progenitor cell</code>, and <code>supporting cell</code>.</p>
<p>The consensus cell type assignments, including both the Cell Ontology term and the associated human-readable name, are available in the processed <code>SingleCellExperiment</code> objects.</p>
<p>Consensus cell type assignments were evaluated by looking at marker gene expression in a set of cell-type specific marker genes.
Marker genes were obtained from the list of Human cell markers on <code>CellMarker2.0</code> <span class="citation" data-cites="MclvBVPD">[<a href="#ref-MclvBVPD" role="doc-biblioref">71</a>]</span>.
We considered only those that are specific to a single cell type, with the exception of hematopoietic precursor cells, which express genes found in other, more differentiated immune cells.</p>
<h3 id="generating-merged-data">Generating merged data</h3>
<p>Merged objects are created with the <code>merge.nf</code> workflow within <code>scpca-nf</code>.
This workflow takes as input the processed <code>SingleCellExperiment</code> objects in a given ScPCA project output by <code>scpca-nf</code> and creates a single merged <code>SingleCellExperiment</code> object containing gene expression data and metadata from all libraries in that project.
The merged object includes both raw and normalized counts for all cells from all libraries.
Because the same reference index was used to quantify all single-cell and single-nuclei RNA-seq data, the set of genes is the same in the merged object and the individual objects.
Library-, cell- and gene-specific metadata from each of the processed <code>SingleCellExperiment</code> objects are also combined and stored in the merged object.
The <code>merge.nf</code> workflow does not perform batch-correction or integration.
The counts in the merged object are therefore not batch-corrected.</p>
<p>The top 2000 shared high-variance genes are identified from the merged counts matrix by modeling variance using <code>scran::modelGeneVar()</code> and specifying library IDs for the <code>block</code> argument.
These genes are used to calculate library-aware principal components with <code>batchelor::multiBatchPCA()</code>.
The top 50 principal components were selected and used to calculate UMAP embeddings for the merged object.</p>
<p>If any libraries included in the ScPCA project contain additional ADT data, the ADT data are also merged and stored in the <code>altExp</code> slot of the merged <code>SingleCellExperiment</code> object.
By contrast, if any libraries included in the ScPCA project are multiplexed and contain HTO data, no merged object is created.
Merged objects were not created for projects with more than 100 samples because of the computational resources that would be required for working with those objects.</p>
<h3 id="converting-singlecellexperiment-objects-to-anndata-objects">Converting SingleCellExperiment objects to AnnData objects</h3>
<p><code>zellkonverter::writeH5AD()</code> <span class="citation" data-cites="11dH4UgRF">[<a href="#ref-11dH4UgRF" role="doc-biblioref">72</a>]</span> was used to convert <code>SingleCellExperiment</code> objects to <code>AnnData</code> format and export the objects as <code>.h5ad</code> files.
For any <code>SingleCellExperiment</code> objects containing an <code>altExp</code> (e.g., ADT data), the RNA and ADT data were exported and saved separately as RNA (<code>_rna.h5ad</code>) and ADT (<code>_adt.h5ad</code>) files.
Multiplexed libraries were not converted to <code>AnnData</code> objects, due to the potential for ambiguity in sample origin assignments.</p>
<p>All merged <code>SingleCellExperiment</code> objects were converted to <code>AnnData</code> objects and saved as <code>.h5ad</code> files.
If a merged <code>SingleCellExperiment</code> object contained any ADT data, the RNA and ADT data were exported and saved separately as RNA (<code>_rna.h5ad</code>) and ADT (<code>_adt.h5ad</code>) objects.
In contrast, if a merged <code>SingleCellExperiment</code> object contained HTO data due to the presence of any multiplexed libraries in the merged object, the HTO data was removed from the <code>SingleCellExperiment</code> object and not included in the exported <code>AnnData</code> object.</p>
<h3 id="analysis-of-bulk-rna-seq-data">Analysis of bulk RNA-seq data</h3>
<h4 id="data-preparation">Data preparation</h4>
<p>We identified solid tumor samples with both bulk and single-cell (or single-nuclei) RNA-seq data in the ScPCA Portal for analysis, with multiplexed samples excluded (N=105).
We removed low-quality samples based on visual inspection of quality control reports (N=8), leaving a total of 97 samples across five ScPCA projects for analysis.</p>
<p>For each project, we transformed and normalized bulk counts matrices for all samples using <code>DESeq2::rlog()</code> <span class="citation" data-cites="w9AOzBMw">[<a href="#ref-w9AOzBMw" role="doc-biblioref">73</a>]</span>.
We obtained pseudobulk counts by summing raw single-cell counts for each sample, and similarly transformed each project’s resulting counts matrix with <code>DESeq2::rlog()</code>.
We filtered out genes which were not observed in either the bulk or pseudobulk raw counts matrices before subsequent analysis.
For each project, we then used the <code>lme4</code> <span class="citation" data-cites="AUoV0xkq">[<a href="#ref-AUoV0xkq" role="doc-biblioref">74</a>]</span> R package to construct a linear model predicting bulk from pseudobulk counts considering a random effect for sample id: <code>bulk ~ pseudobulk + (1|sample_id)</code>.</p>
<h4 id="overrepresentation-analysis">Overrepresentation analysis</h4>
<p>We next asked whether certain cell types might be overrepresented in one modality compared to the other.
For this, we first identified cell types of interest as all consensus cell types possible for each project. <!-- TODO: phrasing. -->
We then created gene sets for each consensus cell type using the project’s <code>CellAssign</code> marker gene reference.
Because a consensus cell type can encompass multiple cell types in the marker gene reference, we defined each consensus cell type’s gene set as the union of all marker genes for each of its constituent reference cell types.</p>
<p>For input to the overrepresentation analysis, we summarized model residuals within each project by taking the median residual for each gene across samples and then transformed these summarized residuals into Z-scores.
We identified outlier genes as those with Z-scores greater than 2.5 (positive outliers) or less than -2.5 (negative outliers).
In this case, positive outliers represent genes with comparatively higher expression in the bulk modality, and negative outliers represent genes with comparatively higher expression in the single-cell modality.</p>
<p>For each consensus cell type gene set, we calculated two odds ratios representing whether genes were overrepresented in the positive outliers (enriched in bulk) or negative outliers (enriched in pseudobulk).
We calculated P-values for both the bulk and pseudobulk enrichment directions via permutation testing with 10,000 replicates.
We defined gene sets with significant overrepresentation as those with a false-discovery-rate-corrected P-value ≤ 0.05 <span class="citation" data-cites="UeFCwwR7">[<a href="#ref-UeFCwwR7" role="doc-biblioref">75</a>]</span>.</p>
<h2 id="code-and-data-availability">Code and data availability</h2>
<p>All summarized gene expression data and de-identified metadata are available for download on the ScPCA Portal, <a href="https://scpca.alexslemonade.org/" class="uri">https://scpca.alexslemonade.org/</a>.</p>
<p>Documentation for the Portal can be found at <a href="https://scpca.readthedocs.io" class="uri">https://scpca.readthedocs.io</a>.</p>
<p>All original code was developed within the following repositories and is publicly available as follows:</p>
<ul>
<li>The <code>scpca-nf</code> workflow used to process all samples available on the Portal can be found at <a href="https://github.com/AlexsLemonade/scpca-nf" class="uri">https://github.com/AlexsLemonade/scpca-nf</a>.</li>
<li>The Single-cell Pediatric Cancer Atlas Portal code can be found at <a href="https://github.com/AlexsLemonade/scpca-portal" class="uri">https://github.com/AlexsLemonade/scpca-portal</a>.</li>
<li>Benchmarking of tools used to build <code>scpca-nf</code> can be found at <a href="https://github.com/AlexsLemonade/alsf-scpca/tree/main/analysis" class="uri">https://github.com/AlexsLemonade/alsf-scpca/tree/main/analysis</a> and <a href="https://github.com/AlexsLemonade/sc-data-integration/tree/main/celltype_annotation" class="uri">https://github.com/AlexsLemonade/sc-data-integration/tree/main/celltype_annotation</a>.</li>
<li>All code for creating the reference files used for consensus cell type assignment can be found at <a href="https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/cell-type-consensus" class="uri">https://github.com/AlexsLemonade/OpenScPCA-analysis/tree/main/analyses/cell-type-consensus</a>.</li>
<li>All code for the underlying figures and analyses can be found at <a href="https://github.com/AlexsLemonade/scpca-paper-figures" class="uri">https://github.com/AlexsLemonade/scpca-paper-figures</a>.</li>
<li>The manuscript can be found at <a href="https://github.com/AlexsLemonade/ScPCA-manuscript" class="uri">https://github.com/AlexsLemonade/ScPCA-manuscript</a>.</li>
</ul>
<h2 id="discussion">Discussion</h2>
<p>Here, we introduced the ScPCA Portal, a downloadable collection of uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata from pediatric tumor samples.
The Portal includes 700 samples from 55 tumor types, making this the most comprehensive collection of publicly available single-cell RNA-seq datasets from pediatric tumor samples to our knowledge.
Summarized data are available at three different processing stages: unfiltered, filtered, or processed objects, permitting users to choose to start from a processed object or perform their own processing, such as filtering and normalization.
Processed objects containing normalized gene expression data, reduced dimensionality results from PCA and UMAP, and cell type annotations are provided to save researchers time and allow users to skip straight to downstream analysis, such as identifying marker genes or exploring genes of interest.
Standardized metadata, containing human-readable values for all fields and ontology term identifiers for a subset of metadata fields, is included in a separate metadata file and the data objects for all samples.
Every library includes a quality control report, which lets users assess data quality and identify low-quality libraries that they may wish to exclude from further downstream analyses.</p>
<p>Data on the Portal is available as either <code>SingleCellExperiment</code> or <code>AnnData</code> objects, so users can work in R or Python with the downloaded data using common analysis systems such as <code>Bioconductor</code> or <code>Scanpy</code>, depending on their preference.
Providing data as <code>AnnData</code> objects also means users can easily integrate ScPCA data with data and tools available on other platforms.
In particular, the format of the provided <code>AnnData</code> objects was designed to be mostly compliant with the requirements of CZI CELLxGENE <span class="citation" data-cites="115InIIhW WrWip7GV 7NobF7Ei">[<a href="#ref-115InIIhW" role="doc-biblioref">76</a>,<a href="#ref-WrWip7GV" role="doc-biblioref">77</a>,<a href="#ref-7NobF7Ei" role="doc-biblioref">78</a>]</span>, but these objects can also be used with UCSC Cell Browser <span class="citation" data-cites="xCP3BjTw VKaQa2Z3">[<a href="#ref-xCP3BjTw" role="doc-biblioref">79</a>,<a href="#ref-VKaQa2Z3" role="doc-biblioref">80</a>]</span> or Kana <span class="citation" data-cites="1DswCJYT0 1FzXrvESX">[<a href="#ref-1DswCJYT0" role="doc-biblioref">81</a>,<a href="#ref-1FzXrvESX" role="doc-biblioref">82</a>]</span>.
Additionally, users can choose to download a merged <code>SingleCellExperiment</code> or <code>AnnData</code> object containing all gene expression data and metadata from all samples in a project.
This is helpful for analyzing multiple samples simultaneously and performing analyses such as differential gene expression or gene set enrichment.</p>
<p>To provide users with cell type annotations, we used two automated methods, <code>SingleR</code> and <code>CellAssign</code>, which use publicly available references.
We then used the correspondence between methods to derive ontology-aware consensus cell type labels.
In addition to providing a consistent labeling scheme across samples in the ScPCA Portal, these consensus cell type labels may be particularly helpful for annotating populations of normal cells, which are represented in the underlying <code>SingleR</code> and <code>CellAssign</code> references, that may be present in tumor samples.
As the publicly available references we used do not contain tumor cells but only normal cells, we recognize that the annotations we provide are limited.
Despite these limitations, the annotations provide a good starting point for further downstream analysis.</p>
<p>We also introduced our open-source and efficient workflow for uniformly processing datasets available on the Portal, <code>scpca-nf</code>, which is available to the entire research community.
In one command, <code>scpca-nf</code> can process raw data from various sequencing types, turning FASTQ files into processed <code>SingleCellExperiment</code> or <code>AnnData</code> objects ready for downstream analyses.
Using Nextflow as the framework for <code>scpca-nf</code> means the workflow is both modular and portable.
This makes it easy to add support for more modalities in the future, such as single-cell ATAC-seq, and allows others to run the workflow on their samples in their computing environment, maintaining the security of protected raw data.
Processed output from running <code>scpca-nf</code> on samples from pediatric tumors, cell lines, or other model organisms is eligible for submission to the ScPCA Portal, enabling us to continue increasing the number of samples available to researchers.</p>
<p>Many samples on the Portal have additional sequencing data, including corresponding ADT data from CITE-seq, cell hashing data, bulk RNA-seq, or spatial transcriptomics, enabling users to gather more information about a single sample than they could from single-cell or single-nuclei RNA-seq alone.
Samples with CITE-seq have additional information about cell-surface protein expression in individual cells, which can help determine cell types and correlate RNA to protein expression <span class="citation" data-cites="abp5rbw2">[<a href="#ref-abp5rbw2" role="doc-biblioref">32</a>]</span>.
Spatial transcriptomics data on the Portal are not single-cell resolution, making it hard to identify cell types and spatial patterns from the spatial data alone.
By providing matching single-cell RNA-seq, users can implement analysis tools, like those that use single-cell RNA-seq to deconvolute spatial data, to gain more insights about the spatial data <span class="citation" data-cites="Whp3r83M">[<a href="#ref-Whp3r83M" role="doc-biblioref">83</a>]</span>.</p>
<p>Similarly, users can gain more insight from bulk RNA-seq data available on the Portal by integrating with single-cell RNA-seq data from the same sample <span class="citation" data-cites="mgOojr60 GpuiAcfU">[<a href="#ref-mgOojr60" role="doc-biblioref">84</a>,<a href="#ref-GpuiAcfU" role="doc-biblioref">85</a>]</span>.
The single-cell RNA-seq data available on the Portal can also be used to deconvolute existing bulk RNA-seq datasets, allowing researchers to infer abundance of different cell types or cell states in bulk RNA-seq data.
Our analysis of this data showed that while expression is generally consistent between matched bulk RNA-seq and single-cell/nuclei libraries in the Portal (Figure <a href="#fig:fig6">6</a>A, Figure <a href="#fig:figS8">S8</a>A), there are potential differences in cell type composition between modalities that may reflect technological differences in sample and library preparation.
Such multimodal comparisons that the ScPCA Portal enables reveal biological and/or technical signal that would otherwise not be apparent from one sequencing modality alone.</p>
<p>Data available on the ScPCA Portal can further be used for external data analyses, for example to support re-analyzing any existing pediatric cancer datasets with bulk RNA-seq, such as the Pediatric Brain Tumor Atlas <span class="citation" data-cites="U5WisTG5 5VXMHJ7N">[<a href="#ref-U5WisTG5" role="doc-biblioref">86</a>,<a href="#ref-5VXMHJ7N" role="doc-biblioref">87</a>]</span>.
This allows researchers to glean more insight from previously published data without obtaining fresh samples, saving time and money.</p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>We thank the data generators and submitters of the Single-cell Pediatric Cancer Atlas.
We also thank Anna Greene for her role in constructing the Single-cell Pediatric Cancer Atlas funding opportunity.</p>
<p>This work was funded through the Alex’s Lemonade Stand Foundation Childhood Cancer Data Lab and Childhood Cancer Data Lab Postdoctoral Fellowship (SMF).</p>
<h2 id="author-contributions">Author Contributions</h2>
<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th>Author</th>
<th>Contributions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Allegra G. Hawkins</td>
<td>Methodology, Software, Investigation, Validation, Formal analysis, Data curation, Writing - Original Draft, Writing - Review &amp; Editing, Visualization</td>
</tr>
<tr class="even">
<td>Joshua A. Shapiro</td>
<td>Methodology, Software, Investigation, Validation, Formal analysis, Resources, Data curation, Writing - Original Draft, Writing - Review &amp; Editing, Visualization</td>
</tr>
<tr class="odd">
<td>Stephanie J. Spielman</td>
<td>Methodology, Software, Investigation, Validation, Formal analysis, Data curation, Writing - Original Draft, Writing - Review &amp; Editing, Visualization</td>
</tr>
<tr class="even">
<td>David S. Mejia</td>
<td>Methodology, Software, Validation, Data curation, Writing - Review &amp; Editing, Resources</td>
</tr>
<tr class="odd">
<td>Deepashree Venkatesh Prasad</td>
<td>Methodology, Software, Validation, Visualization, Writing - Review &amp; Editing</td>
</tr>
<tr class="even">
<td>Nozomi Ichihara</td>
<td>Methodology, Software, Writing - Review &amp; Editing</td>
</tr>
<tr class="odd">
<td>Arkadii Yakovets</td>
<td>Methodology, Software, Validation, Data curation, Resources, Writing - Review &amp; Editing</td>
</tr>
<tr class="even">
<td>Avrohom M. Gottlieb</td>
<td>Methodology, Software, Validation, Data curation, Writing - Review &amp; Editing, Resources</td>
</tr>
<tr class="odd">
<td>Kurt G. Wheeler</td>
<td>Methodology, Software, Validation, Data curation, Resources, Writing - Review &amp; Editing</td>
</tr>
<tr class="even">
<td>Chante J. Bethell</td>
<td>Software, Validation, Writing - Review &amp; Editing</td>
</tr>
<tr class="odd">
<td>Steven M. Foltz</td>
<td>Writing - Review &amp; Editing</td>
</tr>
<tr class="even">
<td>Jennifer O’Malley</td>
<td>Data curation, Supervision, Writing - Review &amp; Editing</td>
</tr>
<tr class="odd">
<td>Casey S. Greene</td>
<td>Conceptualization, Project administration, Supervision, Writing - Review &amp; Editing</td>
</tr>
<tr class="even">
<td>Jaclyn N. Taroni</td>
<td>Conceptualization, Methodology, Investigation, Validation, Data curation, Writing - Original Draft, Writing - Review &amp; Editing, Visualization, Supervision, Project administration</td>
</tr>
</tbody>
</table>
<h2 id="declarations-of-interest">Declarations of Interest</h2>
<p>AGH, JAS, SJS, DSM, DVP, NI, AY, AMG, KGW, CJB, JO, and JNT are or were employees of Alex’s Lemonade Stand Foundation, a sponsor of this research.</p>
<h2 class="page_break_before" id="figure-titles-and-legends">Figure Titles and Legends</h2>
<!-- Figure 1 -->
<div id="fig:fig1" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_1.png?sanitize=true" style="width:7in" data-tag="1" alt="Figure 1: Overview of ScPCA Portal contents." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>Overview of ScPCA Portal contents.</strong></figcaption>
</figure>
</div>
<p>A. Barplots showing sample counts across four main cancer groupings in the ScPCA Portal, with each bar displaying the number of samples for each cancer type.
Each bar is shaded based on the number of samples with each disease timing, and total sample counts for each cancer type are shown to the right of each bar.</p>
<p>B. Barplot showing sample counts across types of modalities present in the ScPCA Portal.
All samples in the portal are shown under the “All Samples” heading.
Samples under the “Samples with additional modalities” heading represent a subset of the total samples with the given additional modality.
Colors shown for each additional modality indicate the suspension type used, either single-cell or single-nuclei RNA-seq.
For example, 75 single-cell samples and 67 single-nuclei samples have accompanying Bulk RNA-seq data.</p>
<p>C. Example of a project card as displayed on the “Browse” page of the ScPCA Portal.
This project card is associated with project <code>SCPCP000009</code> <span class="citation" data-cites="HalHlba8 TffkWyy0">[<a href="#ref-HalHlba8" role="doc-biblioref">26</a>,<a href="#ref-TffkWyy0" role="doc-biblioref">27</a>]</span>.
Project cards include information about the number of samples, technologies and modalities, additional sample metadata information, submitter-provided diagnoses, and a submitter-provided abstract.
Where available, submitter-provided citation information, as well as other databases where this data has been deposited, are also provided.
<br><br><br><br><br></p>
<!-- Figure 2 -->
<div id="fig:fig2" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_2.png?sanitize=true" style="width:7in" data-tag="2" alt="Figure 2: Overview of the scpca-nf workflow." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>Overview of the <code>scpca-nf</code> workflow.</strong></figcaption>
</figure>
</div>
<p>A. Overview of <code>scpca-nf</code>, the primary workflow for processing single-cell and single-nuclei RNA-seq data for the ScPCA Portal.
Mapping is first performed with <code>alevin-fry</code> to generate a gene by cell count matrix, which is read into <code>R</code> and converted into a <code>SingleCellExperiment</code> (<code>SCE</code>) object.
This <code>SCE</code> object is exported as the <code>Unfiltered SCE Object</code> before further post-processing.
Next, empty droplets are filtered out, and the resulting <code>SCE</code> is exported as the <code>Filtered SCE Object</code>.
The filtered object undergoes additional post-processing, including removing low-quality cells, normalizing counts, and performing dimension reduction including principal components analysis and UMAP calculation.
The object undergoes cell type annotation and is exported as the <code>Processed SCE Object</code>.
A summary QC report and a supplemental cell type report are prepared and exported.
Finally, all <code>SCE</code> files are converted to <code>AnnData</code> format and exported.
Panels B-G show example figures that appear in the summary QC report, shown here for <code>SCPCL000001</code>, as follows <span class="citation" data-cites="U3XexzPk">[<a href="#ref-U3XexzPk" role="doc-biblioref">30</a>]</span>.</p>
<p>B. The total UMI count for each cell in the <code>Unfiltered SCE Object</code>, ordered by rank.
Points are colored by the percentage of cells that pass the empty droplets filter.</p>
<p>C. The number of genes detected in each cell passing the empty droplets filter against the total UMI count.
Points are colored by the percentage of mitochondrial reads in the cell.</p>
<p>D. <code>miQC</code> model diagnostic plot showing the percent of mitochondrial reads in each cell against the number of genes detected in the <code>Filtered SCE Object</code>.
Points are colored by the probability that the cell is compromised as determined by <code>miQC</code>.</p>
<p>E. The percent of mitochondrial reads in each cell against the number of genes detected in each cell.
Points are colored by whether the cell was kept or removed, as determined by both <code>miQC</code> and a minimum unique gene count cutoff, prior to normalization and dimensionality reduction.</p>
<p>F. UMAP embeddings of log-normalized RNA expression values where each cell is colored by the number of genes detected.</p>
<p>G. UMAP embeddings of log-normalized RNA expression values for the top four most variable genes, colored by the given gene’s expression.
In the actual summary QC report, the top 12 most highly variable genes are shown.
<br><br></p>
<!-- Figure 3 -->
<div id="fig:fig3" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_3.png?sanitize=true" style="width:7in" data-tag="3" alt="Figure 3: ScPCA Portal project download file structure and merged object workflow." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>ScPCA Portal project download file structure and merged object workflow.</strong></figcaption>
</figure>
</div>
<p>A. File download structure for an ScPCA Portal project download in <code>SingleCellExperiment</code> (<code>SCE</code>) format.
The download folder is named according to both the project ID and the date it was downloaded.
Download folders contain one folder for each sample ID, each containing the three versions (unfiltered, filtered, and processed) of the expression data as well as the summary QC report and cell type report all named according to the ScPCA library ID.
The <code>single_cell_metadata.tsv</code> file contains sample metadata for all samples included in the download.
The <code>README.md</code> file provides information about the contents of each download file, additional contact and citation information, and terms of use for data downloaded from the ScPCA Portal.
The files <code>bulk_quant.tsv</code> and <code>bulk_metadata.tsv</code> are only present for projects that also have bulk RNA-Seq data and contain, respectively, a gene by sample matrix of raw gene expression as quantified by <code>salmon</code>, and associated metadata for all samples with bulk RNA-Seq data.</p>
<p>B. File download structure for an ScPCA Portal merged project download in <code>SCE</code> format.
The download folder is named according to both the project ID and the date it was downloaded.
Download folders contain a single merged object containing all samples in the given project as well as a summary report briefly detailing the contents of the merged object.
All summary QC and cell type reports for each individual library are also provided in the <code>individual_reports</code> folder arranged by their sample ID.
As in panel (A), additional files <code>single_cell_metadata.tsv</code>, <code>bulk_quant.tsv</code>, <code>bulk_metadata.tsv</code>, and <code>README.md</code> are also included.</p>
<p>C. Overview of the merged workflow.
Processed <code>SCE</code> objects associated with a given project are merged into a single object, including ADT counts from CITE-seq data if present, and a merged summary report is generated.
Merged objects are available for download either in <code>SCE</code> or <code>AnnData</code> format.</p>
<p>D. Example of UMAPs as shown in the merged summary report.
A grid of UMAPs is shown for each library in the merged object, with cells in the library of interest shown in red and all other cells belonging to other libraries shown in gray.
The UMAP is constructed from the merged object such that all libraries contribute an equal weight, but no batch correction was performed.
The libraries pictured are a subset of libraries in the ScPCA project <code>SCPCP000003</code>.
For this figure specifically, the merged UMAP was constructed from a merged object containing only these four libraries, but the merged object and summary report on the ScPCA Portal for <code>SCPCP000003</code> contain all of this project’s libraries.
<br><br></p>
<!-- Figure 4 -->
<div id="fig:fig4" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_4.png?sanitize=true" style="width:7in" data-tag="4" alt="Figure 4: Cell type annotation in scpca-nf." />
<figcaption aria-hidden="true"><span>Figure 4:</span> <strong>Cell type annotation in <code>scpca-nf</code>.</strong></figcaption>
</figure>
</div>
<p>A. Expanded view of the process for adding cell type annotations within <code>scpca-nf</code>, as introduced in Figure <a href="#fig:fig2">2</a>A.
Cell type annotation is performed on the <code>Processed SCE Object</code>.
A <code>celldex</code> <span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span> reference dataset with ontology labels is used as input for annotation with <code>SingleR</code> <span class="citation" data-cites="13P6oSrS0">[<a href="#ref-13P6oSrS0" role="doc-biblioref">41</a>]</span>, and a list of marker genes compiled from <code>PanglaoDB</code> <span class="citation" data-cites="T3PlZ3Vh">[<a href="#ref-T3PlZ3Vh" role="doc-biblioref">54</a>]</span> is used as input for annotation with <code>CellAssign</code> <span class="citation" data-cites="HES0WO6k">[<a href="#ref-HES0WO6k" role="doc-biblioref">42</a>]</span>.
Results from cell type annotation are then added to the <code>Processed SCE Object</code>, and a cell type summary report with information about reference sources, comparisons among cell type annotation methods, and diagnostic plots is created.
Although not shown in this panel, cell type annotations are also included in the <code>Processed AnnData Object</code> created from the <code>Processed SCE Object</code> (Figure <a href="#fig:fig2">2</a>A).</p>
<p>B. Example heatmap as shown in the cell type summary report comparing annotations with <code>SingleR</code> and <code>CellAssign</code>.
Heatmap cells are colored by the Jaccard similarity index.
A value of 1 means that there is complete overlap between which cells are annotated with the two labels being compared, and a value of 0 means that there is no overlap between which cells are annotated with the two labels being compared.
The heatmap shown is from library <code>SCPCL000498</code> <span class="citation" data-cites="9F0oWAO0">[<a href="#ref-9F0oWAO0" role="doc-biblioref">25</a>]</span>.
<br><br></p>
<!-- Figure 5 -->
<div id="fig:fig5" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_5.png?sanitize=true" style="width:7in" data-tag="5" alt="Figure 5: Consensus cell type annotations in Brain and CNS tumors." />
<figcaption aria-hidden="true"><span>Figure 5:</span> <strong>Consensus cell type annotations in Brain and CNS tumors.</strong></figcaption>
</figure>
</div>
<p>A. Dot plot showing expression of cell-type-specific marker genes across all libraries from brain and central nervous system (CNS) tumors, excluding multiplexed libraries.
Expression is shown for each broad cell type annotation, where each broad cell type annotation is a collection of similar consensus cell type annotations.
The y-axis displays the broad consensus cell type observed across libraries, with the total number of cells indicated in parentheses.
The x-axis displays marker genes, determined by <code>CellMarker2.0</code> <span class="citation" data-cites="MclvBVPD">[<a href="#ref-MclvBVPD" role="doc-biblioref">71</a>]</span>, used for consensus cell type validation for each cell type shown along the top annotation bar.
Dots are colored by mean gene expression across libraries and sized proportionally to the percent of libraries they are observed in, out of all cells with the same broad cell type annotation in brain and CNS tumor libraries.</p>
<p>B. Barplot showing the percentage of each broad consensus cell type annotation across libraries of brain and CNS tumors, separated into high-grade (left panel) and low-grade (right panel) glioma diagnoses for non-multiplexed libraries.</p>
<p>C. Barplot showing all consensus cell types classified as immune cells across libraries of brain and CNS tumors, separated into high-grade (left panel) and low-grade (right panel) glioma diagnoses for non-multiplexed libraries.
The percentage shown corresponds to the percentage of immune cells classified as the indicated consensus cell type.
Only libraries comprised of at least 1% immune cells, based on consensus cell type annotations, are shown.
Specific consensus cell types for myeloid and lymphocyte immune cells are shown, with all other consensus immune cell types included in “other.”
Notably, granulocytes are also included in “other” because only 1 granulocyte was present in all libraries shown (specifically, <code>SCPCL000793</code>).
<br><br></p>
<!-- Figure 6 -->
<div id="fig:fig6" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_6.png?sanitize=true" style="width:7in" data-tag="6" alt="Figure 6: Comparison of bulk and pseudobulk modalities." />
<figcaption aria-hidden="true"><span>Figure 6:</span> <strong>Comparison of bulk and pseudobulk modalities.</strong></figcaption>
</figure>
</div>
<p>A. Scatter plots colored by point density of <code>DESeq2</code>-transformed and normalized bulk RNA-seq expression compared to pseudobulk expression from single-cell/nuclei RNA-seq.
Samples with RNA-seq for both bulk and single-cell/nuclei modalities, excluding multiplexed samples, from ScPCA projects comprising brain and central nervous system tumors are shown, with the number of samples considered per project shown in parentheses.
The regression line is also shown for each project.
Results from additional projects are shown in Figure <a href="#fig:figS8">S8</a>A.</p>
<p>B. Odds ratios from overrepresentation analysis for the same samples shown in panel A, colored by FDR-corrected significance.
Each odds ratio represents the odds that marker genes for the given cell type were overrepresented in bulk RNA-seq when compared to single-cell/nuclei RNA-seq, relative to other genes.
A total of 36 consensus cell types were evaluated for each project shown here.
Results from additional projects are shown in Figure <a href="#fig:figS8">S8</a>B.</p>
<h2 class="page_break_before" id="supplementary-figures-and-tables">Supplementary Figures and Tables</h2>
<!-- Table S1 -->
<p><strong>Table S1. Overview of ScPCA Portal Datasets.</strong>
This table provides descriptions and sample and library counts for each project in the ScPCA Portal.</p>
<p><code>scpca_project_id</code>: ScPCA project unique identifier.
<code>Diagnosis group</code>: Diagnosis group as shown in Figure <a href="#fig:fig1">1</a>.
<code>Diagnoses</code>: Full set of diagnoses for all samples associated with the project.
<code>Total number of samples (S)</code>: Number of samples associated with the project.
<code>Total number of libraries (L)</code>: Number of libraries associated with the project.
Due to additional sequencing modalities and/or multiplexing, projects may have more libraries than samples.
All remaining columns give the number of libraries (as designated with <code>(L)</code>) with the given suspension type, 10x kit version, or additional modality.
<br><br></p>
<!-- Table S2 -->
<p><strong>Table S2. Summary of references used for cell type annotation with <code>CellAssign</code>.</strong>
This table provides a summary of the references used for assigning cell types for ScPCA projects using <code>CellAssign</code>.
All references were built using all cell types from a specified set of organs present in <code>PanglaoDB</code>’s marker gene list.</p>
<p><code>scpca_project_id</code>: ScPCA project unique identifier.
<code>Diagnoses</code>: Full set of diagnoses for all samples associated with the project.
<code>ScPCA reference name</code>: Name used to describe the custom reference.
<code>PanglaoDB organs included in reference</code>: A list of all organs included in the reference with names of organs corresponding to organs listed in <code>PanglaoDB</code>.
The reference includes marker genes for all cell types present in each organ.
<br><br></p>
<!-- Figure S1 -->
<div id="fig:figS1" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s1.png?sanitize=true" style="width:7in" data-tag="S1" alt="Figure S1: Results from benchmarking alevin-fry and CellRanger performance." />
<figcaption aria-hidden="true"><span>Figure S1:</span> <strong>Results from benchmarking <code>alevin-fry</code> and <code>CellRanger</code> performance.</strong></figcaption>
</figure>
</div>
<p>Each panel compares metrics for six representative ScPCA libraries, including three single-cell and three single-nuclei suspensions, obtained from processing libraries with both <code>alevin-fry</code> and <code>CellRanger</code>.</p>
<p>A. Runtime in minutes (top row) and peak memory in GB (bottom row) for six ScPCA libraries processed with <code>alevin-fry</code> and <code>CellRanger</code>.
Processing with <code>alevin-fry</code> was consistently faster and more memory-efficient compared to processing with <code>CellRanger</code>.</p>
<p>Panels B-D show only cells present in both the <code>alevin-fry</code> and <code>CellRanger</code> output.</p>
<p>B. Comparison of mean gene expression values for six ScPCA libraries processed with <code>alevin-fry</code> and <code>CellRanger</code>, shown on a log-scale.
Each point is a gene, and only genes detected in at least 5 cells are shown.
<span class="math inline">\(R^2\)</span> values shown in the top left corner of each panel reflect broad agreement in mean gene expression values between platforms.</p>
<p>C. Comparison of log total UMI counts for six ScPCA libraries processed with <code>alevin-fry</code> and <code>CellRanger</code>.
Distributions reflect broad agreement in the total UMI count per cell between platforms, although <code>alevin-fry</code> returned slightly higher values for certain single-cell libraries.</p>
<p>D. Comparison of log total genes detected per cell for six ScPCA libraries processed with <code>alevin-fry</code> and <code>CellRanger</code>.
Distributions reflect broad agreement between platforms in the total number of genes detected per cell between platforms, although <code>alevin-fry</code> returned slightly higher values for certain single-cell libraries.
<br><br></p>
<!-- Figure S2 -->
<div id="fig:figS2" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s2.png?sanitize=true" style="width:7in" data-tag="S2" alt="Figure S2: Processing additional single-cell modalities in scpca-nf." />
<figcaption aria-hidden="true"><span>Figure S2:</span> <strong>Processing additional single-cell modalities in <code>scpca-nf</code>.</strong></figcaption>
</figure>
</div>
<p>A. Overview of the <code>scpca-nf</code> workflow for processing libraries with CITE-seq or antibody-derived tag (ADT) derived data.
The workflow mirrors that shown in Figure <a href="#fig:fig2">2</a>A with several differences accounting for the presence of ADT data.
First, both an RNA and ADT FASTQ file are required as input to <code>alevin-fry</code>, along with a TSV file containing information about ADT barcodes.
The gene by cell and ADT by cell count matrices are produced and read into <code>R</code> to create a <code>SingleCellExperiment</code> (SCE) object.
Second, during post-processing, statistics are calculated to filter cells based on ADT counts, but the filter is not applied.
ADT counts are also normalized and included in the <code>Processed SCE Object</code>.
Third, the summary QC report will include a <code>CITE-seq</code> section with additional information about ADT-level processing.
Fourth, the workflow exports <code>SCE</code> objects containing both RNA and ADT results, while separate <code>AnnData</code> objects for RNA and ADT are exported.</p>
<p>Panels B-D show example figures that appear in the CITE-seq section of the summary QC report, shown here for <code>SCPCL000290</code>.</p>
<p>B. The percent of mitochondrial reads in each cell against the number of genes detected in each cell.
The panel labeled “Keep” displays cells that are retained based on both RNA and ADT counts.
The panel labeled “Filter (ADT only)” displays cells that are filtered based on only ADT counts.
The panel labeled “Filter (RNA only)” displays cells that are filtered based on only RNA counts.
The panel labeled “Filter (RNA &amp; ADT)” panel displays cells that are filtered based on both RNA and ADT counts.</p>
<p>C. Density plots of the log-normalized ADT counts shown for the four most variable ADTs in the library.</p>
<p>D. UMAP embeddings of log-normalized RNA expression values where each cell is colored by the expression of the given highly-variable ADT.</p>
<p>E. Overview of the <code>scpca-nf</code> workflow for multiplexed libraries.
The workflow mirrors that shown in Figure <a href="#fig:fig2">2</a>A with several differences accounting for the presence of multiplexed data.
First, both an RNA and HTO FASTQ file are required as input to <code>alevin-fry</code>, along with a TSV file providing information about library pools.
The gene by cell and HTO by cell count matrices are produced and read into <code>R</code> to create a <code>SingleCellExperiment</code> (SCE) object.
Second, in parallel, the RNA FASTQ file, the HTO FASTQ file, and, if available, a corresponding Bulk RNA FASTQ file for each sample present in the multiplexed library are provided to a demultiplexing subprocess.
The workflow calculates demultiplexing results based on HTO counts, as well as genetic demultiplexing results if the library has corresponding bulk RNA FASTQ files.
Demultiplexing results are stored in all exported <code>SCE</code> objects (<code>Unfiltered</code>, <code>Filtered</code>, and <code>Processed</code>), but libraries themselves are not demultiplexed.
Third, only <code>SCE</code> files are provided for multiplexed libraries; no corresponding <code>AnnData</code> files are provided.
<br><br></p>
<!-- Figure S3 -->
<div id="fig:figS3" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s3.png?sanitize=true" style="width:7in" data-tag="S3" alt="Figure S3: Processing other sequencing modalities with scpca-nf." />
<figcaption aria-hidden="true"><span>Figure S3:</span> <strong>Processing other sequencing modalities with <code>scpca-nf</code>.</strong></figcaption>
</figure>
</div>
<p>A. Overview of the bulk RNA-Seq workflow.
A set of FASTQ files from libraries sequenced with bulk RNA-seq are provided as input.
Reads are trimmed using <code>fastp</code>, and <code>salmon</code> is used to map reads and quantify counts.
The quantified gene expression files output from <code>salmon</code> are then grouped by ScPCA Project ID, and a sample by gene count matrix is exported for each Project in TSV format.</p>
<p>B. Overview of the spatial transcriptomics workflow.
The FASTQ file and tissue image for a given library are provided as input to <code>spaceranger</code>.
The workflow directly returns the results from running <code>spaceranger</code> without any further processing.
<br><br></p>
<!--Figure S4-->
<div id="fig:figS4" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s4.png?sanitize=true" style="width:7in" data-tag="S4" alt="Figure S4: Evaluation of references available in the celldex package for use with SingleR." />
<figcaption aria-hidden="true"><span>Figure S4:</span> <strong>Evaluation of references available in the celldex package for use with SingleR.</strong></figcaption>
</figure>
</div>
<p><code>SingleR</code> was used to annotate ScPCA libraries using four different human-specific references from the <code>celldex</code> package.
Libraries represent three different diagnosis groups in the ScPCA Portal - Brain and CNS, Leukemia, and Sarcoma - as indicated in the labels for the individual panels.
The distribution of the delta median statistic, calculated for each cell by subtracting the median delta score from the score of the annotated cell type label, is shown on the y-axis, while the <code>celldex</code> reference used is shown on the x-axis.
Higher values indicate a higher quality cell type annotation, although there is no absolute scale for these values.
Each black point represents a cell, where closed circles denote cells with high-quality annotations and open circles denote cells with low-quality annotations, as assessed by <code>SingleR</code>.
Red diamonds represent the median delta median score for all cells with high-quality annotations in that library.
<br><br></p>
<!--Figure S5-->
<div id="fig:figS5" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s5.png?sanitize=true" style="height:10in" data-tag="S5" alt="Figure S5: Cell type annotation with CellAssign." />
<figcaption aria-hidden="true"><span>Figure S5:</span> <strong>Cell type annotation with <code>CellAssign</code>.</strong></figcaption>
</figure>
</div>
<p>Both plots in this figure are examples of plots that display results from annotating cells with <code>CellAssign</code> that can be found in the cell type summary report, shown here for library <code>SCPCL000498</code> <span class="citation" data-cites="9F0oWAO0">[<a href="#ref-9F0oWAO0" role="doc-biblioref">25</a>]</span>.</p>
<p>A. A grid of UMAPs is shown for each cell type annotated using <code>CellAssign</code>, with the cell type of interest shown in color and all other cells belonging to other cell types shown in gray.
The top four cell types with the greatest number of assigned cells are shown, while all other cells are grouped together and labeled with <code>All remaining cell types</code>.
Any cells that are unable to be assigned by <code>CellAssign</code> are labeled with <code>Unknown cell type</code>.</p>
<p>B. This example heatmap from the cell type summary report compares submitter-provided annotations to annotations with <code>SingleR</code> and <code>CellAssign</code>.
This heatmap is only shown in the cell type summary report if submitters provided cell type annotations.
Heatmap cells are colored by the Jaccard similarity index.
A value of 1 means that there is complete overlap between which cells are annotated with the two labels being compared, and a value of 0 means that there is no overlap between which cells are annotated with the two labels being compared.</p>
<!-- Figure S6 -->
<div id="fig:figS6" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s6.png?sanitize=true" style="width:9in" data-tag="S6" alt="Figure S6: Consensus cell type annotation gene expression in other diagnosis groups." />
<figcaption aria-hidden="true"><span>Figure S6:</span> <strong>Consensus cell type annotation gene expression in other diagnosis groups.</strong></figcaption>
</figure>
</div>
<p>Dot plots showing expression of cell-type-specific marker genes across all libraries from Leukemia (A), Sarcoma (B), and Other solid tumors (C) diagnosis groups.
Expression is shown for each broad cell type annotation, where each broad cell type annotation is a collection of similar consensus cell type annotations.
The y-axis displays the broad consensus cell type observed across libraries, with the total number of cells indicated in parentheses.
The x-axis displays marker genes, determined by <code>CellMarker2.0</code> <span class="citation" data-cites="MclvBVPD">[<a href="#ref-MclvBVPD" role="doc-biblioref">71</a>]</span>, used for consensus cell type validation for each cell type shown along the top annotation bar.
Dots are colored by mean gene expression across libraries and sized proportionally to the percent of libraries they are observed in, out of all cells with the same broad cell type annotation in the given diagnosis.
<br><br></p>
<!-- Figure S7 -->
<div id="fig:figS7" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s7.png?sanitize=true" style="width:7in" data-tag="S7" alt="Figure S7: *Consensus cell type annotation distributions in other diagnosis groups." />
<figcaption aria-hidden="true"><span>Figure S7:</span> *<strong>Consensus cell type annotation distributions in other diagnosis groups.</strong></figcaption>
</figure>
</div>
<p>Barplots of the percentage of cells annotated as each broad consensus cell type annotation across all libraries from Leukemia (A), Sarcoma (B), and Other solid tumors (C) diagnosis groups.
Within each panel, libraries are shown grouped by diagnosis.
Each column represents the distribution of cell types within a single library.
<br><br></p>
<!-- Figure S8 -->
<div id="fig:figS8" class="fignos">
<figure>
<img src="https://raw.githubusercontent.com/AlexsLemonade/scpca-paper-figures/main/figures/compiled_figures/pngs/figure_s8.png?sanitize=true" style="width:7in" data-tag="S8" alt="Figure S8: Comparison of bulk and pseudobulk modalities for additional projects." />
<figcaption aria-hidden="true"><span>Figure S8:</span> <strong>Comparison of bulk and pseudobulk modalities for additional projects.</strong></figcaption>
</figure>
</div>
<p>A. Scatter plots colored by point density of <code>DESeq2</code>-transformed and normalized bulk RNA-seq expression compared to pseudobulk expression from single-nuclei RNA-seq.
Projects with RNA-seq for both bulk and single-cell/nuclei modalities that are not displayed in Figure <a href="#fig:fig6">6</a>A are shown.
All samples shown here are single-nuclei, and the number of samples considered per project is shown in parentheses.
The regression line is also shown for each project.</p>
<p>B. Odds ratios from overrepresentation analysis for the same samples shown in panel A, colored by FDR-corrected significance.
Each odds ratio represents the odds that marker genes for the given cell type were overrepresented in the bulk modality, relative to other genes.
31 consensus cell types were evaluated for project <code>SCPCP000006</code>, and 37 consensus cell types were evaluated for project <code>SCPCP000017</code>.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-ALU8zyix" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Exponential scaling of single-cell RNA-seq in the past decade</strong> <div class="csl-block">Valentine Svensson, Roser Vento-Tormo, Sarah A Teichmann</div> <em>Nature Protocols</em> (2018-03-01) <a href="https://doi.org/gc5ndt">https://doi.org/gc5ndt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nprot.2017.149">10.1038/nprot.2017.149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29494575">29494575</a></div></div>
</div>
<div id="ref-e6vwOccq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Defining cell types and states with single-cell genomics</strong> <div class="csl-block">Cole Trapnell</div> <em>Genome Research</em> (2015-10) <a href="https://doi.org/f7st9g">https://doi.org/f7st9g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/gr.190595.115">10.1101/gr.190595.115</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26430159">26430159</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4579334">PMC4579334</a></div></div>
</div>
<div id="ref-CK2iL88z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Single-cell RNA-seq highlights intratumoral heterogeneity in primary glioblastoma</strong> <div class="csl-block">Anoop P Patel, Itay Tirosh, John J Trombetta, Alex K Shalek, Shawn M Gillespie, Hiroaki Wakimoto, Daniel P Cahill, Brian V Nahed, William T Curry, Robert L Martuza, … Bradley E Bernstein</div> <em>Science</em> (2014-06-20) <a href="https://doi.org/gdm4dv">https://doi.org/gdm4dv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.1254257">10.1126/science.1254257</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24925914">24925914</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4123637">PMC4123637</a></div></div>
</div>
<div id="ref-10Z0RGQgO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>Cancer cell states recur across tumor types and form specific interactions with the tumor microenvironment</strong> <div class="csl-block">Dalia Barkley, Reuben Moncada, Maayan Pour, Deborah A Liberman, Ian Dryg, Gregor Werba, Wei Wang, Maayan Baron, Anjali Rao, Bo Xia, … Itai Yanai</div> <em>Nature Genetics</em> (2022-08) <a href="https://doi.org/gqtn64">https://doi.org/gqtn64</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41588-022-01141-9">10.1038/s41588-022-01141-9</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35931863">35931863</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9886402">PMC9886402</a></div></div>
</div>
<div id="ref-vk9ZInF3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>The Human Cell Atlas</strong> <div class="csl-block">Aviv Regev, Sarah A Teichmann, Eric S Lander, Ido Amit, Christophe Benoist, Ewan Birney, Bernd Bodenmiller, Peter Campbell, Piero Carninci, Menna Clatworthy, … </div> <em>eLife</em> (2017-12-05) <a href="https://doi.org/gcnzcv">https://doi.org/gcnzcv</a> <div class="csl-block">DOI: <a href="https://doi.org/10.7554/elife.27041">10.7554/elife.27041</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29206104">29206104</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5762154">PMC5762154</a></div></div>
</div>
<div id="ref-udPM4KR1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>The Human Tumor Atlas Network: Charting Tumor Transitions across Space and Time at Single-Cell Resolution</strong> <div class="csl-block">Orit Rozenblatt-Rosen, Aviv Regev, Philipp Oberdoerffer, Tal Nawy, Anna Hupalowska, Jennifer E Rood, Orr Ashenberg, Ethan Cerami, Robert J Coffey, Emek Demir, … Xiaowei Zhuang</div> <em>Cell</em> (2020-04) <a href="https://doi.org/ggtkzd">https://doi.org/ggtkzd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cell.2020.03.053">10.1016/j.cell.2020.03.053</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32302568">32302568</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7376497">PMC7376497</a></div></div>
</div>
<div id="ref-Hqoq5ZTI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Cancer in Children and Adolescents - NCI</strong> (2024-08-29) <a href="https://www.cancer.gov/types/childhood-cancers/child-adolescent-cancers-fact-sheet">https://www.cancer.gov/types/childhood-cancers/child-adolescent-cancers-fact-sheet</a></div>
</div>
<div id="ref-yEQ2v42v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>Use case driven evaluation of open databases for pediatric cancer research</strong> <div class="csl-block">Fleur Jeanquartier, Claire Jean-Quartier, Andreas Holzinger</div> <em>BioData Mining</em> (2019-01-15) <a href="https://doi.org/ggjv7q">https://doi.org/ggjv7q</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13040-018-0190-8">10.1186/s13040-018-0190-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30675185">30675185</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6334395">PMC6334395</a></div></div>
</div>
<div id="ref-z8JnDHpr" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Orchestrating single-cell analysis with Bioconductor</strong> <div class="csl-block">Robert A Amezquita, Aaron TL Lun, Etienne Becht, Vince J Carey, Lindsay N Carpp, Ludwig Geistlinger, Federico Marini, Kevin Rue-Albrecht, Davide Risso, Charlotte Soneson, … Stephanie C Hicks</div> <em>Nature Methods</em> (2019-12-02) <a href="https://doi.org/ggdxgx">https://doi.org/ggdxgx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0654-x">10.1038/s41592-019-0654-x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31792435">31792435</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7358058">PMC7358058</a></div></div>
</div>
<div id="ref-qIZTS1II" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>SCANPY: large-scale single-cell gene expression data analysis</strong> <div class="csl-block">FAlexander Wolf, Philipp Angerer, Fabian J Theis</div> <em>Genome Biology</em> (2018-02-06) <a href="https://doi.org/gc22s9">https://doi.org/gc22s9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-017-1382-0">10.1186/s13059-017-1382-0</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29409532">29409532</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5802054">PMC5802054</a></div></div>
</div>
<div id="ref-4XDvZWxk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Nextflow enables reproducible computational workflows</strong> <div class="csl-block">Paolo Di Tommaso, Maria Chatzou, Evan W Floden, Pablo Prieto Barja, Emilio Palumbo, Cedric Notredame</div> <em>Nature Biotechnology</em> (2017-04) <a href="https://doi.org/gfj52z">https://doi.org/gfj52z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nbt.3820">10.1038/nbt.3820</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28398311">28398311</a></div></div>
</div>
<div id="ref-hYvj5GBC" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Alevin-fry unlocks rapid, accurate and memory-frugal quantification of single-cell RNA-seq data</strong> <div class="csl-block">Dongze He, Mohsen Zakeri, Hirak Sarkar, Charlotte Soneson, Avi Srivastava, Rob Patro</div> <em>Nature Methods</em> (2022-03) <a href="https://doi.org/gptg86">https://doi.org/gptg86</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-022-01408-3">10.1038/s41592-022-01408-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35277707">35277707</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8933848">PMC8933848</a></div></div>
</div>
<div id="ref-2n9clgPF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS) - HsapDv</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/hsapdv">https://www.ebi.ac.uk/ols4/ontologies/hsapdv</a></div>
</div>
<div id="ref-Jx3SGqPR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>The anatomy of phenotype ontologies: principles, properties and applications</strong> <div class="csl-block">Georgios V Gkoutos, Paul N Schofield, Robert Hoehndorf</div> <em>Briefings in Bioinformatics</em> (2017-04-06) <a href="https://doi.org/gk3928">https://doi.org/gk3928</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bib/bbx035">10.1093/bib/bbx035</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28387809">28387809</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6169674">PMC6169674</a></div></div>
</div>
<div id="ref-19R1xim7A" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS) - PATO</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/pato">https://www.ebi.ac.uk/ols4/ontologies/pato</a></div>
</div>
<div id="ref-RJLosfFq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>NCBI Taxonomy: a comprehensive update on curation, resources and tools</strong> <div class="csl-block">Conrad L Schoch, Stacy Ciufo, Mikhail Domrachev, Carol L Hotton, Sivakumar Kannan, Rogneda Khovanskaya, Detlef Leipe, Richard Mcveigh, Kathleen O’Neill, Barbara Robbertse, … Ilene Karsch-Mizrachi</div> <em>Database</em> (2020-01-01) <a href="https://doi.org/gg7tjn">https://doi.org/gg7tjn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baaa062">10.1093/database/baaa062</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32761142">32761142</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7408187">PMC7408187</a></div></div>
</div>
<div id="ref-jFq6Uo6f" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>Home - Taxonomy - NCBI</strong> <a href="https://www.ncbi.nlm.nih.gov/taxonomy">https://www.ncbi.nlm.nih.gov/taxonomy</a></div>
</div>
<div id="ref-pZyZllZn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Mondo: Unifying diseases for the world, by the world</strong> <div class="csl-block">Nicole A Vasilevsky, Nicolas A Matentzoglu, Sabrina Toro, Joseph E Flack IV, Harshad Hegde, Deepak R Unni, Gioconda F Alyea, Joanna S Amberger, Larry Babb, James P Balhoff, … Melissa A Haendel</div> <em>Cold Spring Harbor Laboratory</em> (2022-04-16) <a href="https://doi.org/gqx27c">https://doi.org/gqx27c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.04.13.22273750">10.1101/2022.04.13.22273750</a></div></div>
</div>
<div id="ref-ZPHZMiua" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS) - MONDO</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/mondo">https://www.ebi.ac.uk/ols4/ontologies/mondo</a></div>
</div>
<div id="ref-swLRou1q" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Unification of multi-species vertebrate anatomy ontologies for comparative biology in Uberon</strong> <div class="csl-block">Melissa A Haendel, James P Balhoff, Frederic B Bastian, David C Blackburn, Judith A Blake, Yvonne Bradford, Aurelie Comte, Wasila M Dahdul, Thomas A Dececchi, Robert E Druzinsky, … Christopher J Mungall</div> <em>Journal of Biomedical Semantics</em> (2014) <a href="https://doi.org/gtnrz4">https://doi.org/gtnrz4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/2041-1480-5-21">10.1186/2041-1480-5-21</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25009735">25009735</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4089931">PMC4089931</a></div></div>
</div>
<div id="ref-TOC8jpgh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>Uberon, an integrative multi-species anatomy ontology</strong> <div class="csl-block">Christopher J Mungall, Carlo Torniai, Georgios V Gkoutos, Suzanna E Lewis, Melissa A Haendel</div> <em>Genome Biology</em> (2012-01-31) <a href="https://doi.org/fxx6qr">https://doi.org/fxx6qr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/gb-2012-13-1-r5">10.1186/gb-2012-13-1-r5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22293552">22293552</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334586">PMC3334586</a></div></div>
</div>
<div id="ref-XbjdiPPQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS) - UBERON</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/uberon">https://www.ebi.ac.uk/ols4/ontologies/uberon</a></div>
</div>
<div id="ref-9Hno2ty7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>A standardized framework for representation of ancestry data in genomics studies, with application to the NHGRI-EBI GWAS Catalog</strong> <div class="csl-block">Joannella Morales, Danielle Welter, Emily H Bowler, Maria Cerezo, Laura W Harris, Aoife C McMahon, Peggy Hall, Heather A Junkins, Annalisa Milano, Emma Hastings, … Jacqueline AL MacArthur</div> <em>Genome Biology</em> (2018-02-15) <a href="https://doi.org/gf9pk3">https://doi.org/gf9pk3</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-018-1396-2">10.1186/s13059-018-1396-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29448949">29448949</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5815218">PMC5815218</a></div></div>
</div>
<div id="ref-Opr5odMH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS) - Hancestro</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/hancestro">https://www.ebi.ac.uk/ols4/ontologies/hancestro</a></div>
</div>
<div id="ref-9F0oWAO0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>The myogenesis program drives clonal selection and drug resistance in rhabdomyosarcoma</strong> <div class="csl-block">Anand G Patel, Xiang Chen, Xin Huang, Michael R Clay, Natalia L Komarova, Matthew J Krasin, Alberto Pappo, Heather Tillman, Brent A Orr, Justina McEvoy, … Michael A Dyer</div> <em>Developmental Cell</em> (2022-05) <a href="https://doi.org/gtnskk">https://doi.org/gtnskk</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.devcel.2022.04.003">10.1016/j.devcel.2022.04.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35483358">35483358</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9133224">PMC9133224</a></div></div>
</div>
<div id="ref-HalHlba8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>Tumor type and cell type-specific gene expression alterations in diverse pediatric central nervous system tumors identified using single nuclei RNA-seq</strong> <div class="csl-block">Min Kyung Lee, Nasim Azizgolshani, Joshua Shapiro, Lananh Nguyen, Fred Kolling IV, George Zanazzi, Hildredth Frost, Brock Christensen</div> <em>Research Square Platform LLC</em> (2023-02-23) <a href="https://doi.org/gtnskp">https://doi.org/gtnskp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21203/rs.3.rs-2517703/v1">10.21203/rs.3.rs-2517703/v1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36865335">36865335</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9980204">PMC9980204</a></div></div>
</div>
<div id="ref-TffkWyy0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Hydroxymethylation alterations in progenitor-like cell types of pediatric central nervous system tumors are associated with cell type-specific transcriptional changes</strong> <div class="csl-block">Min Kyung Lee, Nasim Azizgolshani, Ze Zhang, Laurent Perreard, Fred Kolling IV, Lananh Nguyen, George Zanazzi, Lucas Salas, Brock Christensen</div> <em>Research Square Platform LLC</em> (2023-02-28) <a href="https://doi.org/gtnskq">https://doi.org/gtnskq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.21203/rs.3.rs-2517758/v1">10.21203/rs.3.rs-2517758/v1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36909536">36909536</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10002842">PMC10002842</a></div></div>
</div>
<div id="ref-sgFUrpYG" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>Orthotopic patient-derived xenografts of paediatric solid tumours</strong> <div class="csl-block">Elizabeth Stewart, Sara M Federico, Xiang Chen, Anang A Shelat, Cori Bradley, Brittney Gordon, Asa Karlstrom, Nathaniel R Twarog, Michael R Clay, Armita Bahrami, … Michael A Dyer</div> <em>Nature</em> (2017-08-30) <a href="https://doi.org/gbs7cx">https://doi.org/gbs7cx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature23647">10.1038/nature23647</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28854174">28854174</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5659286">PMC5659286</a></div></div>
</div>
<div id="ref-PJ7Veojz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Retinoblastoma from human stem cell-derived retinal organoids</strong> <div class="csl-block">Jackie L Norrie, Anjana Nityanandam, Karen Lai, Xiang Chen, Matthew Wilson, Elizabeth Stewart, Lyra Griffiths, Hongjian Jin, Gang Wu, Brent Orr, … Michael A Dyer</div> <em>Nature Communications</em> (2021-07-27) <a href="https://doi.org/gq28m5">https://doi.org/gq28m5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-24781-7">10.1038/s41467-021-24781-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34315877">34315877</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8316454">PMC8316454</a></div></div>
</div>
<div id="ref-U3XexzPk" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Tumor and immune cell types interact to produce heterogeneous phenotypes of pediatric high-grade glioma</strong> <div class="csl-block">John DeSisto, Andrew M Donson, Andrea M Griesinger, Rui Fu, Kent Riemondy, Jean Mulcahy Levy, Julie A Siegenthaler, Nicholas K Foreman, Rajeev Vibhakar, Adam L Green</div> <em>Neuro-Oncology</em> (2023-11-02) <a href="https://doi.org/gtnskm">https://doi.org/gtnskm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/neuonc/noad207">10.1093/neuonc/noad207</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37934854">37934854</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10912009">PMC10912009</a></div></div>
</div>
<div id="ref-9Xh0e2DI" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Single-cell transcriptional mapping reveals genetic and non-genetic determinants of aberrant differentiation in AML</strong> <div class="csl-block">Andy GX Zeng, Ilaria Iacobucci, Sayyam Shah, Amanda Mitchell, Gordon Wong, Suraj Bansal, David Chen, Qingsong Gao, Hyerin Kim, James A Kennedy, … John E Dick</div> <em>Cold Spring Harbor Laboratory</em> (2023-12-27) <a href="https://doi.org/gtnskn">https://doi.org/gtnskn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.12.26.573390">10.1101/2023.12.26.573390</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38234771">38234771</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10793439">PMC10793439</a></div></div>
</div>
<div id="ref-abp5rbw2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Simultaneous epitope and transcriptome measurement in single cells</strong> <div class="csl-block">Marlon Stoeckius, Christoph Hafemeister, William Stephenson, Brian Houck-Loomis, Pratip K Chattopadhyay, Harold Swerdlow, Rahul Satija, Peter Smibert</div> <em>Nature Methods</em> (2017-07-31) <a href="https://doi.org/gfkksd">https://doi.org/gfkksd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.4380">10.1038/nmeth.4380</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28759029">28759029</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5669064">PMC5669064</a></div></div>
</div>
<div id="ref-FZrs73Vp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Cell Hashing with barcoded antibodies enables multiplexing and doublet detection for single cell genomics</strong> <div class="csl-block">Marlon Stoeckius, Shiwei Zheng, Brian Houck-Loomis, Stephanie Hao, Bertrand Z Yeung, William M Mauck III, Peter Smibert, Rahul Satija</div> <em>Genome Biology</em> (2018-12) <a href="https://doi.org/ggbm6p">https://doi.org/ggbm6p</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-018-1603-1">10.1186/s13059-018-1603-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30567574">30567574</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6300015">PMC6300015</a></div></div>
</div>
<div id="ref-1CuNC7bWt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Massively parallel digital transcriptional profiling of single cells</strong> <div class="csl-block">Grace XY Zheng, Jessica M Terry, Phillip Belgrader, Paul Ryvkin, Zachary W Bent, Ryan Wilson, Solongo B Ziraldo, Tobias D Wheeler, Geoff P McDermott, Junjie Zhu, … Jason H Bielas</div> <em>Nature Communications</em> (2017-01-16) <a href="https://doi.org/f9mbtp">https://doi.org/f9mbtp</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/ncomms14049">10.1038/ncomms14049</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28091601">28091601</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5241818">PMC5241818</a></div></div>
</div>
<div id="ref-87PPOPmp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>Cell Ranger</strong> <div class="csl-block">10x Genomics</div> <a href="https://www.10xgenomics.com/support/software/cell-ranger/latest">https://www.10xgenomics.com/support/software/cell-ranger/latest</a></div>
</div>
<div id="ref-159Z7ABXf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>AlexsLemonade/alsf-scpca</strong> <div class="csl-block">Alex's Lemonade Stand Foundation</div> (2021-12-21) <a href="https://github.com/AlexsLemonade/alsf-scpca">https://github.com/AlexsLemonade/alsf-scpca</a></div>
</div>
<div id="ref-U2Fa25dp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>EmptyDrops: distinguishing cells from empty droplets in droplet-based single-cell RNA sequencing data</strong><div class="csl-block">, Aaron TL Lun, Samantha Riesenfeld, Tallulah Andrews, The Phuong Dao, Tomas Gomes, John C Marioni</div> <em>Genome Biology</em> (2019-03-22) <a href="https://doi.org/gfxdhf">https://doi.org/gfxdhf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-019-1662-y">10.1186/s13059-019-1662-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30902100">30902100</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6431044">PMC6431044</a></div></div>
</div>
<div id="ref-17Yhk1HAt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>miQC: An adaptive probabilistic framework for quality control of single-cell RNA-sequencing data</strong> <div class="csl-block">Ariel A Hippen, Matias M Falco, Lukas M Weber, Erdogan Pekcan Erkan, Kaiyang Zhang, Jennifer Anne Doherty, Anna Vähärautio, Casey S Greene, Stephanie C Hicks</div> <em>PLOS Computational Biology</em> (2021-08-24) <a href="https://doi.org/gng37g">https://doi.org/gng37g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1371/journal.pcbi.1009290">10.1371/journal.pcbi.1009290</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34428202">34428202</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8415599">PMC8415599</a></div></div>
</div>
<div id="ref-vpU0FTFY" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Pooling across cells to normalize single-cell RNA sequencing data with many zero counts</strong> <div class="csl-block">Aaron T L. Lun, Karsten Bach, John C Marioni</div> <em>Genome Biology</em> (2016-04-27) <a href="https://doi.org/gfgntn">https://doi.org/gfgntn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-016-0947-7">10.1186/s13059-016-0947-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27122128">27122128</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4848819">PMC4848819</a></div></div>
</div>
<div id="ref-157h5hA34" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</strong> <div class="csl-block">Leland McInnes, John Healy, James Melville</div> <em>arXiv</em> (2020-09-21) <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a></div>
</div>
<div id="ref-13P6oSrS0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Reference-based analysis of lung single-cell sequencing reveals a transitional profibrotic macrophage</strong> <div class="csl-block">Dvir Aran, Agnieszka P Looney, Leqian Liu, Esther Wu, Valerie Fong, Austin Hsu, Suzanna Chak, Ram P Naikawadi, Paul J Wolters, Adam R Abate, … Mallar Bhattacharya</div> <em>Nature Immunology</em> (2019-01-14) <a href="https://doi.org/gfv3p2">https://doi.org/gfv3p2</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41590-018-0276-y">10.1038/s41590-018-0276-y</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30643263">30643263</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6340744">PMC6340744</a></div></div>
</div>
<div id="ref-HES0WO6k" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>Probabilistic cell-type assignment of single-cell RNA-seq for tumor microenvironment profiling</strong> <div class="csl-block">Allen W Zhang, Ciara O’Flanagan, Elizabeth A Chavez, Jamie LP Lim, Nicholas Ceglia, Andrew McPherson, Matt Wiens, Pascale Walters, Tim Chan, Brittany Hewitson, … Sohrab P Shah</div> <em>Nature Methods</em> (2019-09-09) <a href="https://doi.org/ggr7ps">https://doi.org/ggr7ps</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-019-0529-1">10.1038/s41592-019-0529-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31501550">31501550</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7485597">PMC7485597</a></div></div>
</div>
<div id="ref-KRBW8xeZ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>anndata: Annotated data</strong> <div class="csl-block">Isaac Virshup, Sergei Rybakov, Fabian J Theis, Philipp Angerer, FAlexander Wolf</div> <em>Cold Spring Harbor Laboratory</em> (2021-12-19) <a href="https://doi.org/gst7w6">https://doi.org/gst7w6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.12.16.473007">10.1101/2021.12.16.473007</a></div></div>
</div>
<div id="ref-mX3LKFI8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>scuttle</strong> <div class="csl-block">Aaron Lun, Davis McCarthy</div> <em>Bioconductor</em> (2020) <a href="https://doi.org/gtkc7k">https://doi.org/gtkc7k</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18129/b9.bioc.scuttle">10.18129/b9.bioc.scuttle</a></div></div>
</div>
<div id="ref-4kYmrOnP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">45. </div><div class="csl-right-inline"><strong>Chapter 12 Integrating with protein abundance | Advanced Single-Cell Analysis with Bioconductor</strong> <a href="https://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#cite-seq-median-norm">https://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#cite-seq-median-norm</a></div>
</div>
<div id="ref-NqZ0Pm2u" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">46. </div><div class="csl-right-inline"><strong>DropletUtils</strong> <div class="csl-block">Jonathan Griffiths Aaron Lun</div> <em>Bioconductor</em> (2018) <a href="https://doi.org/gtkc7d">https://doi.org/gtkc7d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18129/b9.bioc.dropletutils">10.18129/b9.bioc.dropletutils</a></div></div>
</div>
<div id="ref-zLp25QYu" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">47. </div><div class="csl-right-inline"><strong>Genetic demultiplexing of pooled single-cell RNA-sequencing samples in cancer facilitates effective experimental design</strong> <div class="csl-block">Lukas M Weber, Ariel A Hippen, Peter F Hickey, Kristofer C Berrett, Jason Gertz, Jennifer Anne Doherty, Casey S Greene, Stephanie C Hicks</div> <em>GigaScience</em> (2021-09) <a href="https://doi.org/gmwhsc">https://doi.org/gmwhsc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giab062">10.1093/gigascience/giab062</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34553212">34553212</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8458035">PMC8458035</a></div></div>
</div>
<div id="ref-pt9vlq4w" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">48. </div><div class="csl-right-inline"><strong>fastp: an ultra-fast all-in-one FASTQ preprocessor</strong> <div class="csl-block">Shifu Chen, Yanqing Zhou, Yaru Chen, Jia Gu</div> <em>Bioinformatics</em> (2018-09-01) <a href="https://doi.org/gd9mrb">https://doi.org/gd9mrb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bty560">10.1093/bioinformatics/bty560</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30423086">30423086</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6129281">PMC6129281</a></div></div>
</div>
<div id="ref-vrqQcFyx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">49. </div><div class="csl-right-inline"><strong>Salmon provides fast and bias-aware quantification of transcript expression</strong> <div class="csl-block">Rob Patro, Geet Duggal, Michael I Love, Rafael A Irizarry, Carl Kingsford</div> <em>Nature Methods</em> (2017-03-06) <a href="https://doi.org/gcw9f5">https://doi.org/gcw9f5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28263959">28263959</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5600148">PMC5600148</a></div></div>
</div>
<div id="ref-xo912CsO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">50. </div><div class="csl-right-inline"><strong>Space Ranger</strong> <div class="csl-block">10x Genomics</div> <a href="https://www.10xgenomics.com/support/software/space-ranger/latest">https://www.10xgenomics.com/support/software/space-ranger/latest</a></div>
</div>
<div id="ref-xpZq9dBF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">51. </div><div class="csl-right-inline"><strong>Chapter 4 Annotation diagnostics | Assigning cell types with SingleR</strong> <a href="https://bioconductor.org/books/release/SingleRBook/annotation-diagnostics.html#based-on-the-deltas-across-cells">https://bioconductor.org/books/release/SingleRBook/annotation-diagnostics.html#based-on-the-deltas-across-cells</a></div>
</div>
<div id="ref-MzCvXTgB" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">52. </div><div class="csl-right-inline"><strong>BLUEPRINT: mapping human blood cell epigenomes</strong> <div class="csl-block">JHA Martens, HG Stunnenberg</div> <em>Haematologica</em> (2013-10-01) <a href="https://doi.org/gd2xz4">https://doi.org/gd2xz4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3324/haematol.2013.094243">10.3324/haematol.2013.094243</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24091925">24091925</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3789449">PMC3789449</a></div></div>
</div>
<div id="ref-15J98V2qM" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">53. </div><div class="csl-right-inline"><strong>An integrated encyclopedia of DNA elements in the human genome</strong> <em>Nature</em> (2012-09) <a href="https://doi.org/bg9d">https://doi.org/bg9d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/nature11247">10.1038/nature11247</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22955616">22955616</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3439153">PMC3439153</a></div></div>
</div>
<div id="ref-T3PlZ3Vh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">54. </div><div class="csl-right-inline"><strong>PanglaoDB: a web server for exploration of mouse and human single-cell RNA sequencing data</strong> <div class="csl-block">Oscar Franzén, Li-Ming Gan, Johan LM Björkegren</div> <em>Database</em> (2019-01-01) <a href="https://doi.org/ggkzxr">https://doi.org/ggkzxr</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/database/baz046">10.1093/database/baz046</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30951143">30951143</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6450036">PMC6450036</a></div></div>
</div>
<div id="ref-17SbnOJiL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">55. </div><div class="csl-right-inline"><strong>findCommonAncestors: Find common ancestors in ontoProc: processing of ontologies of anatomy, cell lines, and so on</strong> <a href="https://rdrr.io/bioc/ontoProc/man/findCommonAncestors.html">https://rdrr.io/bioc/ontoProc/man/findCommonAncestors.html</a></div>
</div>
<div id="ref-yxJiICJJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">56. </div><div class="csl-right-inline"><strong>Therapeutic approaches to modulate the immune microenvironment in gliomas</strong> <div class="csl-block">Andreas Sarantopoulos, Chibawanye Ene, Elisa Aquilanti</div> <em>npj Precision Oncology</em> (2024-10-23) <a href="https://doi.org/g9bjzf">https://doi.org/g9bjzf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41698-024-00717-4">10.1038/s41698-024-00717-4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/39443641">39443641</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC11500177">PMC11500177</a></div></div>
</div>
<div id="ref-9U8b91g7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">57. </div><div class="csl-right-inline"><strong>Tumor microenvironment in glioblastoma: Current and emerging concepts</strong> <div class="csl-block">Pratibha Sharma, Ashley Aaroe, Jiyong Liang, Vinay K Puduvalli</div> <em>Neuro-Oncology Advances</em> (2023-01-01) <a href="https://doi.org/g9bjzg">https://doi.org/g9bjzg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/noajnl/vdad009">10.1093/noajnl/vdad009</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36968288">36968288</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10034917">PMC10034917</a></div></div>
</div>
<div id="ref-Vz4pQdwU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">58. </div><div class="csl-right-inline"><strong>Systematic comparison of single-cell and single-nucleus RNA-sequencing methods</strong> <div class="csl-block">Jiarui Ding, Xian Adiconis, Sean K Simmons, Monika S Kowalczyk, Cynthia C Hession, Nemanja D Marjanovic, Travis K Hughes, Marc H Wadsworth, Tyler Burks, Lan T Nguyen, … Joshua Z Levin</div> <em>Nature Biotechnology</em> (2020-04-06) <a href="https://doi.org/ggrksw">https://doi.org/ggrksw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41587-020-0465-8">10.1038/s41587-020-0465-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32341560">32341560</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7289686">PMC7289686</a></div></div>
</div>
<div id="ref-2x15P0J5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">59. </div><div class="csl-right-inline"><strong>The tumor micro-environment in pediatric glioma: friend or foe?</strong> <div class="csl-block">Julie Messiaen, Sandra A Jacobs, Frederik De Smet</div> <em>Frontiers in Immunology</em> (2023-10-13) <a href="https://doi.org/g9bjwt">https://doi.org/g9bjwt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fimmu.2023.1227126">10.3389/fimmu.2023.1227126</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37901250">37901250</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10611473">PMC10611473</a></div></div>
</div>
<div id="ref-DR9mGrXH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">60. </div><div class="csl-right-inline"><strong>Insights into the glioblastoma tumor microenvironment: current and emerging therapeutic approaches</strong> <div class="csl-block">Dev Kumar Tripathy, Lakshmi Priya Panda, Suryanarayan Biswal, Kalpana Barhwal</div> <em>Frontiers in Pharmacology</em> (2024-03-08) <a href="https://doi.org/g9bnw8">https://doi.org/g9bnw8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fphar.2024.1355242">10.3389/fphar.2024.1355242</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/38523646">38523646</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10957596">PMC10957596</a></div></div>
</div>
<div id="ref-1AegtXQtf" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">61. </div><div class="csl-right-inline"><strong>Perspectives on single-nucleus RNA sequencing in different cell types and tissues</strong> <div class="csl-block">Nayoung Kim, Huiram Kang, Areum Jo, Seung-Ah Yoo, Hae-Ock Lee</div> <em>Journal of Pathology and Translational Medicine</em> (2023-01-15) <a href="https://doi.org/g9bnw9">https://doi.org/g9bnw9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.4132/jptm.2022.12.19">10.4132/jptm.2022.12.19</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36623812">36623812</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9846005">PMC9846005</a></div></div>
</div>
<div id="ref-cnXjUoU1" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">62. </div><div class="csl-right-inline"><strong>Dissociation protocols used for sarcoma tissues bias the transcriptome observed in single-cell and single-nucleus RNA sequencing</strong> <div class="csl-block">Danh D Truong, Salah-Eddine Lamhamedi-Cherradi, Robert W Porter, Sandhya Krishnan, Jyothishmathi Swaminathan, Amber Gibson, Alexander J Lazar, JAndrew Livingston, Vidya Gopalakrishnan, Nancy Gordon, … Joseph A Ludwig</div> <em>BMC Cancer</em> (2023-05-31) <a href="https://doi.org/g9dn4b">https://doi.org/g9dn4b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12885-023-10977-1">10.1186/s12885-023-10977-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37254069">37254069</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10230784">PMC10230784</a></div></div>
</div>
<div id="ref-W0TbGvjF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">63. </div><div class="csl-right-inline"><strong>Cellxgene Data Portal</strong> <div class="csl-block">Cellxgene Data Portal</div> <a href="https://cellxgene.cziscience.com/">https://cellxgene.cziscience.com/</a></div>
</div>
<div id="ref-QgMCMjVA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">64. </div><div class="csl-right-inline"><strong>CZI Single cell curation schema 3.0.0</strong> <div class="csl-block">CZI Cellxgene</div> <a href="https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.0.0/schema.md">https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/3.0.0/schema.md</a></div>
</div>
<div id="ref-91inc6cK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">65. </div><div class="csl-right-inline"><strong>Alignment and mapping methodology influence transcript abundance estimation</strong> <div class="csl-block">Avi Srivastava, Laraib Malik, Hirak Sarkar, Mohsen Zakeri, Fatemeh Almodaresi, Charlotte Soneson, Michael I Love, Carl Kingsford, Rob Patro</div> <em>Genome Biology</em> (2020-09-07) <a href="https://doi.org/gg98sd">https://doi.org/gg98sd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-020-02151-8">10.1186/s13059-020-02151-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32894187">32894187</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7487471">PMC7487471</a></div></div>
</div>
<div id="ref-tTu8Ds9Z" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">66. </div><div class="csl-right-inline"><strong>STAR: ultrafast universal RNA-seq aligner</strong> <div class="csl-block">Alexander Dobin, Carrie A Davis, Felix Schlesinger, Jorg Drenkow, Chris Zaleski, Sonali Jha, Philippe Batut, Mark Chaisson, Thomas R Gingeras</div> <em>Bioinformatics</em> (2012-10-25) <a href="https://doi.org/f4h523">https://doi.org/f4h523</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bts635">10.1093/bioinformatics/bts635</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23104886">23104886</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3530905">PMC3530905</a></div></div>
</div>
<div id="ref-rxoRG1BR" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">67. </div><div class="csl-right-inline"><strong>STARsolo: accurate, fast and versatile mapping/quantification of single-cell and single-nucleus RNA-seq data</strong> <div class="csl-block">Benjamin Kaminow, Dinar Yunusov, Alexander Dobin</div> <em>Cold Spring Harbor Laboratory</em> (2021-05-05) <a href="https://doi.org/gqj7ft">https://doi.org/gqj7ft</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.05.05.442755">10.1101/2021.05.05.442755</a></div></div>
</div>
<div id="ref-eyOvxHVn" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">68. </div><div class="csl-right-inline"><strong>Twelve years of SAMtools and BCFtools</strong> <div class="csl-block">Petr Danecek, James K Bonfield, Jennifer Liddle, John Marshall, Valeriu Ohan, Martin O Pollard, Andrew Whitwham, Thomas Keane, Shane A McCarthy, Robert M Davies, Heng Li</div> <em>GigaScience</em> (2021-01-29) <a href="https://doi.org/gjxzc9">https://doi.org/gjxzc9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/gigascience/giab008">10.1093/gigascience/giab008</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33590861">33590861</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7931819">PMC7931819</a></div></div>
</div>
<div id="ref-wd1cI6xh" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">69. </div><div class="csl-right-inline"><strong>Cellsnp-lite: an efficient tool for genotyping single cells</strong> <div class="csl-block">Xianjie Huang, Yuanhua Huang</div> <em>Bioinformatics</em> (2021-05-08) <a href="https://doi.org/gqcggs">https://doi.org/gqcggs</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btab358">10.1093/bioinformatics/btab358</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33963851">33963851</a></div></div>
</div>
<div id="ref-wBFzn1br" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">70. </div><div class="csl-right-inline"><strong>Ontology Lookup Service (OLS)</strong> <a href="https://www.ebi.ac.uk/ols4/ontologies/cl">https://www.ebi.ac.uk/ols4/ontologies/cl</a></div>
</div>
<div id="ref-MclvBVPD" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">71. </div><div class="csl-right-inline"><strong>CellMarker 2.0: an updated database of manually curated cell markers in human/mouse and web tools based on scRNA-seq data</strong> <div class="csl-block">Congxue Hu, Tengyue Li, Yingqi Xu, Xinxin Zhang, Feng Li, Jing Bai, Jing Chen, Wenqi Jiang, Kaiyue Yang, Qi Ou, … Yunpeng Zhang</div> <em>Nucleic Acids Research</em> (2022-10-27) <a href="https://doi.org/gq473c">https://doi.org/gq473c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/nar/gkac947">10.1093/nar/gkac947</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36300619">36300619</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9825416">PMC9825416</a></div></div>
</div>
<div id="ref-11dH4UgRF" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">72. </div><div class="csl-right-inline"><strong>zellkonverter</strong> <div class="csl-block">Luke Zappia, Aaron Lun</div> <em>Bioconductor</em> (2020) <a href="https://doi.org/gtnrz5">https://doi.org/gtnrz5</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18129/b9.bioc.zellkonverter">10.18129/b9.bioc.zellkonverter</a></div></div>
</div>
<div id="ref-w9AOzBMw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">73. </div><div class="csl-right-inline"><strong>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</strong> <div class="csl-block">Michael I Love, Wolfgang Huber, Simon Anders</div> <em>Genome Biology</em> (2014-12-05) <a href="https://doi.org/gd3zvn">https://doi.org/gd3zvn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-014-0550-8">10.1186/s13059-014-0550-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25516281">25516281</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4302049">PMC4302049</a></div></div>
</div>
<div id="ref-AUoV0xkq" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">74. </div><div class="csl-right-inline"><strong>Fitting Linear Mixed-Effects Models Using&lt;b&gt;lme4&lt;/b&gt;</strong> <div class="csl-block">Douglas Bates, Martin Mächler, Ben Bolker, Steve Walker</div> <em>Journal of Statistical Software</em> (2015) <a href="https://doi.org/gcrnkw">https://doi.org/gcrnkw</a> <div class="csl-block">DOI: <a href="https://doi.org/10.18637/jss.v067.i01">10.18637/jss.v067.i01</a></div></div>
</div>
<div id="ref-UeFCwwR7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">75. </div><div class="csl-right-inline"><strong>Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing</strong> <div class="csl-block">Yoav Benjamini, Yosef Hochberg</div> <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em> (1995-01-01) <a href="https://doi.org/gfpkdx">https://doi.org/gfpkdx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/j.2517-6161.1995.tb02031.x">10.1111/j.2517-6161.1995.tb02031.x</a></div></div>
</div>
<div id="ref-115InIIhW" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">76. </div><div class="csl-right-inline"><strong>cellxgene: a performant, scalable exploration platform for high dimensional sparse matrices</strong> <div class="csl-block">Colin Megill, Bruce Martin, Charlotte Weaver, Sidney Bell, Lia Prins, Seve Badajoz, Brian McCandless, Angela Oliveira Pisco, Marcus Kinsella, Fiona Griffin, … Ambrose Carr</div> <em>Cold Spring Harbor Laboratory</em> (2021-04-06) <a href="https://doi.org/gst8vt">https://doi.org/gst8vt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2021.04.05.438318">10.1101/2021.04.05.438318</a></div></div>
</div>
<div id="ref-WrWip7GV" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">77. </div><div class="csl-right-inline"><strong>CZ CELL×GENE Discover: A single-cell data platform for scalable exploration, analysis and modeling of aggregated data</strong><div class="csl-block">, Shibla Abdulla, Brian Aevermann, Pedro Assis, Seve Badajoz, Sidney M Bell, Emanuele Bezzi, Batuhan Cakir, Jim Chaffer, Signe Chambers, … Ambrose Carr</div> <em>Cold Spring Harbor Laboratory</em> (2023-11-02) <a href="https://doi.org/gtk3dd">https://doi.org/gtk3dd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2023.10.30.563174">10.1101/2023.10.30.563174</a></div></div>
</div>
<div id="ref-7NobF7Ei" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">78. </div><div class="csl-right-inline"><strong>Cellxgene Data Portal</strong> <div class="csl-block">Cellxgene Data Portal</div> <a href="https://cellxgene.cziscience.com/">https://cellxgene.cziscience.com/</a></div>
</div>
<div id="ref-xCP3BjTw" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">79. </div><div class="csl-right-inline"><strong>UCSC Cell Browser: visualize your single-cell data</strong> <div class="csl-block">Matthew L Speir, Aparna Bhaduri, Nikolay S Markov, Pablo Moreno, Tomasz J Nowakowski, Irene Papatheodorou, Alex A Pollen, Brian J Raney, Lucas Seninge, WJames Kent, Maximilian Haeussler</div> <em>Bioinformatics</em> (2021-07-09) <a href="https://doi.org/gtk3db">https://doi.org/gtk3db</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/btab503">10.1093/bioinformatics/btab503</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34244710">34244710</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8652023">PMC8652023</a></div></div>
</div>
<div id="ref-VKaQa2Z3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">80. </div><div class="csl-right-inline"><strong>UCSC Cell Browser</strong> <a href="https://cells.ucsc.edu/">https://cells.ucsc.edu/</a></div>
</div>
<div id="ref-1DswCJYT0" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">81. </div><div class="csl-right-inline"><strong>Powering single-cell analyses in the browser with WebAssembly</strong> <div class="csl-block">Aaron Lun, Jayaram Kancherla</div> <em>Cold Spring Harbor Laboratory</em> (2022-03-04) <a href="https://doi.org/gtk3dc">https://doi.org/gtk3dc</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1101/2022.03.02.482701">10.1101/2022.03.02.482701</a></div></div>
</div>
<div id="ref-1FzXrvESX" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">82. </div><div class="csl-right-inline"><strong>kana</strong> <a href="https://www.kanaverse.org/kana/">https://www.kanaverse.org/kana/</a></div>
</div>
<div id="ref-Whp3r83M" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">83. </div><div class="csl-right-inline"><strong>A comprehensive benchmarking with practical guidelines for cellular deconvolution of spatial transcriptomics</strong> <div class="csl-block">Haoyang Li, Juexiao Zhou, Zhongxiao Li, Siyuan Chen, Xingyu Liao, Bin Zhang, Ruochi Zhang, Yu Wang, Shiwei Sun, Xin Gao</div> <em>Nature Communications</em> (2023-03-21) <a href="https://doi.org/gtk3c9">https://doi.org/gtk3c9</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-023-37168-7">10.1038/s41467-023-37168-7</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36941264">36941264</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10027878">PMC10027878</a></div></div>
</div>
<div id="ref-mgOojr60" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">84. </div><div class="csl-right-inline"><strong>Computational deconvolution of transcriptomics data from mixed cell populations</strong> <div class="csl-block">Francisco Avila Cobos, Jo Vandesompele, Pieter Mestdagh, Katleen De Preter</div> <em>Bioinformatics</em> (2018-01-16) <a href="https://doi.org/gctpvd">https://doi.org/gctpvd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1093/bioinformatics/bty019">10.1093/bioinformatics/bty019</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29351586">29351586</a></div></div>
</div>
<div id="ref-GpuiAcfU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">85. </div><div class="csl-right-inline"><strong>Effective methods for bulk RNA-seq deconvolution using scnRNA-seq transcriptomes</strong> <div class="csl-block">Francisco Avila Cobos, Mohammad Javad Najaf Panah, Jessica Epps, Xiaochen Long, Tsz-Kwong Man, Hua-Sheng Chiu, Elad Chomsky, Evgeny Kiner, Michael J Krueger, Diego di Bernardo, … Pavel Sumazin</div> <em>Genome Biology</em> (2023-08-01) <a href="https://doi.org/gsmvqq">https://doi.org/gsmvqq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s13059-023-03016-6">10.1186/s13059-023-03016-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37528411">37528411</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10394903">PMC10394903</a></div></div>
</div>
<div id="ref-U5WisTG5" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">86. </div><div class="csl-right-inline"><strong>The children's brain tumor network (CBTN) - Accelerating research in pediatric central nervous system tumors through collaboration and open science</strong> <div class="csl-block">Jena V Lilly, Jo Lynne Rokita, Jennifer L Mason, Tatiana Patton, Stephanie Stefankiewiz, David Higgins, Gerri Trooskin, Carina A Larouci, Kamnaa Arya, Elizabeth Appert, … Angela J Waanders</div> <em>Neoplasia</em> (2023-01) <a href="https://doi.org/grkvcf">https://doi.org/grkvcf</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.neo.2022.100846">10.1016/j.neo.2022.100846</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36335802">36335802</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9641002">PMC9641002</a></div></div>
</div>
<div id="ref-5VXMHJ7N" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">87. </div><div class="csl-right-inline"><strong>OpenPBTA: The Open Pediatric Brain Tumor Atlas</strong> <div class="csl-block">Joshua A Shapiro, Krutika S Gaonkar, Stephanie J Spielman, Candace L Savonen, Chante J Bethell, Run Jin, Komal S Rathi, Yuankun Zhu, Laura E Egolf, Bailey K Farrow, … Jaclyn N Taroni</div> <em>Cell Genomics</em> (2023-07) <a href="https://doi.org/gr92p6">https://doi.org/gr92p6</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.xgen.2023.100340">10.1016/j.xgen.2023.100340</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/37492101">37492101</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10363844">PMC10363844</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
