## Introduction

Since the introduction of single-cell RNA-seq technology, the number of studies that utilize single-cell RNA-seq has grown rapidly[@doi:10.1038/nprot.2017.149].
Unlike its predecessor, bulk RNA-seq, which averages the profiles of all cells within a sample, single-cell technology quantifies gene expression in individual cells.
Tumors are known to be transcriptionally heterogeneous, so many studies have highlighted the importance of using single-cell RNA-seq in studying tumor samples [@doi:10.1101/gr.190595.115].
Utilizing single-cell RNA-seq for tumor samples can allow for analysis and identification of individual cell populations that may play important roles in tumor growth, resistence, and metastasis [@doi:10.1126/science.1254257].
Additionally, single-cell RNA-seq of tumor samples provides insight into how tumor cells may be interacting with normal cells in the tumor microenvironment[@10.1038/s41588-022-01141-9].

With the growing number of single-cell RNA-seq datasets, efforts have emerged to create central, harmonized sources for datasets.
By harmonizing data across multiple studies and diseases, researchers can better perform joint analysis, taking advantage of more samples to complete their analysis and illuminate previously unknown similarities across different samples and disease types.
These efforts include the Human Cell Atlas (HCA), which aims to use single-cell genomics to provide a comprehensive map of all cell types in the human body [@doi:10.7554/eLife.27041].
This atlas contains uniformly processed single-cell RNA-seq data obtained from normal tissue with few samples derived from diseased tissue.
The Human Tumor Atlas Network (HTAN) also hosts a collection of genomic data collected from tumors across multiple cancer types, including single-cell RNA-seq [@doi:10.1016/j.cell.2020.03.053].

The HCA and HTAN are two of many efforts that have been made to provide access to single-cell RNA-seq data across a variety of tissue types.
These mechanisms have focused on making large quantities of harmonized data from normal tissue or adult tumor samples available to the public, but there are considerably less efforts to harmonize and publicize data from pediatric tumors.
Pediatric cancer is much less common than adult cancer, so the number of available samples from pediatric tumors is smaller compared to the number of adult tumors.
Additionally, not every institution has access to data from pediatric tumors.
These two reasons make it imperative to provide harmonized data from pediatric tumors to all pediatric cancer researchers.
To address this unmet need, Alex's Lemonade Stand Foundation and the Childhood Cancer Data Lab developed and maintain the Single-cell Pediatric Cancer Atlas (ScPCA) Portal (https://scpca.alexslemonade.org/), an open-source data resource for single-cell and single-nuclei RNA sequencing data of pediatric tumors.

The ScPCA Portal holds uniformly processed summarized gene expression from 10X Genomics' droplet-based single-cell and single-nuclei RNA-seq for over 500 samples from a diverse set of over 50 types of cancers.
The original data available on the portal came from 10 projects funded by Alex's Lemonade Stand Foundation, but we have since expanded the Portal to include data contributed by members of the pediatric cancer research community.
In addition to gene expression data from single-cell and single-nuclei RNA-seq, the Portal holds data obtained from bulk RNA-seq, spatial transcriptomics, and feature barcoding methods, such as ADT/CITE-seq and cell hashing.
All data provided on the portal is available in formats ready for downstream analysis, such as SingleCellExperiment or AnnData object with objects containing normalized gene expression counts, dimensionality reduction results and cell type annotations.

To ensure that all current and future data on the Portal is uniformly processed, the Childhood Cancer Data Lab created scpca-nf, a Nextflow-based open-source pipeline (https://github.com/AlexsLemonade/scpca-nf).
The scpca-nf workflow uses alevin-fry for fast and efficient processing to quantify gene expression for all samples on the Portal, including single-cell RNA-seq data and any associated ADT/CITE-seq or cell hash data, spatial transcriptomics data, and bulk RNA-seq data.
Ensuring all data on the Portal is uniformly processed allows users to perform analysis across multiple samples and projects without having to do any re-processing.
The scpca-nf pipeline also serves as a resource for the community to allow others to process their own samples for comparison to samples available on the Portal.
In addition to uniformly processed data across multiple cancer types, we provide comprehensive documentation about data processing and the contents of files on the portal, including a guide to getting started working with an ScPCA dataset (https://scpca.readthedocs.io/).
Here, we present the Single-cell Pediatric Cancer Atlas Portal, which serves as a resource for all pediatric cancer researchers by providing uniformly processed data ready for immediate use to help researchers answer their important biological questions, without the need for time consuming data re-processing and data wrangling.
