# Results

## The Single-cell Pediatric Cancer Atlas Portal

In March of 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata from pediatric tumor samples available for download.
Today, the Portal contains data from 500 samples and over 50 tumor types.
Data available on the Portal was obtained using two different mechanisms.
Raw data was accepted from ALSF-funded investigators and processed using our open-source pipeline, `scpca-nf`, or investigators processed their raw data using `scpca-nf`, producing summarized gene expression data submitted for inclusion on the Portal.

All samples on the Portal include a core set of metadata obtained from investigators, including age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
Some investigators submitted additional metadata, such as treatment and tumor stage also found on the Portal.
All submitted metadata was standardized as much as possible to maintain consistency across projects before adding to the Portal.
In addition to providing a human-readable value for the submitted metadata, we also provide an ontology term ID, if applicable.
The total number of samples for each diagnosis is shown in Figure 1A, along with a breakdown of the proportion of samples from each disease stage within a diagnosis group.
Figure 1A summarizes all samples from patient tumors or patient-derived xenografts currently available on the Portal.
Along with the patient tumors, the Portal contains a handful of samples from human tumor cell lines.

Each available sample has at minimum summarized gene expression data from either single-cell or single-nuclei RNA-seq.
However, some samples include additional data, such as quantified data from tagging cells with Antibody-derived tags (ADT), like CITE-seq[@doi:10.1038/nmeth.4380], or multiplexing samples with hashtag oligonucleotides (HTO)[@doi:10.1186/s13059-018-1603-1].
In some cases, multiple libraries from the same sample were collected to conduct either bulk RNA-seq or spatial transcriptomics.
Downloading a sample on the Portal will include sequencing data from all associated libraries, including data from any additional modalities mentioned here.
A summary of the number of samples with each additional modality is shown in Figure 1B, and a detailed summary of the total samples with each sequencing method broken-down by project, is available in Supplemental Table 1.

Samples on the Portal are organized by project, where each project is a collection of similar samples from a single investigator.
Users can download all samples in a project or navigate to projects of interest and choose individual samples to download.
To identify projects of interest, users can filter based on diagnosis, included modalities (e.g., CITE-seq, bulk RNA-seq), 10X Genomics version (e.g., 10Xv2, 10Xv3), and whether or not a project includes samples derived from patient-derived xenografts or cell lines.
The project card displays an abstract, the total number of samples included, a list of diagnoses for all samples included in the Project, and links to any external information associated with the project, such as publications and links to external data, such as SRA or GEO (Figure 1C).
The project card will also indicate the type(s) of sequencing performed, including the 10X Genomics kit version, the suspension type (cell or nucleus), and if additional sequencing is present, like bulk RNA-seq or multiplexing.

## Uniform processing of data available on the ScPCA Portal

All data available on the portal was uniformly processed using `scpca-nf`, an open-source and efficient Nextflow[@url:https://www.nextflow.io/docs/latest/index.html] workflow for quantifying single-cell and single-nuclei RNA-seq data (github.com/alexslemonade/scpca-nf). 
<!-- TO DO: Does description of Nextflow benefits go here or a separate paragraph at the end?--> 
Using Nextflow as the backbone for the `scpca-nf` workflow ensures both reproducibility and portability.
All dependencies for the workflow are handled automatically and can be run in nearly any computing environment. 
Setup requires organizing input files and configuring Nextflow to run in your computing environment, which involves updating a single configuration file. 
Nextflow will also handle parallelizing sample processing as allowed by your environment, minimizing run time. 

`scpca-nf` uses `salmon alevin` and `alevin-fry` to quantify gene expression data and outputs both raw and normalized counts stored as `SingleCellExperiment` and `AnnData` objects[@doi:10.1038/s41592-022-01408-3]. 
Reads are aligned using the selective alignment option of `salmon alevin` to an index with transcripts corresponding to spliced cDNA and intronic regions, denoted by `alevin-fry` as a `splici` index. 
When building `scpca-nf`, we sought a fast and memory-efficient tool with comparable results to other popular tools, like Cell Ranger[@url:https://www.10xgenomics.com/support/software/cell-ranger/latest].
We ran `alevin-fry` and Cell Ranger on single-cell and single-nuclei samples and compared run time, memory usage, and the quantified gene expression data. 
A decrease in both run time and memory usage in `alevin-fry` compared to Cell Ranger was observed in all samples (Supplemental Figure 1A).
There was no observable difference in the mean gene expression for all genes (Supplemental Figure 1B), total UMIs per cell (Supplemental Figure 1C),  or total genes detected per cell (Supplemental Figure 1D), between `alevin-fry` and  Cell Ranger.
Since both `alevin-fry` and Cell Ranger produce comparable results, we chose to utilize `alevin-fry` in the `scpca-nf` workflow. 
With `alevin-fry`, we can process samples at a fraction of the time and cost.

In addition to quantification of gene expression, `scpca-nf` also performs filtering of empty droplets, removal of low-quality cells, normalization, dimensionality reduction, and cell type annotation (Figure 2A).
The output from `alevin-fry` includes a gene-by-cell count matrix for all barcodes identified, even those that may not contain true cells. 
This unfiltered counts matrix is stored in a `SingleCellExperiment` object and output from the workflow to a `.rds` file with the suffix `_unfiltered.rds`.
The unfiltered gene by cell counts matrices are then filtered using `DropletUtils::emptyDropsCellRanger()` to remove any barcodes that are not likely to contain cells[@doi:10.1186/s13059-019-1662-y]. 
All cells that pass this filtering are saved to a filtered `SingleCellExperiment` object and `.rds` file with the suffix `_filtered.rds`.
This filtered counts matrix is used as input to the post-processing part of the workflow, which includes the removal of low-quality cells with `miQC` [@doi: 10.1371/journal.pcbi.1009290], normalization, and dimensionality reduction. 
The final step of the post-processing performed in `scpca-nf` is the classification of cell types using two automated methods, `SingleR`[@doi:10.1038/s41590-018-0276-y] and `CellAssign`[@doi:10.1038/s41592-019-0529-1]. 
The results from this analysis are stored in a processed `SingleCellExperiment` object saved to a `.rds` file with the suffix `_processed.rds`.
Finally, all `SingleCellExperiment` objects saved as `.rds` files are converted to `AnnData` objects and saved as `.hdf5` files to allow for downstream processing in either R or Python.
On the Portal, users can choose to download data as either `SingleCellExperiment` or `AnnData` objects, and all downloads will contain all three objects output from `scpca-nf`, the unfiltered, filtered, and processed objects. 
Including all three objects allows users to perform their own filtering and normalization or to skip those steps and use the already processed object, which contains normalized data ready for downstream analysis.

Along with outputting the uniformly processed data files, `scpca-nf` also creates and outputs a quality control (QC) report for each library.
This report includes a summary of processing information and library statistics, e.g., the total number of mapped reads, the total number of cells, and relevant versions of tools used within the workflow like `salmon` and `alevin-fry`.
Each report also includes a collection of plots that can be used to evaluate the quality of the library (Figure 2B).
The first plot in the report, the knee plot, shows all droplets prior to the removal of empty droplets.
All droplets are sorted based on the total number of UMIs and those retained after filtering empty droplets are indicated in the plot.
For each cell that remains after filtering empty droplets, the number of total UMIs, genes detected, and mitochondrial reads are calculated and summarized in a scatter plot. 
To remove low-quality cells from the count's matrices, `scpca-nf` applies `miQC`[doi:10.1371/journal.pcbi.1009290], a data-driven approach to identifying low-quality cells.
The `miQC` model and a plot showing which cells are kept and removed after filtering with `miQC` are shown in the QC report.
Finally, the remaining cells are normalized and undergo dimensionality reduction using both principal component analysis (PCA) and UMAP.
The QC report includes a single UMAP where cells are colored by the total number of genes detected and a faceted UMAP where cells are colored by the expression of a top highly variable gene.
All downloads on the Portal will include both the gene expression files (either `SingleCellExperiment` or `AnnData` objects) and the QC report for each library. 

<!--TO DO: Other option for a more extended Nextflow paragraph -->
Using Nextflow as the backbone for the `scpca-nf` workflow ensures both reproducibility and portability. 
Nextflow handles all dependencies automatically, and setup requires organizing input files and configuring Nextflow to run in your environment, which involves updating a single configuration file. 
Nextflow is compatible with a variety of computing environments, including high-performance computing and cloud-based computing, allowing users to run the workflow in their preferred environment. 
Each process in the workflow is run in a docker container, so users are not required to install any packages needed for individual analysis steps. 
The only requirement is to install Nextflow and either Docker or Singularity. 
Nextflow also handles parallelizing processing based on your environment and will configure processing so that run time is minimal.
The combination of being able to execute a Nextflow workflow in any environment and run individual processes in Docker containers makes this workflow easily portable for external use. 

## Making samples with additional modalities available on the Portal

Each sample on the Portal will have summarized gene expression data from either single-cell or single-nuclei RNA-seq. 
For some samples, submitters included data from additional sequencing modalities, including corresponding ADT or CITE-seq data [@doi:10.1038/nmeth.4380], multiplexing using cell hashing [@doi:10.1186/s13059-018-1603-1], spatial transcriptomics, or bulk RNA-seq. 
To make all received data available, we included additional modules in `scpca-nf` to uniformly process samples with these additional sequencing modalities.
For a full summary of the libraries and samples available with additional modalities, see Supplemental Table 1.

For all libraries with associated ADT or CITE-seq data, we provide both the summarized RNA and ADT gene expression data for download on the portal.
To process these libraries, both FASTQ from single-cell or single-nuclei RNA-seq and CITE-seq were provided as input into `scpca-nf` and quantified using `salmon alevin` and `alevin-fry` (Supplemental Figure 2A).
Along with the FASTQ files,  we required a tsv file from each submitter that contained the labels for each ADT used and the associated barcode. 
This file is also input to `scpca-nf` and used to build the index used to quantify ADT expression and create the cell by ADT counts matrix.
Unlike with RNA counts, we do not perform any filtering of cells due to low-quality ADT expression. 
However, we do include the results from running `DropletUtils::cleanTagCounts()` on the ADT counts matrix in both the filtered and processed objects output by `scpca-nf`.
Similar to RNA counts, we normalize ADT data and provide the normalized counts matrix in the processed object. 
Although we do normalize the ADT counts, we do not provide any dimensionality reduction of ADT data; only the RNA counts data is used as input for dimensionality reduction.
For all three `SingleCellExperiment` objects (unfiltered, filtered, and processed), the summarized ADT expression data can be found in the `altExp` slot. 
With `AnnData` objects, the ADT data is provided as a separate `_adt.hdf5` file for each of the three objects.

If a library contains associated ADT data, an additional section will be included in the provided QC report. 
This section includes a summary of statistics, such as how many cells express each ADT.
We also include a collection of additional plots specific to ADT data (Supplemental Figure 2B).  
As mentioned above, we include the results from `DropletUtils::cleanTagCounts()`, but do not filter any ADTs or cells from the object. 
Instead, we include a column in the `colData` of the processed `SingleCellExperiment` object (`.obs` in the `AnnData` object) that indicates if a cell is recommended to be removed due to low expression of ADTs.
In the QC report, we summarize filtering low-quality cells based on both RNA and ADT counts.
The first quadrant shows which cells would be kept if filtering on both RNA and ADT. 
Each of the other facets highlights which cells would be removed if filtering is only done using RNA counts, ADT counts, or both.
The top 4 ADTs with the most variable expression are also identified and visualized using both UMAPs and density plots. 
The UMAPs shown were calculated using the RNA data but cells are colored by ADT expression, while the ridge plots show the normalized ADT expression across all cells. 

Similar to ADT data, if any libraries contain samples that have been multiplexed and have an associated cell hashing library, both the RNA and HTO FASTQ are provided as input to the workflow and quantified with `salmon alevin` and `alevin-fry` (Supplemental Figure 2C). 
`scpca-nf` also requires a tsv file with one row per sample included in the library, which tells the workflow which HTO was used for which sample when multiplexing the library. 
The HTO counts data will be included as an `altExp` in each of the `SingleCellExperiment` objects.
Although we quantify the HTO data and include the cell by HTO counts matrix in all objects, we do not demultiplex the samples so that there is one sample per library. 
Instead, we apply multiple demultiplexing methods, including demultiplexing with `DropletUtils::hashedDrops()`, demultiplexing with `Seurat::HTODemux()`, and genetic demultiplexing when possible.
The genetic demultiplexing used in `scpca-nf` uses the method described in Weber et al [@doi:[10.1093/gigascience/giab062](https://doi.org/10.1093/gigascience/giab062)], which takes bulk RNA-seq data from the same sample as a reference for the expected genotypes found in each sample.
<!--Do we need to list all individual tools used here or are we okay to mention them in the methods section, e.g., STAR, bcftools, STARsolo, cellsnp--> 
Variants among the samples within each pool are identified from both the mapped bulk RNA-seq data and pooled single-cell or single-nuclei RNA-seq. 
After variants from both bulk and single-cell or single-nuclei RNA-seq are genotyped, `vireo` is used to match genotypes and identify the most likely sample of origin [@doi:10.1186/s13059-019-1865-2].
The results from these three demultiplexing methods are included in the filtered and processed objects on the Portal. 
If a sample does not have associated bulk RNA-seq data, then the objects will not contain genetic demultiplexing results. 

If a library has associated HTO data, an additional section is added to the QC report included on the Portal and output by `scpca-nf`.
This section includes a summary of library statistics, such as how many cells express each HTO.
We do not include any additional plots, but we do show a table summarizing how many cells belong to each sample included in the multiplexed library using each of the demultiplexing methods mentioned.

For some samples, multiple libraries were collected, with the additional libraries being used for  bulk RNA-seq and/or spatial transcriptomics. 
Both of these additional sequencing methods are supported by `scpca-nf`, and the output is available for download on the Portal. 
To process bulk RNA-seq data, reads are first trimmed using `fastp` and then aligned using `salmon` (Supplemental Figure 3A). 
The bulk module of `scpca-nf` outputs a single tsv file with the sample by gene counts matrix for all samples in a given ScPCA project.
This sample by gene matrix is included only with project downloads on the Portal. 
As there is not yet support for spatial transcriptomics with `alevin-fry`, `scpca-nf` uses Space Ranger to quantify all spatial transcriptomics data [@url:https://www.10xgenomics.com/support/software/space-ranger/latest] (Supplemental Figure 3B).
The required input includes the RNA FASTQ files and slide image. 
The output includes the spot-by-gene matrix along with a summary report produced by Space Ranger.
If a sample on the Portal has associated spatial transcriptomics data, users will be able to download the summarized gene expression data from single-cell or single-nuclei RNA-seq separately from the spatial output. 


