# Results

## The Single-cell Pediatric Cancer Atlas Portal

1. History of the Portal
  - In 2019, Alex's Lemonade Stand Foundation funded 10 investigators working on single-cell profiling to create a publicly available atlas of single-cell pediatric cancer data.
  ** are there more specific information about how things were funded that I should include?**
  - These 10 investigators produced single-cell and single-nuclei RNA-seq data across a variety of tumors
  - The raw data and associated metadata was sent to the Childhood Cancer Data Lab for uniform processing
  - We created an open-source efficient pipeline to process all samples present on the Portal
  - In 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata available for download.
  - In 2023, we began to expand the Portal to other investigators, putting out a call for contributions.
  - For these projects, investigators used our open-source pipeline to produce summarized gene expression data for inclusion on the portal.

2. Overview of data available on the Portal
  - The Portal currently holds X amount of samples from X amount of tumor types.
  - For all samples we collect a core set of metadata including, age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
  - Fig. 1A shows how many samples we have from each type of tumor. For each diagnosis, we also indicate what proportion of the samples come from each disease stage (e.g., initial diagnosis, recurrence, post-mortem).
  - In addition to single-cell and single-nuclei RNA-seq, many samples have associated bulk RNA-seq, ADT data (CITE-seq), cell hashing, or spatial transcriptomics.
  - Fig. 2B summarizes the total number of samples that are single-cell vs. single-nuclei. Additionally, we show how many of the samples on the portal also have either bulk, CITE, cell hashing, or spatial data.
  - Supplemental Table 1 shows a breakdown of how many of each modality is found in each project.
  ** By submission we will provably have PDX/ cell line information on the portal so we might want to mention that here and add to figures?**

3. Obtaining additional project information
  - On the Portal, samples are organized by project. Each project is a collection of similar samples from a single investigator.
  - To select projects of interest, users can filter based on diagnosis, modality included, single-cell or single-nuclei and 10X version. Additionally, users will be able to filter based on if the project includes cell line samples or xenografts.
  - A summary of each project, including a list of samples found in each project, is displayed on the Portal.
  - Fig.1C shows an example of this summary which include an abstract, links to any external information about the projects such as any associated publication information, and links to external places where data may be stored such as SRA or GEO.
  - If a project includes bulk, CITE, spatial, or multiplexing, this will also be indicated on the project card.


## Uniform processing of data available on the ScPCA Portal

1. Processing data with scpca-nf and alevin-fry
  - All data available on the portal was uniformly processed using scpca-nf, an open-source and efficient Nextflow workflow for quantifying single-cell and single-nuclei RNA-seq data.
  - The workflow uses `salmon alevin` and `alevin-fry` to quantify gene expression data and outputs both raw and normalized counts stored as `SingleCellExperiment` and `AnnData` objects.
  - In building the workflow we sought to look for a tool that was fast and memory efficient with comparable results to other popular tools, like `Cell Ranger`.
  - Reads are aligned using the selective alignment option of `salmon alevin` to an index with transcripts corresponding to spliced cDNA and intronic regions, denoted by `alevin-fry` as a `splici` index.
  - We compared quantification of 3 single-cell and 3 single-nuclei samples with `alevin-fry` and `Cell Ranger` and observed a decrease in both run time and memory usage in `alevin-fry` compared to `Cell Ranger` (FigS1A).
  - When comparing the total UMIs per cell, total genes detected per cell, and mean gene expression, there was no observable difference between `alevin-fry` and `Cell Ranger` (FigS1B-D).
  - By utilizing `alevin-fry` in the `scpca-nf` workflow we can process multiple samples at a fraction of the time and cost.

2. Post-processing of quantified gene expression data (Fig 2A)
  - In addition to quantification of gene expression, `scpca-nf` also performs filtering, normalization, dimensionality reduction, and cell type annotation.
  - The output from  `alevin-fry` includes a gene by cell count matrix for all barcodes identified, even those that may not contain true cells. This matrix is stored in a `SingleCellExperiment` and output from the workflow as an `_unfiltered.rds` file.
  - The unfiltered gene by cell counts matrices are then filtered using `DropletUtils::emptyDropsCellRanger()` to remove any barcodes that are not likely to contain cells. All cells that pass this filtering are saved to a filtered `SingleCellExperiment` object and `_filtered.rds` file.
  - This filtered object is used as input to the post-processing part of the workflow. This includes removal of low-quality cells using `miQC`, normalization, and dimensionality reduction. The final step of the post-processing performed in `scpca-nf` is classification of cell types using automated methods, `SingleR` and `CellAssign`. The results from this analysis are stored in a processed object saved to a `processed.rds`.
  - Finally, all `SingleCellExperiment` objects saved as `.rds` files are converted to `AnnData` objects and saved as `.hdf5` files to allow for downstream processing in either R or Python.
  - On the Portal, users can choose to download data as either `SingleCellExperiment` or `AnnData` objects and all downloads will contain all three objects output from `scpca-nf`, the unfiltered, filtered, and processed objects (do we include the download illustrations in the figure to display this?)

3. QC report (Fig 2B)
  - Along with outputting the uniformly processed data files, `scpca-nf` also includes a step to create a quality control report for each library.
  - This report includes a summary of processing information and library statistics, e.g., the total number of mapped reads, total number of cells, and relevant versions of tools used within the workflow like `salmon` and `alevin-fry`.
  - Each report also includes summarized plots showing the quality of each library.
  - The knee plot shown in the report ranks the total number of UMIs in each droplet and indicates cells that remained after filtering out empty droplets.
  - For each cell that passes filtering out empty droplets, the number of total UMIs, genes detected, and mitochondrial reads is calculated. These cell metrics are summarized in a single plot.
  - To remove low-quality cells from the counts matrices, `scpca-nf` applies `miQC`, a data driven approach to filtering cells. The `miQC` model and a plot showing which cells are kept and removed when filtering with `miQC` are shown in the QC report.
  - Finally, remaining cells are normalized and undergo dimensionality reduction. The QC report includes a single UMAP where cells are colored by the total number of genes detected and a faceted UMAP where cells are colored by the expression of a top highly variable gene.

4. Benefits of scpca-nf/ Nextflow allows for reproducibility and portability (Does this fit here or should it be earlier before describing the workflow?)
  - Using Nextflow as the backbone for the `scpca-nf` workflow ensures reproducibility and portability for users on other systems.
  - Nextflow handles all dependencies automatically and set up generally requires only organizing input files and configuring Nextflow for your environment.
  - Each process in the workflow is run in a docker container, so users only need to install Nextflow and docker to be able to use this workflow.
  - Nextflow also handles parallelizing processing based on your environment and will configure processing so that run time is minimal.
  - In addition to the data that we processed through `scpca-nf`, there are a handful of projects on the Portal that we did not process, as external users were able to implement and run `scpca-nf` on their own raw data prior to submission.
  - Because of the portability of `scpca-nf`, members of the pediatric research community with single-cell and single-nuclei data are able to run the workflow in the environment of their choice and process samples in the same uniform way that samples on the Portal have been processed.
  - This allows us to easily grow the Portal and accept contributions from the community and allows users to process their own data in a quick-efficient manner for easy comparison to samples on the Portal.

## Making samples with additional modalities available on the Portal

1. Processing samples with additional modalities
  - In addition to samples that have single-cell and single-nuclei RNA-seq, we also received samples from submitters with additional sequencing modalities, including CITE-seq, cell hashing, spatial transcriptomics, and bulk RNA-seq.
  - To make all the data that we received available, we included additional modules in `scpca-nf` that would accommodate these additional sequencing modalities.
  - For a full summary of the libraries and samples available with additional modalities, see supplemental Table 1.
  - `scpca-nf` is capable of processing samples that are from a mix of sample types. This means that libraries with and without ADT data or any of the modalities discussed here can be processed together in a single run. Nextflow will handle the parallel processing and that each sample type is run through the correct processes for that modality type.

2. CITE-seq and cell hashing (Fig. S2)
  - For all libraries with associated ADT or CITE-seq data, we provide both the RNA and ADT gene expression data in the files available for download on the portal.
  - Both FASTQ from single-cell/single-nuclei and from ADT were input into `scpca-nf` and quantified using `salmon alevin` and `alevin-fry`.
  - We required a barcodes file from each submitter that contained the ADT labels and the associated barcode. This was used to build the index used for quantification of the ADT FASTQ and creation of the cell by ADT matrix.
  - Unlike with RNA counts, we do not perform any filtering of cells due to low quality ADT expression. However, we do include the results from running `DropletUtils::cleanTagCounts()` in both the filtered and processed objects produced by `scpca-nf`.
  - Similar to RNA counts, we do normalize ADT data and provide the normalized counts matrix in the processed SCE object.
  - ADT data can be found in the `altExp` slot of each `SingleCellExperiment` object or as a separate `_adt.hdf5` file for `AnnData` objects.
  - Similar to ADT data, if any libraries were multiplexed and have an associated cell hashing library, both the RNA and HTO FASTQ are provided as input to the workflow and quantified with `salmon alevin` and `alevin-fry`. We also include a library pools file which indicates which libraries contain which samples and the associated tags used to label each sample.
  - Although we quantify the HTO data and include the cell by HTO counts matrix in all objects, we do not demultiplex the samples so that there is one sample per library. Instead, we apply multiple demultiplexing methods including genetic demultiplexing, demultiplexing with `DropletUtils::hashedDrops()`, and demultipilexing with `Seurat::HTODemux()`. The results from these three methods are included in the filtered and processed objects.
  - Maybe some more details about how we do genetic demultiplexing.

3. Additions to the QC report (Fig. S2C)
  - If a library has associated ADT or cellhash data, an additional section is added to the QC report included on the portal and output by `scpca-nf`.
  - This section includes a summary of statistics such as how many cells express each ADT or HTO.
  - For libraries with ADT, we also include additional plots.
  - As mentioned above, we include the results from `DropletUtils::cleanTagCounts()`, but do not filter any ADTs or cells from the object. Instead we include a column in the `colData` of the processed SCE object that indicates if it is recommended to remove ADTs or not. In the QC report, we summarize filtering taking into account removal of cells because of low quality RNA or ADT. The plot shown in the report highlights which cells would be removed if only filtering using RNA, only ADT, or both.
  - Similar to the UMAPs for the RNA data from single-cell/single-nuclei that highlight the top variable genes, the report includes UMAPs highlighing the 4 most variable ADTs in the data. This is shown with UMAPs and ridge plots.
  - For HTO, we do not include any plots, but we do show a table summarizing how many cells belong to each sample included in the multiplexed library using each of the demultiplexing methods mentioned.

4. Bulk and Spatial
  - Some samples underwent sequencing using both single-cell/single-nuclei and an additional method like bulk RNA-seq or spatial transcriptomics.
  - `scpca-nf` is able to quantify both of these additional sequencing methods.
  - Bulk RNA FASTQ are first trimmed using `fastp` and then aligned using `salmon`. The bulk output is a single tsv file with the sample by gene matrix for all samples in that project.
  - For spatial transcriptomics, the spatial RNA FASTQ and slide image are input into `scpca-nf` and quantified using `spaceranger`. The output includes the spot by gene matrix along with a summary report, produced by `spaceranger`.
