# Results

## The Single-cell Pediatric Cancer Atlas Portal

In March of 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata from pediatric tumor samples available for download.
Data available on the Portal was obtained using two different mechanisms: raw data was accepted from ALSF-funded investigators and processed using our open-source pipeline, `scpca-nf`, or investigators processed their raw data using `scpca-nf` and submitted the output for inclusion on the Portal.

All samples on the Portal include a core set of metadata obtained from investigators, including age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
Some investigators submitted additional metadata, such as treatment and tumor stage, which can also be found on the Portal.
All submitted metadata was standardized to maintain consistency across projects before adding to the Portal.
In addition to providing a human-readable value for the submitted metadata, we also provide an ontology term ID, if applicable.
Submitted metadata was mapped to an associated ontology term IDs obtained from HsapDV (age) [@url:https://www.ebi.ac.uk/ols4/ontologies/hsapdv], PATO (sex) [@url:https://www.ebi.ac.uk/ols4/ontologies/pato], NCBI taxonomy (organism) [@url:https://www.ncbi.nlm.nih.gov/taxonomy], MONDO (disease) [@url:https://www.ebi.ac.uk/ols4/ontologies/mondo], UBERON (tissue) [@url:https://www.ebi.ac.uk/ols4/ontologies/uberon], and Hancestro (ethnicity, if applicable) [@url:https://www.ebi.ac.uk/ols4/ontologies/hancestro]. 
Including ontology term IDs for each sample provides users with standardized metadata terms that can be used across all projects.

The Portal contains data from 500 samples and over 50 tumor types.
<!-- TODO: Update numbers -->
The total number of samples for each diagnosis is shown in Figure 1A, along with a breakdown of the proportion of samples from each disease stage within a diagnosis group.
Figure 1A summarizes all samples from patient tumors or patient-derived xenografts currently available on the Portal.
Most samples found on the Portal were obtained from patients with leukemia (n = 192). 
The Portal also includes samples from brain and central nervous system tumors (n = 154), sarcoma and soft tissue tumors (n = 68), and a variety of other solid tumors (n = 87). 
Most samples were collected at initial diagnosis (n = 424), with a smaller number of samples collected either at recurrence (n = 64), during progressive disease (n = 10), or post-mortem (n = 2). 
Along with the patient tumors, the Portal contains a small number of human tumor cell line samples (n = 4).


Each of the available samples contains summarized gene expression data from either single-cell or single-nuclei RNA sequencing. 
However, some samples also include additional data, such as quantified expression data from tagging cells with Antibody-derived tags (ADT), like CITE-seq antibodies [@doi:10.1038/nmeth.4380], or multiplexing samples with hashtag oligonucleotides (HTO)[@doi:10.1186/s13059-018-1603-1] prior to sequencing. 
Out of the 500 samples, 96 have associated CITE-seq data, and 19 have associated multiplexing data. 
In some cases, multiple libraries from the same sample were collected for additional sequencing, either for bulk RNA-seq or spatial transcriptomics. 
Specifically, 118 samples on the Portal were sequenced using bulk RNA-seq and 94 samples were sequenced using spatial transcriptomics.
A summary of the number of samples with each additional modality is shown in Figure 1B, and a detailed summary of the total samples with each sequencing method broken down by project is available in Supplemental Table 1.

Samples on the Portal are organized by project, where each project is a collection of similar samples from an individual lab.
Users can filter projects based on diagnosis, included modalities (e.g., CITE-seq, bulk RNA-seq), 10X Genomics version (e.g., 10Xv2, 10Xv3), and whether or not a project includes samples derived from patient-derived xenografts or cell lines.
The project card displays an abstract, the total number of samples included, a list of diagnoses for all samples included in the Project, and links to any external information associated with the project, such as publications and links to external data, such as SRA or GEO (Figure 1C).
The project card will also indicate the type(s) of sequencing performed, including the 10X Genomics kit version, the suspension type (cell or nucleus), and if additional sequencing is present, like bulk RNA-seq or multiplexing.

## Uniform processing of data available on the ScPCA Portal

All data available on the Portal was uniformly processed using [`scpca-nf`](https://github.com/AlexsLemonade/scpca-nf), an open-source and efficient Nextflow[@url:https://www.nextflow.io/docs/latest/index.html] workflow for quantifying single-cell and single-nuclei RNA-seq data. 
Using Nextflow as the backbone for the `scpca-nf` workflow ensures both reproducibility and portability.
All dependencies for the workflow are handled automatically, as each process in the workflow is run in a Docker container.
Nextflow is compatible with various computing environments, including high-performance computing and cloud-based computing, allowing users to run the workflow in their preferred environment. 
Setup requires organizing input files and updating a single configuration file for your computing environment after installing Nextflow and either Docker or Singularity. 
Nextflow will also handle parallelizing sample processing as allowed by your environment, minimizing run time. 
The combination of being able to execute a Nextflow workflow in any environment and run individual processes in Docker containers makes this workflow easily portable for external use. 

When building `scpca-nf`, we sought a fast and memory-efficient tool for gene expression quantification to minimize processing costs.
We expected many users of the Portal to have their own single-cell or single-nuclei data processed with Cell Ranger[@url:https://www.10xgenomics.com/support/software/cell-ranger/latest], due to its popularity.
Thus, selecting a tool with comparable results to Cell Ranger was also desirable.
In comparing `alevin-fry` [@doi:10.1038/s41592-022-01408-3] to Cell Ranger, we found `alevin-fry` had a lower run time and memory usage (Supplemental Figure 1A), while retaining comparable mean gene expression for all genes (Supplemental Figure 1B), total UMIs per cell (Supplemental Figure 1C),  or total genes detected per cell (Supplemental Figure 1D). 
<!--TODO: Do we like including this here or is it not worth mentioning? --> 
(All analyses comparing gene expression quantification tools are available in a public analysis repository[@url:https://github.com/AlexsLemonade/alsf-scpca].)
Based on these results, we elected to use `salmon alevin` and `alevin-fry` [@doi:10.1038/s41592-022-01408-3] in `scpca-nf` to quantify gene expression data.

`scpca-nf` takes FASTQ files as input (Figure 2A).
Reads are aligned using the selective alignment option of `salmon alevin` to an index with transcripts corresponding to spliced cDNA and intronic regions, denoted by `alevin-fry` as a `splici` index. 
The output from `alevin-fry` includes a gene-by-cell count matrix for all barcodes identified, even those that may not contain true cells. 
This unfiltered counts matrix is stored in a `SingleCellExperiment` object[@doi:10.1038/s41592-019-0654-x] and output from the workflow to a `.rds` file with the suffix `_unfiltered.rds`.

`scpca-nf` performs filtering of empty droplets, removal of low-quality cells, normalization, dimensionality reduction, and cell type annotation (Figure 2A). 
The unfiltered gene by cell counts matrices are filtered to remove any barcodes that are not likely to contain cells using `DropletUtils::emptyDropsCellRanger()`[@doi:10.1186/s13059-019-1662-y], with all cells that pass being saved to a `SingleCellExperiment` object and `.rds` file with the suffix `_filtered.rds`.
Then, low-quality cells are identified and removed with `miQC` [@doi: 10.1371/journal.pcbi.1009290], which jointly models the proportion of mitochondrial reads and detected genes per cell and calculates a probability that each cell is compromised. 
The remaining cells are normalized [@doi:10.1186/s13059-016-0947-7] and undergo dimensionality reduction using both principal component analysis (PCA) and UMAP.
Finally, cell types are classified using two automated methods, `SingleR`[@doi:10.1038/s41590-018-0276-y] and `CellAssign`[@doi:10.1038/s41592-019-0529-1]. 
The results from this analysis are stored in a processed `SingleCellExperiment` object saved to a `.rds` file with the suffix `_processed.rds`.

To make downloading from the Portal convenient for R and Python users, downloads are available as either `SingleCellExperiment` or `AnnData`[@doi:10.1101/2021.12.16.473007] objects.
All `SingleCellExperiment` objects saved as `.rds` files are converted to `AnnData` objects and saved as `.hdf5` files in `scpca-nf` (Figure 2A).
Downloads contain the unfiltered, filtered, and processed objects from `scpca-nf` to allow users to choose to perform their own filtering and normalization or to start their analysis from a processed object.

All downloads from the Portal include a quality control (QC) report with a summary of processing information (e.g., `alevin-fry` version), library statistics (e.g., the total number of cells), and a collection of diagnostic plots for each library (Figure 2B-G).
The knee plot includes all droplets (i.e., before removing empty droplets) sorted based on the total number of UMIs, and those retained after filtering empty droplets are indicated in the plot (Figure 2B).
For each cell that remains after filtering empty droplets, the number of total UMIs, genes detected, and mitochondrial reads are calculated and summarized in a scatter plot (Figure 2C). 
We include plots showing the `miQC` model and which cells are kept and removed after filtering with `miQC` (Figure 2D-E).
A UMAP plot with cells colored by the total number of genes detected and a faceted UMAP plot where cells are colored by the expression of a top highly variable gene are also available (Figure 2F-G).


## Making samples with additional modalities available on the Portal

Each sample on the Portal will have summarized gene expression data from either single-cell or single-nuclei RNA-seq. 
For some samples, submitters included data from additional sequencing modalities, including corresponding ADT or CITE-seq data [@doi:10.1038/nmeth.4380], multiplexing using cell hashing [@doi:10.1186/s13059-018-1603-1], spatial transcriptomics, or bulk RNA-seq. 
To make all received data available, we included additional modules in `scpca-nf` to uniformly process samples with these additional sequencing modalities.
For a full summary of the libraries and samples available with additional modalities, see Supplemental Table 1.

For all libraries with associated ADT or CITE-seq data, we provide both the summarized RNA and ADT gene expression data for download on the portal.
To process these libraries, both FASTQ from single-cell or single-nuclei RNA-seq and CITE-seq were provided as input into `scpca-nf` and quantified using `salmon alevin` and `alevin-fry` (Supplemental Figure 2A).
Along with the FASTQ files,  we required a tsv file from each submitter that contained the labels for each ADT used and the associated barcode. 
This file is also input to `scpca-nf` and used to build the index used to quantify ADT expression and create the cell by ADT counts matrix.
Unlike with RNA counts, we do not perform any filtering of cells due to low-quality ADT expression. 
However, we do include the results from running `DropletUtils::cleanTagCounts()` on the ADT counts matrix in both the filtered and processed objects output by `scpca-nf`.
Similar to RNA counts, we normalize ADT data and provide the normalized counts matrix in the processed object. 
Although we do normalize the ADT counts, we do not provide any dimensionality reduction of ADT data; only the RNA counts data is used as input for dimensionality reduction.
For all three `SingleCellExperiment` objects (unfiltered, filtered, and processed), the summarized ADT expression data can be found in the `altExp` slot. 
With `AnnData` objects, the ADT data is provided as a separate `_adt.hdf5` file for each of the three objects.

If a library contains associated ADT data, an additional section will be included in the provided QC report. 
This section includes a summary of statistics, such as how many cells express each ADT.
We also include a collection of additional plots specific to ADT data (Supplemental Figure 2B).  
As mentioned above, we include the results from `DropletUtils::cleanTagCounts()`, but do not filter any ADTs or cells from the object. 
Instead, we include a column in the `colData` of the processed `SingleCellExperiment` object (`.obs` in the `AnnData` object) that indicates if a cell is recommended to be removed due to low expression of ADTs.
In the QC report, we summarize filtering low-quality cells based on both RNA and ADT counts.
The first quadrant shows which cells would be kept if filtering on both RNA and ADT. 
Each of the other facets highlights which cells would be removed if filtering is only done using RNA counts, ADT counts, or both.
The top 4 ADTs with the most variable expression are also identified and visualized using both UMAPs and density plots. 
The UMAPs shown were calculated using the RNA data but cells are colored by ADT expression, while the ridge plots show the normalized ADT expression across all cells. 

Similar to ADT data, if any libraries contain samples that have been multiplexed and have an associated cell hashing library, both the RNA and HTO FASTQ are provided as input to the workflow and quantified with `salmon alevin` and `alevin-fry` (Supplemental Figure 2C). 
`scpca-nf` also requires a tsv file with one row per sample included in the library, which tells the workflow which HTO was used for which sample when multiplexing the library. 
The HTO counts data will be included as an `altExp` in each of the `SingleCellExperiment` objects.
Although we quantify the HTO data and include the cell by HTO counts matrix in all objects, we do not demultiplex the samples so that there is one sample per library. 
Instead, we apply multiple demultiplexing methods, including demultiplexing with `DropletUtils::hashedDrops()`, demultiplexing with `Seurat::HTODemux()`, and genetic demultiplexing when possible.
The genetic demultiplexing used in `scpca-nf` uses the method described in Weber et al [@doi:[10.1093/gigascience/giab062](https://doi.org/10.1093/gigascience/giab062)], which takes bulk RNA-seq data from the same sample as a reference for the expected genotypes found in each sample.
<!--Do we need to list all individual tools used here or are we okay to mention them in the methods section, e.g., STAR, bcftools, STARsolo, cellsnp--> 
Variants among the samples within each pool are identified from both the mapped bulk RNA-seq data and pooled single-cell or single-nuclei RNA-seq. 
After variants from both bulk and single-cell or single-nuclei RNA-seq are genotyped, `vireo` is used to match genotypes and identify the most likely sample of origin [@doi:10.1186/s13059-019-1865-2].
The results from these three demultiplexing methods are included in the filtered and processed objects on the Portal. 
If a sample does not have associated bulk RNA-seq data, then the objects will not contain genetic demultiplexing results. 

If a library has associated HTO data, an additional section is added to the QC report included on the Portal and output by `scpca-nf`.
This section includes a summary of library statistics, such as how many cells express each HTO.
We do not include any additional plots, but we do show a table summarizing how many cells belong to each sample included in the multiplexed library using each of the demultiplexing methods mentioned.

For some samples, multiple libraries were collected, with the additional libraries being used for  bulk RNA-seq and/or spatial transcriptomics. 
Both of these additional sequencing methods are supported by `scpca-nf`, and the output is available for download on the Portal. 
To process bulk RNA-seq data, reads are first trimmed using `fastp` and then aligned using `salmon` (Supplemental Figure 3A). 
The bulk module of `scpca-nf` outputs a single tsv file with the sample by gene counts matrix for all samples in a given ScPCA project.
This sample by gene matrix is included only with project downloads on the Portal. 
As there is not yet support for spatial transcriptomics with `alevin-fry`, `scpca-nf` uses Space Ranger to quantify all spatial transcriptomics data [@url:https://www.10xgenomics.com/support/software/space-ranger/latest] (Supplemental Figure 3B).
The required input includes the RNA FASTQ files and slide image. 
The output includes the spot-by-gene matrix along with a summary report produced by Space Ranger.
If a sample on the Portal has associated spatial transcriptomics data, users will be able to download the summarized gene expression data from single-cell or single-nuclei RNA-seq separately from the spatial output. 


## Downloading projects from the ScPCA Portal

1. Users can download all samples for a given project together
  - The portal has two different options to allow users to download data for all samples in a given ScPCA Project, either as invididual files for each sample or as a single merged file.
  - By default, when downloading a project, the download will include a folder for each sample that is included in the project.
  - That folder will contain all individual `SingleCellExperiment` objects as `.rds` files or `AnnData` objects as `.hdf5` files, depending on the file format chosen by the user (Fig. 3A).
  - Each of these objects contains the gene expression data and metadata for a single library.
  - If a given project has associated bulk RNA-seq, then a sample by gene counts matrix, `bulk_quant.tsv`, including the quantified gene expression data for all samples in a project with associated bulk RNA-seq will be included.
2. Merged objects
  - Providing all data from all libraries withing a single file makes it easier for users to perform joint analysis on multiple samples at the same time. 
  Specifically, these objects can be useful for comparing gene-level metrics across multiple samples, such as differential expression analysis and gene set enrichment analysis.
  - Therefore, we make a single, merged `SingleCellExperiment` or `AnnData` object (Fig. 3B) available for each project (without batch-correction or integration).
  - This file contains one object with all raw and normalized gene expression data and metadata for all single-cell and single-nuclei RNA-seq libraries within a given ScPCA project
  - If downloading a project that contains at least one library with CITE-seq, the quantified CITE-seq expression data will also be merged. In SCEs this is provided as an `altExp` within the main object, but for `AnnData` objects, the quantified CITE-seq data is provided as a separate file.

2. The merged object workflow (Fig. 3C and 3D)
  - To create the merged objects, we created an additional stand-alone workflow for merging the output from `scpca-nf`, `merge.nf` (Fig. 3C).
  - Following processing of each `SingleCellExperiment` object with `scpca-nf`, all processed objects from all libraries and samples within a project are input to the merge workflow, which combines all input data into a single merged object.
  - The merged object contains raw and normalized gene expression counts for all cells in all libraries. The same index was used for processing all individual libraries, so the genes found will be the same as in an invididual object.
  - After merging, the top 2000 high-variance genes are calculated by modeling variance within each library included in the merged object.
  - These high-variance genes are used to calculate new PCA coordinates using `batchelor::multiBatchPCA()` and specifying librares as batches.
  - The top 50 PCs were selected and used as input to calculate new UMAP embeddings on the merged object.
  - Similar to `scpca-nf`, the merged `SingleCellExperiment` object is converted to a merged `AnnData` object and both formats are provided as download options on the Portal.
  - Along with the merged objects, for each project, a merged summary report is created and output.
  - This report includes a brief summary of the samples and libraries included in the merged object, including a summary of the type of libraries (e.g., single-cell, single-nuclei, with CITE-seq) and sample diagnoses included in the object.
  - The report also contains a UMAP showing all cells from all libraries included in the merged object. For each library, a separate panel is shown, and cells from that library are colored while all other cells are gray (Fig. 3D).
  
