# Results

## The Single-cell Pediatric Cancer Atlas Portal

1. History of the Portal
  - In 2019, Alex's Lemonade Stand Foundation funded 10 investigators working on single-cell profiling to create a publicly available atlas of single-cell pediatric cancer data.
  ** are there more specific information about how things were funded that I should include?**
  - These 10 investigators produced single-cell and single-nuclei RNA-seq data across a variety of tumors
  - The raw data and associated metadata was sent to the Childhood Cancer Data Lab for uniform processing
  - We created an open-source efficient pipeline to process all samples present on the Portal
  - In 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata available for download.
  - In 2023, we began to expand the Portal to other investigators, putting out a call for contributions.
  - For these projects, investigators used our open-source pipeline to produce summarized gene expression data for inclusion on the portal.

2. Overview of samples on the Portal
  - The Portal currently holds X amount of samples from X amount of tumor types.
  - For all samples we collect a core set of metadata including, age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
  - Fig. 1A shows how many samples we have from each type of tumor. For each diagnosis, we also indicate what proportion of the samples come from each disease stage (e.g., initial diagnosis, recurrence, post-mortem).
  - In addition to single-cell and single-nuclei RNA-seq, many samples have associated bulk RNA-seq, ADT data (CITE-seq), cell hashing, or spatial transcriptomics.
  - Fig. 2B summarizes the total number of samples that are single-cell vs. single-nuclei. Additionally, we show how many of the samples on the portal also have either bulk, CITE, cell hashing, or spatial data.
  - Supplemental Table 1 shows a breakdown of how many of each modality is found in each project.
  ** By submission we will provably have PDX/ cell line information on the portal so we might want to mention that here and add to figures?**

3. Obtaining data and additional project information
  - On the Portal, samples are organized by project. Each project is a collection of similar samples from a single investigator.
  - To select projects of interest, users can filter based on diagnosis, modality included, single-cell or single-nuclei and 10X version. Additionally, users will be able to filter based on if the project includes cell line samples or xenografts.
  - A summary of each project, including a list of samples found in each project, is displayed on the Portal.
  - Fig.1C shows an example of this summary which include an abstract, links to any external information about the projects such as any associated publication information, and links to external places where data may be stored such as SRA or GEO.
  - Users can download all samples from an entire project or choose individual samples.
  - Along with the summarized gene expression data, a file containing de-identified clinical metadata for each sample in the download is included.
  - For more information on what is included in each download see download section of the docs.
