# Results

## The Single-cell Pediatric Cancer Atlas Portal

In March of 2022, the Childhood Cancer Data Lab launched the Single-cell Pediatric Cancer Atlas (ScPCA) Portal to make uniformly processed, summarized single-cell and single-nuclei RNA-seq data and de-identified metadata from pediatric tumor samples available for download.
Data available on the Portal was obtained using two different mechanisms: raw data was accepted from ALSF-funded investigators and processed using our open-source pipeline, `scpca-nf`, or investigators processed their raw data using `scpca-nf` and submitted the output for inclusion on the Portal.

All samples on the Portal include a core set of metadata obtained from investigators, including age, sex, diagnosis, subdiagnosis (if applicable), tissue location, and disease stage.
Some investigators submitted additional metadata, such as treatment and tumor stage, which can also be found on the Portal.
All submitted metadata was standardized to maintain consistency across projects before adding to the Portal.
In addition to providing a human-readable value for the submitted metadata, we also provide an ontology term ID, if applicable.
Submitted metadata was mapped to an associated ontology term IDs obtained from HsapDV (age) [@url:https://www.ebi.ac.uk/ols4/ontologies/hsapdv], PATO (sex) [@url:https://www.ebi.ac.uk/ols4/ontologies/pato], NCBI taxonomy (organism) [@url:https://www.ncbi.nlm.nih.gov/taxonomy], MONDO (disease) [@url:https://www.ebi.ac.uk/ols4/ontologies/mondo], UBERON (tissue) [@url:https://www.ebi.ac.uk/ols4/ontologies/uberon], and Hancestro (ethnicity, if applicable) [@url:https://www.ebi.ac.uk/ols4/ontologies/hancestro]. 
Including ontology term IDs for each sample provides users with standardized metadata terms that can be used across all projects.

The Portal contains data from 500 samples and over 50 tumor types.
<!-- TODO: Update numbers -->
The total number of samples for each diagnosis is shown in Figure 1A, along with a breakdown of the proportion of samples from each disease stage within a diagnosis group.
Figure 1A summarizes all samples from patient tumors or patient-derived xenografts currently available on the Portal.
Most samples found on the Portal were obtained from patients with leukemia (n = 192). 
The Portal also includes samples from brain and central nervous system tumors (n = 154), sarcoma and soft tissue tumors (n = 68), and a variety of other solid tumors (n = 87). 
Most samples were collected at initial diagnosis (n = 424), with a smaller number of samples collected either at recurrence (n = 64), during progressive disease (n = 10), or post-mortem (n = 2). 
Along with the patient tumors, the Portal contains a small number of human tumor cell line samples (n = 4).


Each of the available samples contains summarized gene expression data from either single-cell or single-nuclei RNA sequencing. 
However, some samples also include additional data, such as quantified expression data from tagging cells with Antibody-derived tags (ADT), like CITE-seq antibodies [@doi:10.1038/nmeth.4380], or multiplexing samples with hashtag oligonucleotides (HTO)[@doi:10.1186/s13059-018-1603-1] prior to sequencing. 
Out of the 500 samples, 96 have associated CITE-seq data, and 19 have associated multiplexing data. 
In some cases, multiple libraries from the same sample were collected for additional sequencing, either for bulk RNA-seq or spatial transcriptomics. 
Specifically, 118 samples on the Portal were sequenced using bulk RNA-seq and 94 samples were sequenced using spatial transcriptomics.
A summary of the number of samples with each additional modality is shown in Figure 1B, and a detailed summary of the total samples with each sequencing method broken down by project is available in Supplemental Table 1.

Samples on the Portal are organized by project, where each project is a collection of similar samples from an individual lab.
Users can filter projects based on diagnosis, included modalities (e.g., CITE-seq, bulk RNA-seq), 10X Genomics version (e.g., 10Xv2, 10Xv3), and whether or not a project includes samples derived from patient-derived xenografts or cell lines.
The project card displays an abstract, the total number of samples included, a list of diagnoses for all samples included in the Project, and links to any external information associated with the project, such as publications and links to external data, such as SRA or GEO (Figure 1C).
The project card will also indicate the type(s) of sequencing performed, including the 10X Genomics kit version, the suspension type (cell or nucleus), and if additional sequencing is present, like bulk RNA-seq or multiplexing.

## Uniform processing of data available on the ScPCA Portal

All data available on the Portal was uniformly processed using [`scpca-nf`](https://github.com/AlexsLemonade/scpca-nf), an open-source and efficient Nextflow[@url:https://www.nextflow.io/docs/latest/index.html] workflow for quantifying single-cell and single-nuclei RNA-seq data. 
Using Nextflow as the backbone for the `scpca-nf` workflow ensures both reproducibility and portability.
All dependencies for the workflow are handled automatically, as each process in the workflow is run in a Docker container.
Nextflow is compatible with various computing environments, including high-performance computing and cloud-based computing, allowing users to run the workflow in their preferred environment. 
Setup requires organizing input files and updating a single configuration file for your computing environment after installing Nextflow and either Docker or Singularity. 
Nextflow will also handle parallelizing sample processing as allowed by your environment, minimizing run time. 
The combination of being able to execute a Nextflow workflow in any environment and run individual processes in Docker containers makes this workflow easily portable for external use. 

When building `scpca-nf`, we sought a fast and memory-efficient tool for gene expression quantification to minimize processing costs.
We expected many users of the Portal to have their own single-cell or single-nuclei data processed with Cell Ranger[@url:https://www.10xgenomics.com/support/software/cell-ranger/latest], due to its popularity.
Thus, selecting a tool with comparable results to Cell Ranger was also desirable.
In comparing `alevin-fry` [@doi:10.1038/s41592-022-01408-3] to Cell Ranger, we found `alevin-fry` had a lower run time and memory usage (Supplemental Figure 1A), while retaining comparable mean gene expression for all genes (Supplemental Figure 1B), total UMIs per cell (Supplemental Figure 1C),  or total genes detected per cell (Supplemental Figure 1D). 
(All analyses comparing gene expression quantification tools are available in a public analysis repository[@url:https://github.com/AlexsLemonade/alsf-scpca].)
Based on these results, we elected to use `salmon alevin` and `alevin-fry` [@doi:10.1038/s41592-022-01408-3] in `scpca-nf` to quantify gene expression data.

`scpca-nf` takes FASTQ files as input (Figure 2A).
Reads are aligned using the selective alignment option of `salmon alevin` to an index with transcripts corresponding to spliced cDNA and intronic regions, denoted by `alevin-fry` as a `splici` index. 
The output from `alevin-fry` includes a gene by cell count matrix for all barcodes identified, even those that may not contain true cells. 
This unfiltered counts matrix is stored in a `SingleCellExperiment` object[@doi:10.1038/s41592-019-0654-x] and output from the workflow to a `.rds` file with the suffix `_unfiltered.rds`.

`scpca-nf` performs filtering of empty droplets, removal of low-quality cells, normalization, dimensionality reduction, and cell type annotation (Figure 2A). 
The unfiltered gene by cell counts matrices are filtered to remove any barcodes that are not likely to contain cells using `DropletUtils::emptyDropsCellRanger()`[@doi:10.1186/s13059-019-1662-y], with all cells that pass being saved to a `SingleCellExperiment` object and `.rds` file with the suffix `_filtered.rds`.
Then, low-quality cells are identified and removed with `miQC` [@doi:10.1371/journal.pcbi.1009290], which jointly models the proportion of mitochondrial reads and detected genes per cell and calculates a probability that each cell is compromised. 
The remaining cells are normalized [@doi:10.1186/s13059-016-0947-7] and undergo dimensionality reduction using both principal component analysis (PCA) and UMAP.
Finally, cell types are classified using two automated methods, `SingleR`[@doi:10.1038/s41590-018-0276-y] and `CellAssign`[@doi:10.1038/s41592-019-0529-1]. 
The results from this analysis are stored in a processed `SingleCellExperiment` object saved to a `.rds` file with the suffix `_processed.rds`.

To make downloading from the Portal convenient for R and Python users, downloads are available as either `SingleCellExperiment` or `AnnData`[@doi:10.1101/2021.12.16.473007] objects.
All `SingleCellExperiment` objects saved as `.rds` files are converted to `AnnData` objects and saved as `.hdf5` files in `scpca-nf` (Figure 2A).
Downloads contain the unfiltered, filtered, and processed objects from `scpca-nf` to allow users to choose to perform their own filtering and normalization or to start their analysis from a processed object.

All downloads from the Portal include a quality control (QC) report with a summary of processing information (e.g., `alevin-fry` version), library statistics (e.g., the total number of cells), and a collection of diagnostic plots for each library (Figure 2B-G).
The knee plot includes all droplets (i.e., before removing empty droplets) sorted based on the total number of UMIs, and those retained after filtering empty droplets are indicated in the plot (Figure 2B).
For each cell that remains after filtering empty droplets, the number of total UMIs, genes detected, and mitochondrial reads are calculated and summarized in a scatter plot (Figure 2C). 
We include plots showing the `miQC` model and which cells are kept and removed after filtering with `miQC` (Figure 2D-E).
A UMAP plot with cells colored by the total number of genes detected and a faceted UMAP plot where cells are colored by the expression of a top highly variable gene are also available (Figure 2F-G).


## Processing samples with additional modalities

Each sample on the Portal will have summarized gene expression data from either single-cell or single-nuclei RNA-seq. 
For some samples, submitters included data from additional sequencing modalities, including corresponding ADT or CITE-seq data [@doi:10.1038/nmeth.4380], multiplexing using cell hashing [@doi:10.1186/s13059-018-1603-1], spatial transcriptomics, or bulk RNA-seq. 
`scpca-nf` includes additional modules to ensure uniform processing of samples with additional sequencing modalities.

To process ADT libraries, the ADT FASTQ were provided as input into `scpca-nf` and quantified using `salmon alevin` and `alevin-fry` (Supplemental Figure 2A).
Along with the FASTQ files,  `scpca-nf` takes a tab-separated values (tsv) file containing one row for each ADT used.
This file contains the name used for the ADT and associated barcode and is required for building an ADT-specific index for quantifying ADT expression with `alevin-fry`. 
The output from `alevin-fry` is the unfiltered ADT by cell counts matrix.
The ADT by cell counts matrix is read into R alongside the gene by cell counts matrix and saved as an alternative experiment (`altExp`) within the main `SingleCellExperiment` containing the unfiltered RNA counts. 
This `SingleCellExperiment` object containing both RNA and ADT counts is output from the workflow to a `.rds` file with the suffix `_unfiltered.rds`.

`scpca-nf` does not filter any cells based on ADT expression. 
`DropletUtils::emptyDropsCellRanger()` is only applied to the unfiltered RNA counts matrix to remove empty droplets. 
Any cells removed after filtering empty droplets are also removed from the ADT counts matrix.
`scpca-nf` also does not remove cells with low-quality ADT expression, but the workflow does calculate QC statistics for ADT counts using `DropletUtils::cleanTagCounts()`. 
These ADT QC statistics are stored alongside the ADT by cell counts matrix in the filtered `SingleCellExperiment` object. 
The `SingleCellExperiment` object containing both the filtered RNA and ADT counts matrix, along with associated ADT QC statistics, is saved to an `.rds` file with the suffix `_filtered.rds`.

Similar to processing RNA gene expression data, the ADT by cell counts matrix is normalized.
Normalized counts are not calculated for any cells that would be removed because of low-quality ADT expression, indicated by `DropletUtils::cleanTagCounts()`. 
Although `scpca-nf` normalizes ADT counts, the workflow does not perform any dimensionality reduction of ADT data; only the RNA counts data is used as input for dimensionality reduction.
The normalized ADT data is saved as an `altExp` within the processed `SingleCellExperiment` containing the normalized RNA data and is output to a `.rds` file with the suffix `_processed.rds`.
All `.rds` files containing `SingleCellExperiment` objects and associated `altExp` objects, are converted to `AnnData` objects and exported as separate RNA (`_rna.hdf5`) and ADT (`_adt.hdf5`) AnnData objects. 

If a library contains associated ADT data, the QC report output by `scpca-nf` will include an additional section. 
This section includes a summary of ADT-related statistics, such as how many cells express each ADT.
The report also includes a collection of additional plots specific to ADT data (Supplemental Figure 2B-D).  
As mentioned above, `scpca-nf` uses `DropletUtils::cleanTagCounts()` to calculate QC statistics for each cell using ADT expression but does not filter any cells from the object. 
A summary of removing low-quality cells based on both RNA and ADT counts is included in the QC report (Supplemental Figure 2B).
The first quadrant indicates which cells would be kept if the object was filtered on both RNA and ADT. 
The other facets highlight which cells would be removed if filtering was done using only RNA counts, only ADT counts, or both.
The top 4 ADTs with the most variable expression are also identified and visualized using UMAPs (Supplemental Figure 2C) and density plots (Supplemental Figure 2D). 
The UMAPs shown were calculated using the RNA data, and cells are colored by ADT expression, while the density plots show the normalized ADT expression across all cells. 

To process multiplexed libraries, the HTO FASTQ is input to `scpca-nf` and quantified using `salmon alevin` and `alevin-fry` (Supplemental Figure 2C).
Along with the FASTQ files,  `scpca-nf` requires two tsv files for processing multiplexed data.
The first is similar to the barcode file required when quantifying ADT expression and contains the HTO name and associated barcode. 
This file is needed to build an HTO-specific index for quantifying HTO expression with `alevin-fry`. 
The second tsv file contains one row for each sample included in the multiplexed library and tells the workflow which HTO was used for which sample when multiplexing the library. 
The output from `alevin-fry` is the HTO by cell counts matrix.
The HTO by cell counts matrix is read into R alongside the gene by cell counts matrix.  
The unfiltered HTO by cell matrix is saved as an alternative experiment (`altExp`) within the main `SingleCellExperiment` containing the unfiltered RNA counts. 
This `SingleCellExperiment` object containing both RNA and HTO counts is output from the workflow to a `.rds` file with the suffix `_unfiltered.rds`.

As with ADT data, `scpca-nf` does not filter any cells based on HTO expression. 
`DropletUtils::emptyDropsCellRanger()` is only applied to the unfiltered RNA counts matrix to remove empty droplets, and any cells removed after filtering empty droplets are also removed from the HTO counts matrix and saved to an `.rds` file with the `_filtered.rds` suffix.
`scpca-nf` does not perform any additional filtering or processing of the HTO by cell counts matrix, so the same filtered matrix is saved to the processed `.rds` file with the `_processed.rds` suffix.

Although `scpca-nf` quantifies the HTO data and includes an HTO by cell counts matrix in all objects, `scpca-nf` does not demultiplex the samples into one sample per library. 
Instead, `scpca-nf` applies multiple demultiplexing methods, including demultiplexing with `DropletUtils::hashedDrops()`, demultiplexing with `Seurat::HTODemux()`, and genetic demultiplexing, if possible.
The genetic demultiplexing used in `scpca-nf` uses the method described in Weber et al [@doi:10.1093/gigascience/giab062], which takes bulk RNA-seq data and single-cell RNA-seq data from the same sample. 
The bulk RNA-seq serves as a reference for the expected genotypes found in each sample.
If a sample lacks associated bulk RNA-seq data, then no genetic demultiplexing is performed.
The results from these three demultiplexing methods are saved in the filtered and processed `SingleCellExperiment` objects.

If a library has associated HTO data, an additional section is included in the QC report output by `scpca-nf`.
This section summarizes HTO-specific library statistics, such as how many cells express each HTO.
No additional plots are produced, but a table summarizing the results from all three demultiplexing methods is included. 

For some samples, multiple libraries were collected, with the additional libraries being used for  bulk RNA-seq and/or spatial transcriptomics. 
Both of these additional sequencing methods are supported by `scpca-nf`. 
`scpca-nf` takes FASTQ from bulk RNA-seq as input, trims reads using `fastp`, and then aligns reads with `salmon` (Supplemental Figure 3A). 
The output is single tsv file with the sample by gene counts matrix for all samples in a given ScPCA project.
This sample by gene matrix is included only with project downloads on the Portal. 

To quantify spatial transcriptomics data, `scpca-nf` takes the RNA FASTQ and slide image as input (Supplemental Figure 3B). 
As there is not yet support for spatial transcriptomics with `alevin-fry`, `scpca-nf` uses Space Ranger to quantify all spatial transcriptomics data [@url:https://www.10xgenomics.com/support/software/space-ranger/latest].
The output includes the spot by gene matrix along with a summary report produced by Space Ranger.

## Downloading projects from the ScPCA Portal

On the Portal, users can select to download data from individual samples or all data from an entire ScPCA project. 
When downloading data for an entire project, users can choose between receiving the individual files for each sample (default) or one file containing the gene expression data and metadata for all samples in the project.
Users also have the option to choose their desired format and receive the data as `SingleCellExperiment` (`.rds`) or `AnnData` (`.hdf5`) objects.
If the download contains all samples as individual files, the download folder will include a sub-folder for each sample in the project (Figure 3A).
Each sample folder contains all three object types (unfiltered, filtered, and processed) as either `SingleCellExperiment` (`.rds`) or `AnnData` (`.hdf5`) objects for all libraries from the given sample. 
These objects house the summarized gene expression data and associated metadata for the library indicated in the filename. 

Providing data for all libraries within a single file makes it easier for users to perform joint analyses on multiple samples simultaneously. 
Specifically, these objects can be used for comparing gene-level metrics across multiple samples, such as differential expression analysis and gene set enrichment analysis.
Therefore, we make a single, merged object available for each project.
This merged object contains all raw and normalized gene expression data and metadata for all single-cell and single-nuclei RNA-seq libraries within a given ScPCA project.
The data in the merged object has simply been combined, and no batch-corrected or integrated data is included. 
If downloading data from a ScPCA project as a single, merged file, the download will include a single `.rds` or `.hdf5` file, depending on the chosen file format, a summary report for the merged object, and a folder with all individual QC and cell type reports for each library found in the merged object (Figure 3B).

Regardless of downloading all samples in a project as individual files or as a merged object, the download will include a metadata file, `single_cell_metadata.tsv`, containing relevant metadata for all samples, and a `README.md`.  
If the chosen ScPCA project has samples with bulk RNA-seq, two additional files are included: a sample by gene counts matrix (`bulk_quant.tsv`) with the quantified gene expression data for all samples in the project, and a metadata file (`bulk_metadata.tsv`). 

To build the merged objects, we created an additional stand-alone workflow for merging the output from `scpca-nf`, `merge.nf` (Figure 3C).
`merge.nf` takes as input the processed `SingleCellExperiment` objects output by `scpca-nf` for all single-cell and single-nuclei libraries included in a given ScPCA project.
The gene expression data stored in all `SingleCellExperiment` objects are then merged to produce a single merged gene by cell counts matrix. 
This matrix contains all cells from all libraries and all shared genes.
All objects on the Portal were quantified using the same index with the same set of genes, so the genes found in the merged object will be the same as those found in each individual object.
Any metadata found in the individual processed objects will also be merged (e.g., `colData`, `rowData`, and `metadata`). 
The merged normalized counts matrix is used to select high-variance genes in a library-aware manner before performing dimensionality reduction with both PCA and UMAP.
`merge.nf` outputs the merged and processed object as a `SingleCellExperiment` object.

If at least one library in a project contains ADT data, the raw and normalized ADT data are also merged and saved as an `altExp` in the merged `SingleCellExperiment` object. 
If any libraries in a project are multiplexed, the HTO data is not merged and is not included in the merged object. 
All merged `SingleCellExperiment` objects are converted to `AnnData` objects and exported as `.hdf5` files. 
If the merged object contains an `altExp` with merged ADT data, two `AnnData` objects are exported to create separate RNA (`_rna.hdf5`) and ADT (`_adt.hdf5`) objects.

`merge.nf` outputs a summary report for each merged object. 
This report includes a brief summary of the types of samples and libraries included in the project, including tables showing the total number of samples for each diagnosis and the number of libraries from each library type represented in the project. 
A faceted UMAP showing all cells from all libraries is also shown. 
Here, each panel represents a different library included in the merged object. 
All cells from the specified library shown in that panel are colored, while all other cells are gray. 
An example of this UMAP showing a subset of libraries from an example ScPCA project is available in Figure 3D. 


  
