## Materials and Methods

### Data generation
  - how data was generated in different labs using 10X and then sent to the Data Lab

### Data processing (do we need this section?)
  - Mention that all data was processing using `scpca-nf` either by us or external submitters

### Processing single-cell and single-nuclei RNA-seq data with alevin-fry
  - Use of salmon alevin and alevin-fry to process all raw FASTQ files
  - Information on index used
  - Parameter choices for alevin-fry

### Post alevin-fry processing of single-cell and single-nuclei RNA-seq data
  - filtering of empty droplets
  - removal of low quality cells
  - normalization
  - HVG selection
  - PCA and UMAP calculation

### Quantifying gene expression for libraries with CITE-seq or cell hashing
  - How we used alevin-fry to quantify ADT and HTO libraries

### Processing CITE-seq expression data
  - Filtering low quality cells based on ADT data
  - Normalization of ADT data

### Genetic demultiplexing
  - Use of vireo and matching bulk RNA-seq

### HTO demultiplexing
  - Seurat
  - DropletUtils

### Quantification of spatial transcriptomics data
  - Use of space ranger

### Quantification of bulk RNA-seq data
  - Use of salmon

### Cell type annotation
  - Implementation of SingleR and CellAssign
  - Description of metrics used (e.g., what is the delta median and where does the probability come from)

### Generating merged data
  - combining counts data and metadata

### Converting SingleCellExperiment objects to AnnData objects
  - use of zellkonverter


There are probably more sections here we might want to include about the Portal itself? Anything related to building it?
Do we need to mention anything about the terms of use with the data here?

### Code and data availability

### Limitations section (do we need this?)
